{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import math\n",
    "import seaborn as sns\n",
    "import CPTAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter CPTAC.embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import CPTAC\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Proteomics_Participant_ID', 'Proteomics_TMT_batch',\n",
       "       'Proteomics_TMT_plex', 'Proteomics_TMT_channel',\n",
       "       'Proteomics_Parent_Sample_IDs', 'Proteomics_Aliquot_ID',\n",
       "       'Proteomics_Tumor_Normal', 'Proteomics_OCT', 'Country',\n",
       "       'Histologic_Grade_FIGO',\n",
       "       ...\n",
       "       'RNAseq_R1_sample_type', 'RNAseq_R1_filename', 'RNAseq_R1_UUID',\n",
       "       'RNAseq_R2_sample_type', 'RNAseq_R2_filename', 'RNAseq_R2_UUID',\n",
       "       'miRNAseq_sample_type', 'miRNAseq_UUID', 'Methylation_available',\n",
       "       'Methylation_quality'],\n",
       "      dtype='object', length=162)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### step 1 getting you clinical data set up right\n",
    "\n",
    "clinical_full = CPTAC.get_clinical() \n",
    "clinical_full.columns\n",
    "###### fix the above import to get only certian columns\n",
    "clinical = clinical_full[['Histologic_Grade_FIGO']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Proteomics_TMT_batch</th>\n",
       "      <th>Proteomics_TMT_plex</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Parent_Sample_IDs</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Proteomics_OCT</th>\n",
       "      <th>Country</th>\n",
       "      <th>Histologic_Grade_FIGO</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_R1_sample_type</th>\n",
       "      <th>RNAseq_R1_filename</th>\n",
       "      <th>RNAseq_R1_UUID</th>\n",
       "      <th>RNAseq_R2_sample_type</th>\n",
       "      <th>RNAseq_R2_filename</th>\n",
       "      <th>RNAseq_R2_UUID</th>\n",
       "      <th>miRNAseq_sample_type</th>\n",
       "      <th>miRNAseq_UUID</th>\n",
       "      <th>Methylation_available</th>\n",
       "      <th>Methylation_quality</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>idx</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>128N</td>\n",
       "      <td>C3L-00006-01</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 1</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8a1efc47-1c29-417f-a425-cdbd09565dcb</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>37bcba98-1094-459e-83ae-c23a602416fb</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3L-00008-01</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 1</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>555725e8-cba5-4676-9b0a-80100cbf9f47</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>15235b12-b67a-4678-acc4-ed03d642bd5e</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>492b50d8-ec35-46e7-a65d-06512aaee394</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>131</td>\n",
       "      <td>C3L-00032-01</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 2</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>9ae968f3-691d-4db3-9977-1ab3e5af9085</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>423b6b09-02aa-4f47-9241-f75c1dad1161</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>1794ff56-db2d-4d1a-8758-cab7fe3d98c1</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S004</th>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>129N</td>\n",
       "      <td>C3L-00084-01</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>b0a7cdf2-2ad8-4442-91b0-548ea4975554</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>c83987a5-1c13-4af4-b46c-218fe5f60c34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3L-00090-01</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 2</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>8ce5618d-9ff6-40f9-aeea-8d8e1633ae38</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>06d3fd4a-a623-4146-8500-4f1f17235253</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>a6524c2d-d7dd-4629-980e-b45dbdc92c49</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 162 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Proteomics_Participant_ID  Proteomics_TMT_batch  Proteomics_TMT_plex  \\\n",
       "idx                                                                         \n",
       "S001                 C3L-00006                     2                    5   \n",
       "S002                 C3L-00008                     4                   16   \n",
       "S003                 C3L-00032                     1                    2   \n",
       "S004                 C3L-00084                     3                   11   \n",
       "S005                 C3L-00090                     3                   12   \n",
       "\n",
       "     Proteomics_TMT_channel Proteomics_Parent_Sample_IDs  \\\n",
       "idx                                                        \n",
       "S001                   128N                 C3L-00006-01   \n",
       "S002                   130N                 C3L-00008-01   \n",
       "S003                    131                 C3L-00032-01   \n",
       "S004                   129N                 C3L-00084-01   \n",
       "S005                   129C                 C3L-00090-01   \n",
       "\n",
       "     Proteomics_Aliquot_ID Proteomics_Tumor_Normal Proteomics_OCT  \\\n",
       "idx                                                                 \n",
       "S001         CPT0001460012                   Tumor             No   \n",
       "S002         CPT0001300009                   Tumor             No   \n",
       "S003         CPT0001420009                   Tumor             No   \n",
       "S004         CPT0000820012                   Tumor             No   \n",
       "S005         CPT0001140003                   Tumor             No   \n",
       "\n",
       "            Country Histologic_Grade_FIGO         ...          \\\n",
       "idx                                               ...           \n",
       "S001  United States          FIGO grade 1         ...           \n",
       "S002  United States          FIGO grade 1         ...           \n",
       "S003  United States          FIGO grade 2         ...           \n",
       "S004            NaN                   NaN         ...           \n",
       "S005  United States          FIGO grade 2         ...           \n",
       "\n",
       "     RNAseq_R1_sample_type                                 RNAseq_R1_filename  \\\n",
       "idx                                                                             \n",
       "S001                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "S002                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "S003                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "S004                 Tumor  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "S005                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "\n",
       "                            RNAseq_R1_UUID RNAseq_R2_sample_type  \\\n",
       "idx                                                                \n",
       "S001  8a1efc47-1c29-417f-a425-cdbd09565dcb                 Tumor   \n",
       "S002  555725e8-cba5-4676-9b0a-80100cbf9f47                 Tumor   \n",
       "S003  9ae968f3-691d-4db3-9977-1ab3e5af9085                 Tumor   \n",
       "S004  b0a7cdf2-2ad8-4442-91b0-548ea4975554                 Tumor   \n",
       "S005  8ce5618d-9ff6-40f9-aeea-8d8e1633ae38                 Tumor   \n",
       "\n",
       "                                     RNAseq_R2_filename  \\\n",
       "idx                                                       \n",
       "S001  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "S002  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "S003  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "S004  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "S005  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "\n",
       "                            RNAseq_R2_UUID miRNAseq_sample_type  \\\n",
       "idx                                                               \n",
       "S001  8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda                Tumor   \n",
       "S002  15235b12-b67a-4678-acc4-ed03d642bd5e                Tumor   \n",
       "S003  423b6b09-02aa-4f47-9241-f75c1dad1161                Tumor   \n",
       "S004  c83987a5-1c13-4af4-b46c-218fe5f60c34                  NaN   \n",
       "S005  06d3fd4a-a623-4146-8500-4f1f17235253                Tumor   \n",
       "\n",
       "                             miRNAseq_UUID Methylation_available  \\\n",
       "idx                                                                \n",
       "S001  37bcba98-1094-459e-83ae-c23a602416fb                   YES   \n",
       "S002  492b50d8-ec35-46e7-a65d-06512aaee394                   YES   \n",
       "S003  1794ff56-db2d-4d1a-8758-cab7fe3d98c1                   YES   \n",
       "S004                                   NaN                   YES   \n",
       "S005  a6524c2d-d7dd-4629-980e-b45dbdc92c49                   YES   \n",
       "\n",
       "     Methylation_quality  \n",
       "idx                       \n",
       "S001                PASS  \n",
       "S002                PASS  \n",
       "S003                PASS  \n",
       "S004                PASS  \n",
       "S005                PASS  \n",
       "\n",
       "[5 rows x 162 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Step 2 getting TP53 mutation status done in a simple df\n",
    "\n",
    "\n",
    "somatic = CPTAC.get_somatic()\n",
    "gene = 'TP53' \n",
    "protein = 'TP53'\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hotspot_mutations = ['p.P250L', 'p.R248W', 'p.S241C', 'p.S241del', 'p.K164E', 'p.R249S', 'p.R273H', 'p.R282W',\n",
    "                    'p.S127Y', 'p.G244D', 'p.A159P']\n",
    "mutated_hotspot = somatic.loc[(somatic['Location'].isin(hotspot_mutations)) & (somatic['Gene'] == gene)]\n",
    "hotspot_patients = mutated_hotspot['Clinical_Patient_Key']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Proteomics_Participant_ID\n",
      "Power_divergenceResult(statistic=110.99999999999999, pvalue=nan)\n",
      "Proteomics_TMT_batch\n",
      "Ttest_indResult(statistic=0.5417103612376254, pvalue=0.5888164733972103)\n",
      "Proteomics_TMT_plex\n",
      "Ttest_indResult(statistic=0.24396540137364253, pvalue=0.8075887754475264)\n",
      "Proteomics_TMT_channel\n",
      "Power_divergenceResult(statistic=3.0000000000000004, pvalue=nan)\n",
      "Proteomics_Parent_Sample_IDs\n",
      "Power_divergenceResult(statistic=141.0, pvalue=nan)\n",
      "Proteomics_Aliquot_ID\n",
      "Power_divergenceResult(statistic=141.0, pvalue=nan)\n",
      "Proteomics_Tumor_Normal\n",
      "Power_divergenceResult(statistic=36.0, pvalue=nan)\n",
      "Proteomics_OCT\n",
      "Power_divergenceResult(statistic=3.0, pvalue=nan)\n",
      "Country\n",
      "Power_divergenceResult(statistic=4.666666666666666, pvalue=nan)\n",
      "Histologic_Grade_FIGO\n",
      "Power_divergenceResult(statistic=0.4, pvalue=nan)\n",
      "Myometrial_invasion_Specify\n",
      "Power_divergenceResult(statistic=8.0, pvalue=nan)\n",
      "Histologic_type\n",
      "Power_divergenceResult(statistic=8.666666666666666, pvalue=nan)\n",
      "Treatment_naive\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "Tumor_purity\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "Case_quality\n",
      "Power_divergenceResult(statistic=8.333333333333334, pvalue=nan)\n",
      "Path_Stage_Primary_Tumor-pT\n",
      "Power_divergenceResult(statistic=21.0, pvalue=nan)\n",
      "Path_Stage_Reg_Lymph_Nodes-pN\n",
      "Power_divergenceResult(statistic=3.3333333333333335, pvalue=nan)\n",
      "Clin_Stage_Dist_Mets-cM\n",
      "Power_divergenceResult(statistic=13.5, pvalue=nan)\n",
      "Path_Stage_Dist_Mets-pM\n",
      "Power_divergenceResult(statistic=10.5, pvalue=nan)\n",
      "tumor_Stage-Pathological\n",
      "Power_divergenceResult(statistic=18.0, pvalue=nan)\n",
      "FIGO_stage\n",
      "Power_divergenceResult(statistic=26.666666666666668, pvalue=nan)\n",
      "LVSI\n",
      "Ttest_indResult(statistic=3.1125923663236543, pvalue=0.002424381623858236)\n",
      "BMI\n",
      "Ttest_indResult(statistic=-2.058110623823925, pvalue=0.042206970081571746)\n",
      "Age\n",
      "Ttest_indResult(statistic=0.8961775408729353, pvalue=0.3723319180019674)\n",
      "Diabetes\n",
      "Power_divergenceResult(statistic=16.54545454545455, pvalue=nan)\n",
      "Race\n",
      "Power_divergenceResult(statistic=27.999999999999993, pvalue=nan)\n",
      "Ethnicity\n",
      "Power_divergenceResult(statistic=13.999999999999998, pvalue=nan)\n",
      "Gender\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "Tumor_Site\n",
      "Power_divergenceResult(statistic=10.5, pvalue=nan)\n",
      "Tumor_Site_Other\n",
      "Power_divergenceResult(statistic=30.000000000000004, pvalue=nan)\n",
      "Tumor_Focality\n",
      "Power_divergenceResult(statistic=1.3333333333333333, pvalue=nan)\n",
      "Tumor_Size_cm\n",
      "Ttest_indResult(statistic=0.5019597130458164, pvalue=0.616821782294074)\n",
      "Estrogen_Receptor\n",
      "Power_divergenceResult(statistic=6.666666666666667, pvalue=nan)\n",
      "Estrogen_Receptor_%\n",
      "Ttest_indResult(statistic=-1.8619593538672972, pvalue=0.07078769662973088)\n",
      "Progesterone_Receptor\n",
      "Power_divergenceResult(statistic=6.0, pvalue=nan)\n",
      "Progesterone_Receptor_%\n",
      "Ttest_indResult(statistic=-2.005156704049731, pvalue=0.05295943867167082)\n",
      "MLH1\n",
      "Power_divergenceResult(statistic=12.666666666666666, pvalue=nan)\n",
      "MLH2\n",
      "Power_divergenceResult(statistic=12.666666666666666, pvalue=nan)\n",
      "MSH6\n",
      "Power_divergenceResult(statistic=8.666666666666666, pvalue=nan)\n",
      "PMS2\n",
      "Power_divergenceResult(statistic=8.666666666666666, pvalue=nan)\n",
      "p53\n",
      "Power_divergenceResult(statistic=18.83333333333333, pvalue=nan)\n",
      "Other_IHC_specify\n",
      "Power_divergenceResult(statistic=25.000000000000014, pvalue=nan)\n",
      "MLH1_Promoter_Hypermethylation\n",
      "Power_divergenceResult(statistic=24.0, pvalue=nan)\n",
      "Num_full_term_pregnancies\n",
      "Power_divergenceResult(statistic=4.0, pvalue=nan)\n",
      "EPIC_Bcells\n",
      "Ttest_indResult(statistic=1.3236864142260267, pvalue=0.18865572103568126)\n",
      "EPIC_CAFs\n",
      "Ttest_indResult(statistic=-1.5187663109668772, pvalue=0.13200849499668596)\n",
      "EPIC_CD4_Tcells\n",
      "Ttest_indResult(statistic=0.24647696422415638, pvalue=0.8058233773495231)\n",
      "EPIC_CD8_Tcells\n",
      "Ttest_indResult(statistic=-1.387473521185667, pvalue=0.16841274793897115)\n",
      "EPIC_Endothelial\n",
      "Ttest_indResult(statistic=-1.5635826470823477, pvalue=0.12110667382383634)\n",
      "EPIC_Macrophages\n",
      "Ttest_indResult(statistic=1.5830508956628215, pvalue=0.11659936454712048)\n",
      "EPIC_NKcells\n",
      "Ttest_indResult(statistic=-0.7465201039401458, pvalue=0.4571224383009178)\n",
      "EPIC_otherCells\n",
      "Ttest_indResult(statistic=1.8840049695749943, pvalue=0.06249851347854914)\n",
      "CIBERSORT_B _cells _naive\n",
      "Ttest_indResult(statistic=0.4595360710326206, pvalue=0.6468567756344638)\n",
      "CIBERSORT_B _cells _memory\n",
      "Ttest_indResult(statistic=1.3122793977798437, pvalue=0.1924603233862566)\n",
      "CIBERSORT_Plasma _cells\n",
      "Ttest_indResult(statistic=0.8900470796370993, pvalue=0.3755984015798566)\n",
      "CIBERSORT_T _cells _CD8\n",
      "Ttest_indResult(statistic=1.0631210182877586, pvalue=0.29031342188502884)\n",
      "CIBERSORT_T _cells _CD4 _naive\n",
      "Ttest_indResult(statistic=nan, pvalue=nan)\n",
      "CIBERSORT_T _cells _CD4 _memory _resting\n",
      "Ttest_indResult(statistic=-2.143034953838503, pvalue=0.034561117880055434)\n",
      "CIBERSORT_T _cells _CD4 _memory _activated\n",
      "Ttest_indResult(statistic=1.6793241835237211, pvalue=0.09624236181496836)\n",
      "CIBERSORT_T _cells _follicular _helper\n",
      "Ttest_indResult(statistic=1.1516363770821443, pvalue=0.25224415036130254)\n",
      "CIBERSORT_T _cells _regulatory _(Tregs)\n",
      "Ttest_indResult(statistic=-0.4780512623805488, pvalue=0.6336676434741593)\n",
      "CIBERSORT_T _cells _gamma _delta\n",
      "Ttest_indResult(statistic=-0.8859823867038107, pvalue=0.37777406044096695)\n",
      "CIBERSORT_NK _cells _resting\n",
      "Ttest_indResult(statistic=-0.9451134817704964, pvalue=0.3469019224799048)\n",
      "CIBERSORT_NK _cells _activated\n",
      "Ttest_indResult(statistic=2.3712599757418733, pvalue=0.019662994233101878)\n",
      "CIBERSORT_Monocytes\n",
      "Ttest_indResult(statistic=-2.0194452027015477, pvalue=0.04614385925175008)\n",
      "CIBERSORT_Macrophages _M0\n",
      "Ttest_indResult(statistic=1.8331270663311066, pvalue=0.06978874185650309)\n",
      "CIBERSORT_Macrophages _M1\n",
      "Ttest_indResult(statistic=2.3346379979698604, pvalue=0.021582001178565315)\n",
      "CIBERSORT_Macrophages _M2\n",
      "Ttest_indResult(statistic=0.84195475114366, pvalue=0.4018425898203699)\n",
      "CIBERSORT_Dendritic _cells _resting\n",
      "Ttest_indResult(statistic=-1.4307878558813392, pvalue=0.15563996746776487)\n",
      "CIBERSORT_Dendritic _cells _activated\n",
      "Ttest_indResult(statistic=0.7448911503626491, pvalue=0.45810220589929773)\n",
      "CIBERSORT_Mast _cells _resting\n",
      "Ttest_indResult(statistic=-2.1312102080603217, pvalue=0.03554801579611658)\n",
      "CIBERSORT_Mast _cells _activated\n",
      "Ttest_indResult(statistic=1.1964454786881489, pvalue=0.23438024402503801)\n",
      "CIBERSORT_Eosinophils\n",
      "Ttest_indResult(statistic=-0.1195563668209847, pvalue=0.9050768653729697)\n",
      "CIBERSORT_Neutrophils\n",
      "Ttest_indResult(statistic=-0.06905148135504909, pvalue=0.945087921774973)\n",
      "CIBERSORT_Absolute _score\n",
      "Ttest_indResult(statistic=1.0493923382730896, pvalue=0.2965526042101489)\n",
      "ESTIMATE_StromalScore\n",
      "Ttest_indResult(statistic=-0.5183140417486213, pvalue=0.605395680461335)\n",
      "ESTIMATE_ImmuneScore\n",
      "Ttest_indResult(statistic=1.171648429103034, pvalue=0.24415029241733235)\n",
      "ESTIMATE_ESTIMATEScore\n",
      "Ttest_indResult(statistic=0.4109757161857539, pvalue=0.6819785718671744)\n",
      "Stemness_score\n",
      "Ttest_indResult(statistic=3.5666892675485333, pvalue=0.0005331321142559024)\n",
      "Pathway_activity_EGFR\n",
      "Ttest_indResult(statistic=-0.33777014457731663, pvalue=0.7362515732690988)\n",
      "Pathway_activity_Hypoxia\n",
      "Ttest_indResult(statistic=0.05242222541507055, pvalue=0.9582978979502212)\n",
      "Pathway_activity_JAK.STAT\n",
      "Ttest_indResult(statistic=4.417597395431041, pvalue=2.554104145672744e-05)\n",
      "Pathway_activity_MAPK\n",
      "Ttest_indResult(statistic=-0.4673454035246565, pvalue=0.6412798088789067)\n",
      "Pathway_activity_NFkB\n",
      "Ttest_indResult(statistic=2.151417634548559, pvalue=0.033875968966394894)\n",
      "Pathway_activity_PI3K\n",
      "Ttest_indResult(statistic=1.3356308618574708, pvalue=0.1847325164480498)\n",
      "Pathway_activity_TGFb\n",
      "Ttest_indResult(statistic=-0.6898388866737868, pvalue=0.4919094226686238)\n",
      "Pathway_activity_TNFa\n",
      "Ttest_indResult(statistic=1.5201746022906033, pvalue=0.1316545472585252)\n",
      "Pathway_activity_Trail\n",
      "Ttest_indResult(statistic=0.9809684861654098, pvalue=0.3289994982978355)\n",
      "Pathway_activity_VEGF\n",
      "Ttest_indResult(statistic=0.39564742977962075, pvalue=0.6932164596781671)\n",
      "Pathway_activity_p53\n",
      "Ttest_indResult(statistic=-3.664586805471489, pvalue=0.00040038181894262216)\n",
      "TP53_ATM\n",
      "Ttest_indResult(statistic=-1.0649731165230572, pvalue=0.2894274613582694)\n",
      "TP53_CHEK2\n",
      "Ttest_indResult(statistic=-0.7355345320745459, pvalue=0.463718676077184)\n",
      "TP53_MDM4\n",
      "Ttest_indResult(statistic=-0.36158586334310605, pvalue=0.7184168669720088)\n",
      "TP53_RPS6KA3\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "TP53_TP53\n",
      "Ttest_indResult(statistic=8.402496161711282, pvalue=2.8878593567722003e-13)\n",
      "TP53_pathway\n",
      "Ttest_indResult(statistic=6.074994687977802, pvalue=2.2051007801644186e-08)\n",
      "PI3K_AKT1\n",
      "Ttest_indResult(statistic=0.8435072436666403, pvalue=0.40093813141872225)\n",
      "PI3K_AKT2\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_AKT3\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "PI3K_DEPDC5\n",
      "Ttest_indResult(statistic=-0.7355345320745459, pvalue=0.463718676077184)\n",
      "PI3K_DEPTOR\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_INPP4B\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "PI3K_MAPKAP1\n",
      "Ttest_indResult(statistic=-0.7355345320745459, pvalue=0.463718676077184)\n",
      "PI3K_MLST8\n",
      "Ttest_indResult(statistic=-0.361585863343106, pvalue=0.7184168669720088)\n",
      "PI3K_MTOR\n",
      "Ttest_indResult(statistic=-1.3369334053416047, pvalue=0.18424790523932905)\n",
      "PI3K_NPRL2\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_NPRL3\n",
      "Ttest_indResult(statistic=-0.6333619788817667, pvalue=0.5279288882983969)\n",
      "PI3K_PDK1\n",
      "Ttest_indResult(statistic=-0.36158586334310605, pvalue=0.7184168669720088)\n",
      "PI3K_PIK3CA\n",
      "Ttest_indResult(statistic=-1.046182856128859, pvalue=0.29797391567908843)\n",
      "PI3K_PIK3CB\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "PI3K_PIK3R1\n",
      "Ttest_indResult(statistic=-1.5478604841714108, pvalue=0.12478370190404489)\n",
      "PI3K_PIK3R2\n",
      "Ttest_indResult(statistic=-0.05228054025945097, pvalue=0.9584084218476157)\n",
      "PI3K_PPP2R1A\n",
      "Ttest_indResult(statistic=-0.27734633064080044, pvalue=0.7820810601161129)\n",
      "PI3K_PTEN\n",
      "Ttest_indResult(statistic=-3.028204420366421, pvalue=0.003122781667343892)\n",
      "PI3K_RHEB\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_RICTOR\n",
      "Ttest_indResult(statistic=-0.7355345320745459, pvalue=0.463718676077184)\n",
      "PI3K_RPS6\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_RPS6KB1\n",
      "Ttest_indResult(statistic=-0.36158586334310605, pvalue=0.7184168669720088)\n",
      "PI3K_RPTOR\n",
      "Ttest_indResult(statistic=-0.5142244141600536, pvalue=0.6082184263455863)\n",
      "PI3K_STK11\n",
      "Ttest_indResult(statistic=-0.36158586334310605, pvalue=0.7184168669720088)\n",
      "PI3K_TSC1\n",
      "Ttest_indResult(statistic=0.5917952126914056, pvalue=0.555310098726014)\n",
      "PI3K_TSC2\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "PI3K_pathway\n",
      "Ttest_indResult(statistic=-4.281796271532132, pvalue=4.230291642636117e-05)\n",
      "HRD_BRCA1\n",
      "Ttest_indResult(statistic=-0.9902436691399973, pvalue=0.3244214528535031)\n",
      "HRD_BRCA2\n",
      "Ttest_indResult(statistic=-0.8271199081845632, pvalue=0.4101190006367572)\n",
      "HRD_BRCA1_or_BRCA2\n",
      "Ttest_indResult(statistic=-1.2052862111436864, pvalue=0.23090915897353362)\n",
      "Log2_SNP\n",
      "Ttest_indResult(statistic=-1.1842984863600587, pvalue=0.23907446849846975)\n",
      "Log2_INDEL\n",
      "Ttest_indResult(statistic=-1.8261099195087873, pvalue=0.07078800747287779)\n",
      "Mutation_signature_C>A\n",
      "Ttest_indResult(statistic=-0.8247530172126227, pvalue=0.4114554362774979)\n",
      "Mutation_signature_C>G\n",
      "Ttest_indResult(statistic=3.6963906062966116, pvalue=0.0003555868021822147)\n",
      "Mutation_signature_C>T\n",
      "Ttest_indResult(statistic=-2.046406987675715, pvalue=0.043314230959521416)\n",
      "Mutation_signature_T>C\n",
      "Ttest_indResult(statistic=-1.830938624481076, pvalue=0.07005795989575032)\n",
      "Mutation_signature_T>A\n",
      "Ttest_indResult(statistic=3.2690443776392644, pvalue=0.001475868439105736)\n",
      "Mutation_signature_T>G\n",
      "Ttest_indResult(statistic=1.7187895769159676, pvalue=0.08871639514979729)\n",
      "CNV_clustering\n",
      "Power_divergenceResult(statistic=3.0, pvalue=nan)\n",
      "Purity_Immune\n",
      "Ttest_indResult(statistic=1.1574952179594644, pvalue=0.24997205968071176)\n",
      "Purity_Cancer\n",
      "Ttest_indResult(statistic=0.5914588342741838, pvalue=0.5556178333111768)\n",
      "Purity_Stroma\n",
      "Ttest_indResult(statistic=-1.180898425988845, pvalue=0.2405911569437833)\n",
      "MSI_status\n",
      "Power_divergenceResult(statistic=8.333333333333334, pvalue=nan)\n",
      "POLE_subtype\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "JAK1_MS_INDEL\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "JAK1_Mutation\n",
      "Power_divergenceResult(statistic=71.99999999999999, pvalue=nan)\n",
      "WXS_normal_sample_type\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "WXS_normal_filename\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WXS_normal_UUID\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n",
      "WXS_tumor_sample_type\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "WXS_tumor_filename\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n",
      "WXS_tumor_UUID\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n",
      "WGS_normal_sample_type\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "WGS_normal_UUID\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n",
      "WGS_tumor_sample_type\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "WGS_tumor_UUID\n",
      "Power_divergenceResult(statistic=90.99999999999999, pvalue=nan)\n",
      "RNAseq_R1_sample_type\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "RNAseq_R1_filename\n",
      "Power_divergenceResult(statistic=105.0, pvalue=nan)\n",
      "RNAseq_R1_UUID\n",
      "Power_divergenceResult(statistic=105.0, pvalue=nan)\n",
      "RNAseq_R2_sample_type\n",
      "Power_divergenceResult(statistic=12.0, pvalue=nan)\n",
      "RNAseq_R2_filename\n",
      "Power_divergenceResult(statistic=105.0, pvalue=nan)\n",
      "RNAseq_R2_UUID\n",
      "Power_divergenceResult(statistic=105.0, pvalue=nan)\n",
      "miRNAseq_sample_type\n",
      "Power_divergenceResult(statistic=8.0, pvalue=nan)\n",
      "miRNAseq_UUID\n",
      "Power_divergenceResult(statistic=96.0, pvalue=nan)\n",
      "Methylation_available\n",
      "Power_divergenceResult(statistic=0.0, pvalue=nan)\n",
      "Methylation_quality\n",
      "Power_divergenceResult(statistic=10.0, pvalue=nan)\n"
     ]
    }
   ],
   "source": [
    "clinical['Within_Hotspot'] = False\n",
    "clinical.loc[hotspot_patients, 'Within_Hotspot'] = True\n",
    "\n",
    "hotspot = clinical.loc[clinical['Within_Hotspot'] == True]\n",
    "non_hotspot = clinical.loc[clinical['Within_Hotspot'] == False]\n",
    "TTest = 0\n",
    "ChiSquared = 0\n",
    "for feature in clinical:\n",
    "    if str(clinical[feature].dtypes) == 'float64' or str(clinical[feature].dtypes) == 'int64':\n",
    "        ttest = scipy.stats.ttest_ind(hotspot[feature].dropna(axis=0), non_hotspot[feature].dropna(axis = 0))\n",
    "        print(feature)\n",
    "        print(ttest)\n",
    "        TTest += 1\n",
    "    elif str(clinical[feature].dtype) == 'object':\n",
    "        cat_num = 0\n",
    "        cat_dict = {}\n",
    "        index = 1\n",
    "        chi_obs = list()\n",
    "        test = 0\n",
    "        \n",
    "        for value in clinical[feature].dropna(axis = 0):\n",
    "            if value in cat_dict:\n",
    "                test += 1\n",
    "            else:\n",
    "                cat_num += 1\n",
    "                cat_dict[value] = cat_num\n",
    "                \n",
    "        while index <= cat_num:\n",
    "            cat_obs = 0\n",
    "            for value in hotspot[feature].dropna(axis = 0):\n",
    "                if cat_dict[value] == index:\n",
    "                    cat_obs +=1\n",
    "            index += 1\n",
    "            chi_obs.append(cat_obs)\n",
    "        dof = cat_num - 1\n",
    "        \n",
    "        \n",
    "        Chi_Test = scipy.stats.chisquare(chi_obs, f_exp=None, ddof=dof, axis=0)\n",
    "        print(feature)\n",
    "        print(Chi_Test)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p-value < 0.01\n",
    "LVSI\n",
    "Stemness_Score\n",
    "Pathway_activity_p53\n",
    "PI3K_PTEN\n",
    "Mutation_signature_C>G\n",
    "Mutation_signature_T>A\n",
    "\n",
    "p-value < 0.05\n",
    "BMI\n",
    "CIBERSORT_T _cells _CD4 _memory _resting\n",
    "CIBERSORT_NK _cells _activated\n",
    "CIBERSORT_Monocytes\n",
    "CIBERSORT_Macrophages _M1\n",
    "CIBERSORT_Mast _cells _resting\n",
    "Pathway_activity_NFkB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Binary specification of whether the specified patient has lymphovascular space invasion.\n"
     ]
    }
   ],
   "source": [
    "CPTAC.define(\"LVSI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "###Tumor Attributes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jonjarman/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/Users/jonjarman/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:543: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item] = s\n"
     ]
    }
   ],
   "source": [
    "interest = list()\n",
    "for header in columns:\n",
    "    if header[0:5] == 'Tumor':\n",
    "        interest.append(header)\n",
    "        \n",
    "        \n",
    "p53_mutation = CPTAC.get_cohort_clinical(interest)\n",
    "p53_mutation['Within_Hotspot'] = False\n",
    "p53_mutation.loc[hotspot_patients, 'Within_Hotspot'] = True\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ttest_indResult(statistic=0.38204990429970553, pvalue=0.7035655006095476)\n"
     ]
    }
   ],
   "source": [
    "hotspot = p53_mutation.loc[p53_mutation['Within_Hotspot'] == True]\n",
    "non_hotspot = p53_mutation.loc[p53_mutation['Within_Hotspot'] == False]\n",
    "#hotspot = hotspot.dropna(axis=0,how = 'any')\n",
    "#non_hotspsot = non_hotspot.dropna(axis=0, how = 'any')\n",
    "ttest = scipy.stats.ttest_ind(hotspot['Tumor_Size_cm'], non_hotspot['Tumor_Size_cm'])\n",
    "print(ttest)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "###Building a Chi Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FIGO</th>\n",
       "      <th>S001</th>\n",
       "      <th>S002</th>\n",
       "      <th>S003</th>\n",
       "      <th>S005</th>\n",
       "      <th>S007</th>\n",
       "      <th>S008</th>\n",
       "      <th>S010</th>\n",
       "      <th>S011</th>\n",
       "      <th>S012</th>\n",
       "      <th>S014</th>\n",
       "      <th>S017</th>\n",
       "      <th>S018</th>\n",
       "      <th>S019</th>\n",
       "      <th>S020</th>\n",
       "      <th>S021</th>\n",
       "      <th>S022</th>\n",
       "      <th>S023</th>\n",
       "      <th>S024</th>\n",
       "      <th>S025</th>\n",
       "      <th>S026</th>\n",
       "      <th>S027</th>\n",
       "      <th>S028</th>\n",
       "      <th>S029</th>\n",
       "      <th>S030</th>\n",
       "      <th>S031</th>\n",
       "      <th>S032</th>\n",
       "      <th>S033</th>\n",
       "      <th>S034</th>\n",
       "      <th>S035</th>\n",
       "      <th>S036</th>\n",
       "      <th>S037</th>\n",
       "      <th>S038</th>\n",
       "      <th>S039</th>\n",
       "      <th>S040</th>\n",
       "      <th>S044</th>\n",
       "      <th>S045</th>\n",
       "      <th>S046</th>\n",
       "      <th>S047</th>\n",
       "      <th>S048</th>\n",
       "      <th>S049</th>\n",
       "      <th>S050</th>\n",
       "      <th>S051</th>\n",
       "      <th>S053</th>\n",
       "      <th>S054</th>\n",
       "      <th>S055</th>\n",
       "      <th>S056</th>\n",
       "      <th>S057</th>\n",
       "      <th>S059</th>\n",
       "      <th>S060</th>\n",
       "      <th>S061</th>\n",
       "      <th>S062</th>\n",
       "      <th>S063</th>\n",
       "      <th>S064</th>\n",
       "      <th>S065</th>\n",
       "      <th>S066</th>\n",
       "      <th>S067</th>\n",
       "      <th>S068</th>\n",
       "      <th>S069</th>\n",
       "      <th>S070</th>\n",
       "      <th>S073</th>\n",
       "      <th>S074</th>\n",
       "      <th>S075</th>\n",
       "      <th>S076</th>\n",
       "      <th>S077</th>\n",
       "      <th>S078</th>\n",
       "      <th>S079</th>\n",
       "      <th>S080</th>\n",
       "      <th>S081</th>\n",
       "      <th>S082</th>\n",
       "      <th>S083</th>\n",
       "      <th>S084</th>\n",
       "      <th>S085</th>\n",
       "      <th>S086</th>\n",
       "      <th>S087</th>\n",
       "      <th>S088</th>\n",
       "      <th>S089</th>\n",
       "      <th>S090</th>\n",
       "      <th>S091</th>\n",
       "      <th>S092</th>\n",
       "      <th>S093</th>\n",
       "      <th>S094</th>\n",
       "      <th>S095</th>\n",
       "      <th>S098</th>\n",
       "      <th>S099</th>\n",
       "      <th>S100</th>\n",
       "      <th>S101</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>FIGO grade 1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIGO grade 2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIGO grade 3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FIGO          S001  S002  S003  S005  S007  S008  S010  S011  S012  S014  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     1     1     0     0     1     0     1     1     1     0   \n",
       "FIGO grade 2     0     0     1     1     0     1     0     0     0     1   \n",
       "FIGO grade 3     0     0     0     0     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S017  S018  S019  S020  S021  S022  S023  S024  S025  S026  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     1     1     1     0     1     1     1     0     1     0   \n",
       "FIGO grade 2     0     0     0     1     0     0     0     1     0     1   \n",
       "FIGO grade 3     0     0     0     0     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S027  S028  S029  S030  S031  S032  S033  S034  S035  S036  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     1     0     0     0     0     1     0     0     0     1   \n",
       "FIGO grade 2     0     1     1     0     0     0     1     1     1     0   \n",
       "FIGO grade 3     0     0     0     1     1     0     0     0     0     0   \n",
       "\n",
       "FIGO          S037  S038  S039  S040  S044  S045  S046  S047  S048  S049  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     1     0     1     0     1     0     1     0     1     1   \n",
       "FIGO grade 2     0     1     0     1     0     1     0     1     0     0   \n",
       "FIGO grade 3     0     0     0     0     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S050  S051  S053  S054  S055  S056  S057  S059  S060  S061  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     0     0     1     0     1     1     0     0     0     1   \n",
       "FIGO grade 2     1     0     0     0     0     0     1     1     1     0   \n",
       "FIGO grade 3     0     1     0     1     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S062  S063  S064  S065  S066  S067  S068  S069  S070  S073  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     1     0     0     1     0     1     0     0     0     0   \n",
       "FIGO grade 2     0     1     1     0     0     0     1     1     0     1   \n",
       "FIGO grade 3     0     0     0     0     1     0     0     0     1     0   \n",
       "\n",
       "FIGO          S074  S075  S076  S077  S078  S079  S080  S081  S082  S083  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     0     1     1     1     0     0     1     0     1     0   \n",
       "FIGO grade 2     1     0     0     0     1     1     0     1     0     1   \n",
       "FIGO grade 3     0     0     0     0     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S084  S085  S086  S087  S088  S089  S090  S091  S092  S093  \\\n",
       "id                                                                         \n",
       "FIGO grade 1     0     0     0     1     0     0     1     0     0     1   \n",
       "FIGO grade 2     1     1     1     0     1     1     0     1     1     0   \n",
       "FIGO grade 3     0     0     0     0     0     0     0     0     0     0   \n",
       "\n",
       "FIGO          S094  S095  S098  S099  S100  S101  \n",
       "id                                                \n",
       "FIGO grade 1     0     0     0     0     0     0  \n",
       "FIGO grade 2     0     1     1     1     0     1  \n",
       "FIGO grade 3     1     0     0     0     1     0  "
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "chi_cat_row = np.array(clinical.index)    \n",
    "chi_cat_column = np.array(clinical['Histologic_Grade_FIGO']) \n",
    "chi_cat = pd.Categorical(clinical['Histologic_Grade_FIGO'])\n",
    "chi_cat\n",
    "\n",
    "chi_cat_table = pd.crosstab(chi_cat_row, chi_cat, rownames=['id'], colnames=['FIGO'])\n",
    "chi_cat_table\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
