{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "   # TP53: Effects of Mutation on Interacting Proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>Standard imports for playing with and plotting data frames.</b>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>Import CPTAC data</b>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter CPTAC.embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import CPTAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "somatic_mutations = CPTAC.get_somatic()\n",
    "proteomics = CPTAC.get_proteomics()\n",
    "phos = CPTAC.get_phosphoproteomics()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List of proteins that interact with TP53 (according to Uniprot and String)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protList = ['TGFB1', 'AXIN1', 'EP300', 'HRMT1L2', 'CARM1', 'TAF1', 'ING4', 'CABLES1', \n",
    "            'TP73', 'HIPK1', 'HIPK2', 'TP53INP1', 'TP53BP', 'WWOX', 'HCV', 'USP7', 'SYVN1', 'HSP90AB1',\n",
    "           'CHD8', 'ARMC10', 'BANP', 'CDKN2AIP', 'NUAK1', 'STK11', 'LKB1', 'UHRF2', 'E4F1', 'YWHAZ',\n",
    "           'MAML1', 'MKRN1', 'PML', 'MDM2', 'FBXO42', 'ATM', 'PP2R2A', 'AURKA', 'DAXX', 'BRD7', 'TRIM24',\n",
    "           'L3MBTL1', 'GRK5', 'CAK', 'PTK2B', 'PYK2', 'MDM2', 'PRKCG', 'PPIF', 'KAT6A', 'UBC9', 'ZNF385B',\n",
    "           'ZNF385A', 'ANKRD2', 'RFFL', 'RNF34', 'MTA1', 'COP1', 'CCAR2', 'MORC3', 'POU4F2', 'AFG1L', 'UBD',\n",
    "           'TAF6', 'HPV', 'UL123', 'E1B-55K', 'BAX', 'FAS', 'BCL2', 'CREBBP', 'CDK2', 'MDM2', 'CDKN2A', 'ATM',\n",
    "            'CCNB1', 'TERT', 'IGF1R', 'ALB', 'MMP1']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## t-test for effects of missense mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "tested = [];\n",
    "p_vals = [];\n",
    "for protein in protList: \n",
    "    if protein in proteomics.columns:\n",
    "        tested.append(protein)\n",
    "        cross = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "        cross = cross.loc[cross['Patient_Type'] == 'Tumor'][[\"Mutation\", protein]].dropna(axis=0)\n",
    "        mutated = cross.loc[cross[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "        wt = cross.loc[cross[\"Mutation\"] == \"Wildtype\"]\n",
    "        ttest = scipy.stats.ttest_ind(mutated[protein], wt[protein])\n",
    "        p_vals.append(ttest[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_vals)[0]\n",
    "pvals = statsmodels.stats.multitest.fdrcorrection(p_vals)[1]\n",
    "significant_proteins = np.array(tested)[np.array(areSignificant)]\n",
    "significant_pvals = np.array(pvals)[np.array(areSignificant)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List of significantly affected proteins and their respective p-values (FDR adjusted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CABLES1' 'HSP90AB1' 'CHD8' 'CDKN2AIP' 'UHRF2' 'YWHAZ' 'AURKA' 'DAXX'\n",
      " 'AFG1L' 'TAF6' 'FAS' 'CDKN2A']\n",
      "[0.01198163 0.02933247 0.01198163 0.03426805 0.02920811 0.01198163\n",
      " 0.01036627 0.01554115 0.03943006 0.03426805 0.00441439 0.02874655]\n"
     ]
    }
   ],
   "source": [
    "print(significant_proteins)\n",
    "print(significant_pvals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## t-test for effects of frame shift and nonsense mutations (truncating)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "tested = [];\n",
    "p_vals = [];a\n",
    "for protein in protList: \n",
    "    if protein in proteomics.columns:\n",
    "        tested.append(protein)\n",
    "        cross = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "        cross = cross.loc[cross['Patient_Type'] == 'Tumor'][[\"Mutation\", protein]].dropna(axis=0)\n",
    "        mutated = cross.loc[(cross[\"Mutation\"] == \"Frame_Shift_Del\") | \n",
    "                            (cross[\"Mutation\"] == \"Frame_Shift_Ins\") |\n",
    "                            (cross[\"Mutation\"] == 'Nonsense_Mutation')]\n",
    "        wt = cross.loc[cross[\"Mutation\"] == \"Wildtype\"]\n",
    "        ttest = scipy.stats.ttest_ind(mutated[protein], wt[protein])\n",
    "        p_vals.append(ttest[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_vals)[0]\n",
    "pvals = statsmodels.stats.multitest.fdrcorrection(p_vals)[1]\n",
    "significant_proteins = np.array(tested)[np.array(areSignificant)]\n",
    "significant_pvals = np.array(pvals)[np.array(areSignificant)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CABLES1' 'CHD8' 'FAS' 'CDK2' 'CDKN2A' 'CCNB1']\n",
      "[0.0059016  0.00027794 0.0059016  0.0408439  0.00625023 0.00866919]\n"
     ]
    }
   ],
   "source": [
    "print(significant_proteins)\n",
    "print(significant_pvals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## t-test for effects of missense and truncating mutations combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "tested = [];\n",
    "p_vals = [];\n",
    "for protein in protList: \n",
    "    if protein in proteomics.columns:\n",
    "        tested.append(protein)\n",
    "        cross = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "        cross = cross.loc[cross['Patient_Type'] == 'Tumor'][[\"Mutation\", protein]].dropna(axis=0)\n",
    "        mutated = cross.loc[(cross[\"Mutation\"] == \"Frame_Shift_Del\") | \n",
    "                            (cross[\"Mutation\"] == \"Frame_Shift_Ins\") |\n",
    "                            (cross[\"Mutation\"] == 'Nonsense_Mutation') |\n",
    "                            (cross[\"Mutation\"] == 'Missense_Mutation')]\n",
    "        wt = cross.loc[cross[\"Mutation\"] == \"Wildtype\"]\n",
    "        ttest = scipy.stats.ttest_ind(mutated[protein], wt[protein])\n",
    "        p_vals.append(ttest[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_vals)[0]\n",
    "pvals = statsmodels.stats.multitest.fdrcorrection(p_vals)[1]\n",
    "significant_proteins = np.array(tested)[np.array(areSignificant)]\n",
    "significant_pvals = np.array(pvals)[np.array(areSignificant)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['TAF1' 'CABLES1' 'HSP90AB1' 'CHD8' 'CDKN2AIP' 'YWHAZ' 'AURKA' 'DAXX'\n",
      " 'AFG1L' 'TAF6' 'FAS' 'CDK2' 'CDKN2A' 'CCNB1']\n",
      "[3.34175606e-02 4.03994861e-04 3.14875026e-03 1.76268430e-04\n",
      " 1.01630359e-02 1.71636933e-02 2.56707953e-03 2.56707953e-03\n",
      " 2.98265751e-02 9.46604364e-03 5.39752132e-05 1.25904262e-02\n",
      " 1.83899711e-03 2.56707953e-03]\n"
     ]
    }
   ],
   "source": [
    "print(significant_proteins)\n",
    "print(significant_pvals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## t-test for effects of all mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "tested = [];\n",
    "p_vals = [];\n",
    "for protein in protList: \n",
    "    if protein in proteomics.columns:\n",
    "        tested.append(protein)\n",
    "        cross = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "        cross = cross.loc[cross['Patient_Type'] == 'Tumor'][[\"Mutation\", protein]].dropna(axis=0)\n",
    "        mutated = cross.loc[cross[\"Mutation\"] != \"Wildtype\"]\n",
    "        wt = cross.loc[cross[\"Mutation\"] == \"Wildtype\"]\n",
    "        ttest = scipy.stats.ttest_ind(mutated[protein], wt[protein])\n",
    "        p_vals.append(ttest[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_vals)[0]\n",
    "pvals = statsmodels.stats.multitest.fdrcorrection(p_vals)[1]\n",
    "significant_proteins = np.array(tested)[np.array(areSignificant)]\n",
    "significant_pvals = np.array(pvals)[np.array(areSignificant)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CABLES1' 'HSP90AB1' 'CHD8' 'CDKN2AIP' 'STK11' 'YWHAZ' 'AURKA' 'DAXX'\n",
      " 'AFG1L' 'FAS' 'CDK2' 'CDKN2A' 'CCNB1']\n",
      "[2.78773236e-04 1.74402946e-03 2.58375695e-04 7.19163403e-03\n",
      " 1.54582617e-02 4.39318009e-03 6.07336786e-04 6.07336786e-04\n",
      " 1.04749693e-02 1.47599338e-05 3.77291095e-03 4.56910336e-04\n",
      " 5.87412307e-04]\n"
     ]
    }
   ],
   "source": [
    "print(significant_proteins)\n",
    "print(significant_pvals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Build the dataframe for plotting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'significantResults' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-f7556d7cce86>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mplotdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mplotdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mplotdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Patient_Type'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'Tumor'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Mutation'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mprotein\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msignificantResults\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m     \u001b[0;32mif\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprotein\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m'TP53'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m         \u001b[0mproteindf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCPTAC\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompare_mutations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mproteomics\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprotein\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mprotein\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'significantResults' is not defined"
     ]
    }
   ],
   "source": [
    "plotdf = CPTAC.compare_mutations(proteomics, gene)\n",
    "plotdf = plotdf.loc[plotdf['Patient_Type'] == 'Tumor'][['Mutation', gene]]\n",
    "\n",
    "for protein in significantResults:\n",
    "    if(protein != 'TP53'):\n",
    "        proteindf = CPTAC.compare_mutations(proteomics, protein, gene)[[protein]]\n",
    "        plotdf = plotdf.join(proteindf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotdf = plotdf.replace({'Mutation': {'Frame_Shift_Del': 'Truncating',\n",
    "                                      'Frame_Shift_Ins': 'Truncating',\n",
    "                                      'Nonsense_Mutation': 'Truncating',\n",
    "                                      'Wildtype': 'Control',\n",
    "                                      'Missense_Mutation': 'Missense'\n",
    "                                     }})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotdf = pd.melt(plotdf, id_vars=['Mutation'], var_name='Protein', value_name='Proteomics')\n",
    "plotdf = plotdf.loc[(plotdf[\"Mutation\"] == \"Control\") | (plotdf[\"Mutation\"] == \"Missense\") | (plotdf[\"Mutation\"] == \"Truncating\")]\n",
    "plotdf = plotdf.dropna(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a4_dims = (23.4, 16.54)\n",
    "fig, ax = plt.subplots(figsize=a4_dims)\n",
    "my_pal = {\"Missense\": \"#6C906D\", \"Control\": \"#596B94\", \"Truncating\": \"#A35857\"}\n",
    "\n",
    "#Create the plot\n",
    "boxplt = sns.boxplot(x='Protein', y='Proteomics', hue='Mutation', data=plotdf, palette=my_pal)\n",
    "boxplt = sns.stripplot(data=plotdf, x='Protein', y='Proteomics', hue='Mutation', dodge=True, jitter=True, color='.3')\n",
    "\n",
    "#Add styling\n",
    "boxplt.set_xlabel('')\n",
    "boxplt.set_ylabel('Protein Level', fontsize='40')\n",
    "boxplt.tick_params(labelsize='20')\n",
    "\n",
    "#Adjust legend\n",
    "handles, labels = ax.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], ['Missense', 'Control', 'Truncating'], title='TP53 Status', fontsize='20', frameon=False)\n",
    "boxplt.get_legend().get_title().set_fontsize('20')\n",
    "boxplt.get_legend().set_bbox_to_anchor((0.15, 0.15, 0, 0))\n",
    "\n",
    "#Add significance indicator: TP53/Missense\n",
    "x1, x2 = -0.25, -0.05\n",
    "y, h, col = 4.45, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: TP53/Truncation\n",
    "x1, x2 = 0.05, 0.25\n",
    "y, h, col = 4.25, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"ns\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: CABLES1/Missense\n",
    "x1, x2 = 0.75, 0.95\n",
    "y, h, col = 1.3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"***\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: CABLES1/Truncation\n",
    "x1, x2 = 1.05, 1.25\n",
    "y, h, col = 0.8 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: HSP90AB1/Missense\n",
    "x1, x2 = 1.75, 1.95\n",
    "y, h, col = 1 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: HSP90AB1/Truncation\n",
    "x1, x2 = 2.05, 2.25\n",
    "y, h, col = 0.8 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"***\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: CHD8/Missense\n",
    "x1, x2 = 2.75, 2.95\n",
    "y, h, col = 1 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: CHD8/Truncation\n",
    "x1, x2 = 3.05, 3.25\n",
    "y, h, col = 0.8 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: UHRF2/Missense\n",
    "x1, x2 = 3.75, 3.95\n",
    "y, h, col = 1 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: UHRF2/Truncation\n",
    "x1, x2 = 4.05, 4.25\n",
    "y, h, col = 0.8 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"ns\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: AURKA/Missense\n",
    "x1, x2 = 4.75, 4.95\n",
    "y, h, col = 3 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"****\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "#Add significance indicator: AURKA/Truncation\n",
    "x1, x2 = 5.05, 5.25\n",
    "y, h, col = 2.8 + .3, .3, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y+h, y+h, y+h, y+h], lw=1.5, c=col)\n",
    "plt.text((x1+x2)*.5, y+h, \"*\", ha='center', va='bottom', color=col, fontsize='23')\n",
    "\n",
    "fig = boxplt.get_figure()\n",
    "fig.savefig('p53_trans.png', dpi=300)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phosphoproteome abundance of interacting proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This is the same list as before, with some of the end numbers of the proteins removed; less specificity in protein names captures more potentially significant results in the phosphoproteomic data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "phosProtList = ['TGFB1', 'AXIN1', 'EP300', 'HRMT1L2', 'CARM1', 'TAF1', 'ING4', 'CABLES1', \n",
    "            'TP73', 'HIPK1', 'HIPK2', 'TP53INP1', 'TP53BP', 'WWOX', 'HCV', 'USP7', 'SYVN1', 'HSP90AB1',\n",
    "           'CHD8', 'ARMC10', 'BANP', 'CDKN2AIP', 'NUAK1', 'STK11', 'LKB1', 'UHRF2', 'E4F1', 'YWHAZ',\n",
    "           'MAML1', 'MKRN1', 'PML', 'MDM2', 'FBXO42', 'ATM', 'PP2R2A', 'AURKA', 'DAXX', 'BRD7', 'TRIM24',\n",
    "           'L3MBTL1', 'GRK5', 'CAK', 'PTK2B', 'PYK2', 'MDM2', 'PRKCG', 'PPIF', 'KAT6A', 'UBC9', 'ZNF385B',\n",
    "           'ZNF385A', 'ANKRD2', 'RFFL', 'RNF34', 'MTA1', 'COP1', 'CCAR2', 'MORC3', 'POU4F2', 'AFG1L', 'UBD',\n",
    "           'TAF6', 'HPV', 'UL123', 'E1B-55K', 'BAX', 'FAS', 'BCL2', 'CREBBP', 'CDK2', 'MDM2', 'CDKN2A', 'ATM',\n",
    "            'CCNB1', 'TERT', 'IGF1R', 'ALB', 'MMP1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TP53-S315</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Patient_Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>0.646000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>-0.800000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S004</th>\n",
       "      <td>1.660000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>3.760000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S007</th>\n",
       "      <td>-0.597000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S008</th>\n",
       "      <td>0.346000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S009</th>\n",
       "      <td>-0.266000</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S010</th>\n",
       "      <td>0.059100</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S011</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S012</th>\n",
       "      <td>0.095100</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S013</th>\n",
       "      <td>-0.118500</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S014</th>\n",
       "      <td>0.350000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S015</th>\n",
       "      <td>3.820000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S016</th>\n",
       "      <td>3.670000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S017</th>\n",
       "      <td>0.376000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S018</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S019</th>\n",
       "      <td>-0.511900</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S020</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S021</th>\n",
       "      <td>0.000164</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S022</th>\n",
       "      <td>0.164000</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S023</th>\n",
       "      <td>0.629000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S024</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S025</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S026</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S027</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S028</th>\n",
       "      <td>0.230000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S029</th>\n",
       "      <td>-0.246000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S030</th>\n",
       "      <td>0.533000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S124</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S125</th>\n",
       "      <td>-1.090000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S126</th>\n",
       "      <td>-0.456000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S127</th>\n",
       "      <td>-1.230000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S128</th>\n",
       "      <td>-0.195000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S129</th>\n",
       "      <td>1.190000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S130</th>\n",
       "      <td>-0.335000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S131</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S132</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S133</th>\n",
       "      <td>-0.182000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S134</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S135</th>\n",
       "      <td>-0.147000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S136</th>\n",
       "      <td>-7.190000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S137</th>\n",
       "      <td>-0.339500</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S138</th>\n",
       "      <td>-1.370000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S139</th>\n",
       "      <td>0.008580</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S140</th>\n",
       "      <td>-0.136000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S141</th>\n",
       "      <td>-0.696000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S142</th>\n",
       "      <td>0.375000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S143</th>\n",
       "      <td>-0.344000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S144</th>\n",
       "      <td>-0.948000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S145</th>\n",
       "      <td>-0.782000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S146</th>\n",
       "      <td>-0.131000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S147</th>\n",
       "      <td>-0.395000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S148</th>\n",
       "      <td>0.236000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S149</th>\n",
       "      <td>-0.630000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S150</th>\n",
       "      <td>-0.000523</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S151</th>\n",
       "      <td>-0.690000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S152</th>\n",
       "      <td>-1.090000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S153</th>\n",
       "      <td>0.520000</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      TP53-S315           Mutation Patient_Type\n",
       "S001        NaN  Missense_Mutation        Tumor\n",
       "S002   0.646000           Wildtype        Tumor\n",
       "S003  -0.800000           Wildtype        Tumor\n",
       "S004   1.660000  Missense_Mutation        Tumor\n",
       "S005        NaN           Wildtype        Tumor\n",
       "S006   3.760000  Missense_Mutation        Tumor\n",
       "S007  -0.597000           Wildtype        Tumor\n",
       "S008   0.346000  Missense_Mutation        Tumor\n",
       "S009  -0.266000  Nonsense_Mutation        Tumor\n",
       "S010   0.059100           Wildtype        Tumor\n",
       "S011        NaN           Wildtype        Tumor\n",
       "S012   0.095100           Wildtype        Tumor\n",
       "S013  -0.118500    Frame_Shift_Del        Tumor\n",
       "S014   0.350000           Wildtype        Tumor\n",
       "S015   3.820000  Missense_Mutation        Tumor\n",
       "S016   3.670000  Missense_Mutation        Tumor\n",
       "S017   0.376000           Wildtype        Tumor\n",
       "S018        NaN           Wildtype        Tumor\n",
       "S019  -0.511900           Wildtype        Tumor\n",
       "S020        NaN  Missense_Mutation        Tumor\n",
       "S021   0.000164           Wildtype        Tumor\n",
       "S022   0.164000  Nonsense_Mutation        Tumor\n",
       "S023   0.629000           Wildtype        Tumor\n",
       "S024        NaN           Wildtype        Tumor\n",
       "S025        NaN           Wildtype        Tumor\n",
       "S026        NaN           Wildtype        Tumor\n",
       "S027        NaN           Wildtype        Tumor\n",
       "S028   0.230000           Wildtype        Tumor\n",
       "S029  -0.246000           Wildtype        Tumor\n",
       "S030   0.533000           Wildtype        Tumor\n",
       "...         ...                ...          ...\n",
       "S124        NaN           Wildtype       Normal\n",
       "S125  -1.090000           Wildtype       Normal\n",
       "S126  -0.456000           Wildtype       Normal\n",
       "S127  -1.230000           Wildtype       Normal\n",
       "S128  -0.195000           Wildtype       Normal\n",
       "S129   1.190000           Wildtype       Normal\n",
       "S130  -0.335000           Wildtype       Normal\n",
       "S131        NaN           Wildtype       Normal\n",
       "S132        NaN           Wildtype       Normal\n",
       "S133  -0.182000           Wildtype       Normal\n",
       "S134        NaN           Wildtype       Normal\n",
       "S135  -0.147000           Wildtype       Normal\n",
       "S136  -7.190000           Wildtype       Normal\n",
       "S137  -0.339500           Wildtype       Normal\n",
       "S138  -1.370000           Wildtype       Normal\n",
       "S139   0.008580           Wildtype       Normal\n",
       "S140  -0.136000           Wildtype       Normal\n",
       "S141  -0.696000           Wildtype       Normal\n",
       "S142   0.375000           Wildtype       Normal\n",
       "S143  -0.344000           Wildtype       Normal\n",
       "S144  -0.948000           Wildtype       Normal\n",
       "S145  -0.782000           Wildtype       Normal\n",
       "S146  -0.131000           Wildtype       Normal\n",
       "S147  -0.395000           Wildtype       Normal\n",
       "S148   0.236000           Wildtype       Normal\n",
       "S149  -0.630000           Wildtype       Normal\n",
       "S150  -0.000523           Wildtype       Normal\n",
       "S151  -0.690000           Wildtype       Normal\n",
       "S152  -1.090000           Wildtype       Normal\n",
       "S153   0.520000           Wildtype       Normal\n",
       "\n",
       "[153 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sitedf = CPTAC.compare_mutations(phos, 'TP53-S315', gene)\n",
    "sitedf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACAP2-S379\n",
      "Ttest_indResult(statistic=5.584922959854958, pvalue=2.0928008546907557e-07)\n",
      "AIF1L-S160\n",
      "Ttest_indResult(statistic=5.627812345689005, pvalue=1.7351033560196802e-07)\n",
      "CENPF-S1722\n",
      "Ttest_indResult(statistic=5.855601131175288, pvalue=1.40618346876695e-07)\n",
      "CHEK2-S163\n",
      "Ttest_indResult(statistic=6.6305274586646785, pvalue=1.7725206790447915e-08)\n",
      "DNMT3A-S390\n",
      "Ttest_indResult(statistic=5.686647783676332, pvalue=3.198722606290891e-07)\n",
      "ESF1-S153\n",
      "Ttest_indResult(statistic=7.364643602517957, pvalue=2.1839400676812538e-10)\n",
      "FOXK1-S223\n",
      "Ttest_indResult(statistic=5.852582373689908, pvalue=6.429375955330947e-08)\n",
      "FXR1-S409\n",
      "Ttest_indResult(statistic=6.438040207019647, pvalue=4.4953070675887535e-09)\n",
      "FXR1-T411\n",
      "Ttest_indResult(statistic=7.601164481535482, pvalue=1.7816680559272247e-11)\n",
      "HJURP-S473\n",
      "Ttest_indResult(statistic=5.422631596564276, pvalue=4.7066148378401633e-07)\n",
      "HMGA1-S44\n",
      "Ttest_indResult(statistic=5.338271235986477, pvalue=6.06923229540704e-07)\n",
      "HSP90AB1-S255\n",
      "Ttest_indResult(statistic=5.988606892109238, pvalue=3.497088250848597e-08)\n",
      "HSP90AB1-S261\n",
      "Ttest_indResult(statistic=6.124427993790099, pvalue=1.8929372454310217e-08)\n",
      "IRF2BPL-S547\n",
      "Ttest_indResult(statistic=5.386369216523543, pvalue=4.940263316918059e-07)\n",
      "JADE3-S774\n",
      "Ttest_indResult(statistic=6.106617148546644, pvalue=2.4001389828219054e-08)\n",
      "KMT2B-S844\n",
      "Ttest_indResult(statistic=5.538630089339659, pvalue=3.26423186849982e-07)\n",
      "LMNA-S429\n",
      "Ttest_indResult(statistic=-5.464637803722743, pvalue=3.527543504001237e-07)\n",
      "MCM3-S726\n",
      "Ttest_indResult(statistic=6.753494406889159, pvalue=2.822256266703885e-09)\n",
      "MCM6-S762\n",
      "Ttest_indResult(statistic=7.755312677317857, pvalue=1.648781027133358e-11)\n",
      "MYBL2-S282\n",
      "Ttest_indResult(statistic=6.01999519790588, pvalue=3.0361251671439404e-08)\n",
      "NCL-S619\n",
      "Ttest_indResult(statistic=5.595592168502766, pvalue=1.9975833398021239e-07)\n",
      "NCL-S67\n",
      "Ttest_indResult(statistic=6.0520943031016055, pvalue=2.6266735395487437e-08)\n",
      "NIPBL-T599\n",
      "Ttest_indResult(statistic=6.038472156342363, pvalue=4.082296447792649e-08)\n",
      "PHF3-S1133\n",
      "Ttest_indResult(statistic=5.482917537885181, pvalue=3.259597277167453e-07)\n",
      "PTPN2-S304\n",
      "Ttest_indResult(statistic=6.152907752947197, pvalue=1.9523777470836474e-08)\n",
      "RECQL5-S491\n",
      "Ttest_indResult(statistic=5.65995996873664, pvalue=1.507036232707932e-07)\n",
      "RIF1-S1873\n",
      "Ttest_indResult(statistic=6.144458543988556, pvalue=6.334361226471046e-08)\n",
      "RIF1-S1876\n",
      "Ttest_indResult(statistic=6.958998212676846, pvalue=1.1929844019601943e-07)\n",
      "RIOK1-S22\n",
      "Ttest_indResult(statistic=7.274363661246851, pvalue=8.651302886259795e-11)\n",
      "RRP36-S73\n",
      "Ttest_indResult(statistic=5.653047160593592, pvalue=2.0628807423278951e-07)\n",
      "SLC4A1AP-S312\n",
      "Ttest_indResult(statistic=5.587421118472314, pvalue=2.7864612689889264e-07)\n",
      "SLC4A1AP-S466\n",
      "Ttest_indResult(statistic=6.175006958182189, pvalue=1.5040196839314483e-08)\n",
      "SMC1A-S358\n",
      "Ttest_indResult(statistic=10.964768205317732, pvalue=2.2024473944628135e-10)\n",
      "SMC1A-S360\n",
      "Ttest_indResult(statistic=8.106568549348813, pvalue=1.877656118372916e-09)\n",
      "SMC3-S1065\n",
      "Ttest_indResult(statistic=6.302426103798918, pvalue=8.398688128075596e-09)\n",
      "SMG1-S34\n",
      "Ttest_indResult(statistic=5.731602465840903, pvalue=5.701715439775803e-07)\n",
      "SON-S152\n",
      "Ttest_indResult(statistic=5.61513542250127, pvalue=1.8340682190852294e-07)\n",
      "SORBS2-S402\n",
      "Ttest_indResult(statistic=-5.434558356379387, pvalue=4.65103096399622e-07)\n",
      "SP100-S231\n",
      "Ttest_indResult(statistic=6.016110746210287, pvalue=3.089753154646366e-08)\n",
      "STMN1-S63\n",
      "Ttest_indResult(statistic=6.005180498787388, pvalue=3.245709961458589e-08)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3157: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCOF1-S583\n",
      "Ttest_indResult(statistic=5.680148669901479, pvalue=1.379139610380399e-07)\n",
      "TOP2A-S1247\n",
      "Ttest_indResult(statistic=5.717240450741866, pvalue=1.330193085994742e-07)\n",
      "TOP2A-S1377\n",
      "Ttest_indResult(statistic=5.584633402366059, pvalue=2.0954461206568472e-07)\n",
      "TOP2A-S1393\n",
      "Ttest_indResult(statistic=5.889800441682831, pvalue=2.422370278562358e-07)\n",
      "TP53-S315\n",
      "Ttest_indResult(statistic=5.461805676393611, pvalue=5.572977461429709e-07)\n",
      "TP53BP1-S1763\n",
      "Ttest_indResult(statistic=7.687154722386778, pvalue=3.307433804821374e-11)\n",
      "TP53BP1-S1764\n",
      "Ttest_indResult(statistic=5.650805451544988, pvalue=2.3686027832938366e-07)\n",
      "USP51-S26\n",
      "Ttest_indResult(statistic=10.428651518636508, pvalue=2.271386374366031e-07)\n",
      "USP8-S719\n",
      "Ttest_indResult(statistic=-5.405582807470919, pvalue=4.5492049360333684e-07)\n",
      "UTP14A-S445\n",
      "Ttest_indResult(statistic=5.7659415412569475, pvalue=1.508473305168412e-07)\n",
      "WDR75-S779\n",
      "Ttest_indResult(statistic=5.363174905068154, pvalue=5.456352230422431e-07)\n",
      "ZFAND3-S129\n",
      "Ttest_indResult(statistic=5.505315146304779, pvalue=2.9583731219450865e-07)\n",
      "ZNF318-S1243\n",
      "Ttest_indResult(statistic=5.719748217437114, pvalue=2.6074029827125374e-07)\n"
     ]
    }
   ],
   "source": [
    "pcutoff = 0.05 / len(phos.columns)\n",
    "gene = 'TP53'\n",
    "p_values = []\n",
    "site_names = []\n",
    "\n",
    "for site in phos.columns:\n",
    "    if(site != gene and site != 'idx'):\n",
    "        sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "        sitedf = sitedf.loc[sitedf['Patient_Type'] == 'Tumor'].dropna(axis=0)\n",
    "        mutated_site = sitedf.loc[sitedf['Mutation'] != 'Wildtype']\n",
    "        wt_site = sitedf.loc[sitedf['Mutation'] == 'Wildtype']\n",
    "        if len(mutated_site) > 2:\n",
    "            ttest = scipy.stats.ttest_ind(mutated_site[site], wt_site[site])\n",
    "            if ttest[1] <= pcutoff:\n",
    "                print(site)\n",
    "                print(ttest)                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATXN2-S684\n",
      "Ttest_indResult(statistic=3.655877883159149, pvalue=0.0004143865925924668)\n",
      "BACH2-S510\n",
      "Ttest_indResult(statistic=-4.483054616952817, pvalue=0.0007484241489183486)\n",
      "CARD11-S448\n",
      "Ttest_indResult(statistic=-5.780529532874567, pvalue=0.0002658892492140297)\n",
      "CGN-S168\n",
      "Ttest_indResult(statistic=4.327756281512896, pvalue=0.0005975148332866843)\n",
      "CTNNB1-T41\n",
      "Ttest_indResult(statistic=-7.323236669966868, pvalue=0.0007442445738937455)\n",
      "DAXX-T420\n",
      "Ttest_indResult(statistic=-4.669739577116847, pvalue=0.0008811556706892382)\n",
      "DES-S47\n",
      "Ttest_indResult(statistic=-4.295944907747133, pvalue=0.0008697834923291664)\n",
      "EXOC4-S468\n",
      "Ttest_indResult(statistic=-5.032285910854635, pvalue=6.924481714683101e-06)\n",
      "FAM120A-Y395\n",
      "Ttest_indResult(statistic=11.352798979430741, pvalue=0.0003432439478581016)\n",
      "FERMT2-S673\n",
      "Ttest_indResult(statistic=-3.4157079773770427, pvalue=0.0009447398272045617)\n",
      "GATAD1-S235\n",
      "Ttest_indResult(statistic=14.381372503515209, pvalue=0.00013585610818085095)\n",
      "HELZ-S1767\n",
      "Ttest_indResult(statistic=3.4006777412554223, pvalue=0.0009739278142552264)\n",
      "IFI16-T114\n",
      "Ttest_indResult(statistic=-4.115623225751573, pvalue=0.0007219116033165132)\n",
      "IPCEF1-S335\n",
      "Ttest_indResult(statistic=-6.704211305664939, pvalue=0.0005347403130513619)\n",
      "KIAA0355-S723\n",
      "Ttest_indResult(statistic=-3.4985329812092174, pvalue=0.000743248141662276)\n",
      "KIAA1217-S1329\n",
      "Ttest_indResult(statistic=13.587316935724077, pvalue=0.00016986116619984766)\n",
      "KMT2A-S3514\n",
      "Ttest_indResult(statistic=-3.469145442722512, pvalue=0.0007774937844088817)\n",
      "LRBA-S1005\n",
      "Ttest_indResult(statistic=3.5256913272570283, pvalue=0.0006636042418640248)\n",
      "LTB4R-S313\n",
      "Ttest_indResult(statistic=-4.350554736896303, pvalue=0.0002562530408621131)\n",
      "MPHOSPH6-T147\n",
      "Ttest_indResult(statistic=3.9202983280319676, pvalue=0.00036840094175859885)\n",
      "NCKAP5L-T1262\n",
      "Ttest_indResult(statistic=17.3662276497492, pvalue=6.453414191741867e-05)\n",
      "NES-S1016\n",
      "Ttest_indResult(statistic=-3.5042755634593616, pvalue=0.0007809992210136065)\n",
      "NOLC1-T188\n",
      "Ttest_indResult(statistic=5.029655981622827, pvalue=0.0007096991699040373)\n",
      "NT5C2-S535\n",
      "Ttest_indResult(statistic=-13.15591760651327, pvalue=0.0001928061605925224)\n",
      "NUFIP2-T335\n",
      "Ttest_indResult(statistic=8.745647930415343, pvalue=0.0009419912332621163)\n",
      "PCM1-S1412\n",
      "Ttest_indResult(statistic=3.5336756201101456, pvalue=0.0007202168715041542)\n",
      "PLCE1-T2270\n",
      "Ttest_indResult(statistic=10.639180735454474, pvalue=0.0004419399597415622)\n",
      "PLEKHN1-S571\n",
      "Ttest_indResult(statistic=-4.249967925518245, pvalue=0.0003019890249829488)\n",
      "PPP6R2-T431\n",
      "Ttest_indResult(statistic=3.585608838793728, pvalue=0.0009903071581002234)\n",
      "PRRC2A-S159\n",
      "Ttest_indResult(statistic=-4.724805818258825, pvalue=0.0002289208698491465)\n",
      "RAF1-S316\n",
      "Ttest_indResult(statistic=-3.6346663017796943, pvalue=0.0005669994538310606)\n",
      "RAPH1-S1160\n",
      "Ttest_indResult(statistic=-12.80507597983372, pvalue=5.169228002798576e-05)\n",
      "SAMD4B-S172\n",
      "Ttest_indResult(statistic=-3.456476085016165, pvalue=0.0009548067444037111)\n",
      "SETD1B-S1659\n",
      "Ttest_indResult(statistic=-4.15035456696699, pvalue=0.00010816757838408726)\n",
      "TCF7L2-S58\n",
      "Ttest_indResult(statistic=-10.917071169022437, pvalue=0.0003997753495677089)\n",
      "TNS1-T1173\n",
      "Ttest_indResult(statistic=-3.503168445332504, pvalue=0.0009427823869010998)\n",
      "TNS2-S1257\n",
      "Ttest_indResult(statistic=-4.978155524831363, pvalue=0.00013676606237397352)\n",
      "USP6NL-S757\n",
      "Ttest_indResult(statistic=6.546073962315258, pvalue=6.505037671143995e-05)\n"
     ]
    }
   ],
   "source": [
    "pcutoff = 0.001\n",
    "gene = 'PIK3CA'\n",
    "p_values = []\n",
    "site_names = []\n",
    "\n",
    "for site in phos.columns:\n",
    "    if(site != gene and site != 'idx'):\n",
    "        sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "        sitedf = sitedf.loc[sitedf['Patient_Type'] == 'Tumor'].dropna(axis=0)\n",
    "        mutated_site = sitedf.loc[sitedf['Mutation'] != 'Wildtype']\n",
    "        wt_site = sitedf.loc[sitedf['Mutation'] == 'Wildtype']\n",
    "        if len(mutated_site) > 2 and len(wt_site) > 2:\n",
    "            ttest = scipy.stats.ttest_ind(mutated_site[site], wt_site[site])\n",
    "            if ttest[1] <= pcutoff:\n",
    "                print(site)\n",
    "                print(ttest)     "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot phosphorylation at one site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "sitedf = sitedf.loc[sitedf['Patient_Type'] == 'Tumor'].dropna(axis=0)\n",
    "mutated_site = sitedf.loc[sitedf['Mutation'] != 'Wildtype']\n",
    "wt_site = sitedf.loc[sitedf['Mutation'] == 'Wildtype']\n",
    "ttest = scipy.stats.ttest_ind(mutated_site[site], wt_site[site])\n",
    "print(ttest)\n",
    "\n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Within Hotspot', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Level', fontsize='20')\n",
    "boxplt.set_title(\"Effect of TP53 mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Look at truncating mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['TAF1D-S206' 'HIPK1-T1027' 'TP53BP1-S1431' 'TP53BP1-S1683'\n",
      " 'TP53BP1-S1706' 'TP53BP1-S1763' 'TP53BP1-S1764' 'TP53BP1-S867'\n",
      " 'HSP90AB1-S226' 'HSP90AB1-S255' 'HSP90AB1-S261' 'CHD8-S1420' 'CHD8-S1424'\n",
      " 'CHD8-S1995' 'CHD8-S549' 'FASN-S2198' 'FASN-T2204' 'BCL2L12-S273']\n",
      "[0.04492884 0.03343621 0.04492884 0.03343621 0.03271287 0.00202\n",
      " 0.04503339 0.03558846 0.0272901  0.00202    0.00498404 0.03343621\n",
      " 0.03558846 0.03343621 0.03271287 0.02017401 0.03271287 0.03343621]\n"
     ]
    }
   ],
   "source": [
    "sites = phos.columns\n",
    "p_values = []\n",
    "site_names = []\n",
    "\n",
    "for protein in phosProtList:\n",
    "    pattern = re.compile(protein)\n",
    "    isInList = filter(pattern.search, sites)\n",
    "    if next(isInList, None) is not None:\n",
    "        phosphositesdf = CPTAC.compare_mutations(phos, protein, gene)\n",
    "        phosphositesdf = phosphositesdf.loc[phosphositesdf['Patient_Type'] == 'Tumor'].drop('Patient_Type', axis = 1)\n",
    "        for site in phosphositesdf.columns:\n",
    "            if (site is not 'Mutation'):\n",
    "                sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "                mutateddf = sitedf.loc[(sitedf[\"Mutation\"] == \"Frame_Shift_Del\") | \n",
    "                            (sitedf[\"Mutation\"] == \"Frame_Shift_Ins\") |\n",
    "                            (sitedf[\"Mutation\"] == 'Nonsense_Mutation')].dropna(axis=0)\n",
    "                wtdf = sitedf.loc[sitedf['Mutation'] == 'Wildtype'].dropna(axis=0)\n",
    "                ttest = scipy.stats.ttest_ind(mutateddf[site], wtdf[site])\n",
    "                p_values.append(ttest[1])\n",
    "                site_names.append(site)     \n",
    "                \n",
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_values)[0]\n",
    "pvalues = statsmodels.stats.multitest.fdrcorrection(p_values)[1]\n",
    "significant_sites = np.array(site_names)[np.array(areSignificant)]\n",
    "significant_pvalues = np.array(pvalues)[np.array(areSignificant)]\n",
    "\n",
    "print(significant_sites)\n",
    "print(significant_pvalues)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Missense and truncating"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['TGFB1I1-S192' 'CABLES1-S373' 'CABLES1-T415' 'HIPK1-S1200' 'HIPK1-T1027'\n",
      " 'TP53BP1-S1109' 'TP53BP1-S1431' 'TP53BP1-S1435' 'TP53BP1-S1683'\n",
      " 'TP53BP1-S1706' 'TP53BP1-S1763' 'TP53BP1-S1764' 'TP53BP1-S400'\n",
      " 'TP53BP1-S403' 'TP53BP1-S557' 'TP53BP1-S640' 'TP53BP1-S644'\n",
      " 'TP53BP1-S645' 'TP53BP1-S776' 'TP53BP1-S867' 'TP53BP1-S998'\n",
      " 'TP53BP1-T1653' 'HSP90AB1-S226' 'HSP90AB1-S255' 'HSP90AB1-S261'\n",
      " 'HSP90AB1-S445' 'CHD8-S1420' 'CHD8-S1995' 'CHD8-S2008' 'CHD8-S549'\n",
      " 'CHD8-S562' 'CHD8-T1982' 'CDKN2AIP-S131' 'DAXX-S680' 'MTA1-T564'\n",
      " 'FASN-S207' 'FASN-T2204' 'BCL2L12-S273' 'BCL2L13-S444' 'CREBBP-T974'\n",
      " 'CDKN2AIP-S131' 'RALBP1-S30']\n",
      "[4.83396943e-02 1.02748279e-02 1.71077907e-02 2.26293623e-02\n",
      " 3.31176397e-03 5.10409707e-03 1.59419677e-05 3.56867481e-05\n",
      " 4.05486152e-05 4.91755490e-07 7.77961292e-07 1.04397887e-03\n",
      " 3.86388492e-02 5.10409707e-03 3.98470422e-02 9.00020989e-03\n",
      " 4.27340920e-02 1.18523003e-02 3.14587547e-02 1.59419677e-05\n",
      " 4.91988605e-02 1.07792725e-02 4.05486152e-05 4.91755490e-07\n",
      " 2.38123612e-05 4.18215372e-03 3.11044243e-03 3.31176397e-03\n",
      " 1.58587568e-02 5.21294131e-03 1.13654894e-02 2.21669234e-02\n",
      " 5.10409707e-03 4.06499687e-02 1.58587568e-02 9.92360295e-03\n",
      " 1.72933722e-02 1.92222507e-03 1.18523003e-02 9.35659272e-03\n",
      " 5.10409707e-03 4.06499687e-02]\n"
     ]
    }
   ],
   "source": [
    "sites = phos.columns\n",
    "p_values = []\n",
    "site_names = []\n",
    "\n",
    "for protein in phosProtList:\n",
    "    pattern = re.compile(protein)\n",
    "    isInList = filter(pattern.search, sites)\n",
    "    if next(isInList, None) is not None:\n",
    "        phosphositesdf = CPTAC.compare_mutations(phos, protein, gene)\n",
    "        phosphositesdf = phosphositesdf.loc[phosphositesdf['Patient_Type'] == 'Tumor'].drop('Patient_Type', axis = 1)\n",
    "        for site in phosphositesdf.columns:\n",
    "            if (site is not 'Mutation'):\n",
    "                sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "                mutateddf = sitedf.loc[(sitedf[\"Mutation\"] == \"Frame_Shift_Del\") | \n",
    "                        (sitedf[\"Mutation\"] == \"Frame_Shift_Ins\") |\n",
    "                        (sitedf[\"Mutation\"] == 'Nonsense_Mutation') |\n",
    "                        (sitedf[\"Mutation\"] == 'Missense_Mutation')            ].dropna(axis=0)\n",
    "                wtdf = sitedf.loc[sitedf['Mutation'] == 'Wildtype'].dropna(axis=0)\n",
    "                ttest = scipy.stats.ttest_ind(mutateddf[site], wtdf[site])\n",
    "                p_values.append(ttest[1])\n",
    "                site_names.append(site)       \n",
    "\n",
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_values)[0]\n",
    "pvalues = statsmodels.stats.multitest.fdrcorrection(p_values)[1]\n",
    "significant_sites = np.array(site_names)[np.array(areSignificant)]\n",
    "significant_pvalues = np.array(pvalues)[np.array(areSignificant)]\n",
    "\n",
    "print(significant_sites)\n",
    "print(significant_pvalues)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Look at all mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['TGFB1I1-S137' 'TGFB1I1-S143' 'TGFB1I1-S192' 'TGFB1I1-S68' 'TAF15-S97'\n",
      " 'CABLES1-S109' 'CABLES1-S290' 'CABLES1-S373' 'CABLES1-S418'\n",
      " 'CABLES1-T415' 'HIPK1-S1200' 'HIPK1-T1027' 'TP53BP1-S1004'\n",
      " 'TP53BP1-S1099' 'TP53BP1-S1109' 'TP53BP1-S1431' 'TP53BP1-S1435'\n",
      " 'TP53BP1-S1683' 'TP53BP1-S1706' 'TP53BP1-S1763' 'TP53BP1-S1764'\n",
      " 'TP53BP1-S400' 'TP53BP1-S403' 'TP53BP1-S525' 'TP53BP1-S557'\n",
      " 'TP53BP1-S640' 'TP53BP1-S644' 'TP53BP1-S645' 'TP53BP1-S776'\n",
      " 'TP53BP1-S867' 'TP53BP1-T1653' 'TP53BP1-T307' 'HSP90AB1-S226'\n",
      " 'HSP90AB1-S255' 'HSP90AB1-S261' 'HSP90AB1-S445' 'CHD8-S1420' 'CHD8-S1995'\n",
      " 'CHD8-S2008' 'CHD8-S549' 'CHD8-S562' 'CHD8-T1982' 'CDKN2AIP-S131'\n",
      " 'STK11IP-S599' 'DAXX-S680' 'TRIM24-S1025' 'TRIM24-S1028' 'ANKRD26-S631'\n",
      " 'MTA1-T564' 'TAF6-S673' 'FASN-S207' 'FASN-S2198' 'FASN-T2204'\n",
      " 'BCL2L12-S273' 'BCL2L13-S444' 'CREBBP-S121' 'CREBBP-T974' 'CDKN2AIP-S131'\n",
      " 'RALBP1-S30']\n",
      "[4.73928331e-02 4.24683358e-02 1.30612019e-02 1.96788056e-02\n",
      " 4.87173544e-02 4.04222291e-02 2.81107614e-02 5.42094133e-03\n",
      " 2.09663430e-02 6.17934786e-03 2.55412949e-02 9.99937979e-03\n",
      " 3.78259978e-02 2.81107614e-02 1.49026382e-03 7.90494374e-07\n",
      " 1.57438520e-06 2.41290803e-06 3.73010059e-08 3.73010059e-08\n",
      " 7.78857041e-05 1.47993669e-02 2.15188496e-03 4.73928331e-02\n",
      " 4.87173544e-02 3.08885029e-03 4.43785531e-02 1.14420067e-02\n",
      " 3.32173526e-02 1.79499077e-06 2.00013985e-03 3.35051783e-02\n",
      " 6.37779242e-06 3.73010059e-08 3.50570262e-06 9.23826549e-04\n",
      " 3.08885029e-03 1.49026382e-03 2.37495864e-02 3.08885029e-03\n",
      " 7.37469189e-03 2.55412949e-02 6.19448264e-03 4.87173544e-02\n",
      " 1.53392569e-02 4.87173544e-02 4.73928331e-02 2.37495864e-02\n",
      " 2.00013985e-03 3.32173526e-02 6.48376054e-03 4.87173544e-02\n",
      " 8.66561436e-03 6.71318064e-04 1.72393000e-02 4.94488928e-02\n",
      " 3.08885029e-03 6.19448264e-03 2.37495864e-02]\n"
     ]
    }
   ],
   "source": [
    "sites = phos.columns\n",
    "p_values = []\n",
    "site_names = []\n",
    "\n",
    "for protein in phosProtList:\n",
    "    pattern = re.compile(protein)\n",
    "    isInList = filter(pattern.search, sites)\n",
    "    if next(isInList, None) is not None:\n",
    "        phosphositesdf = CPTAC.compare_mutations(phos, protein, gene)\n",
    "        phosphositesdf = phosphositesdf.loc[phosphositesdf['Patient_Type'] == 'Tumor'].drop('Patient_Type', axis = 1)\n",
    "        for site in phosphositesdf.columns:\n",
    "            if (site is not 'Mutation'):\n",
    "                sitedf = CPTAC.compare_mutations(phos, site, gene)\n",
    "                mutateddf = sitedf.loc[sitedf[\"Mutation\"] != 'Wildtype'].dropna(axis=0)\n",
    "                wtdf = sitedf.loc[sitedf['Mutation'] == 'Wildtype'].dropna(axis=0)\n",
    "                ttest = scipy.stats.ttest_ind(mutateddf[site], wtdf[site])\n",
    "                p_values.append(ttest[1])\n",
    "                site_names.append(site)       \n",
    "\n",
    "areSignificant = statsmodels.stats.multitest.fdrcorrection(p_values)[0]\n",
    "pvalues = statsmodels.stats.multitest.fdrcorrection(p_values)[1]\n",
    "significant_sites = np.array(site_names)[np.array(areSignificant)]\n",
    "significant_pvalues = np.array(pvalues)[np.array(areSignificant)]\n",
    "\n",
    "print(significant_sites)\n",
    "print(significant_pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
