{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter CPTAC.embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "from scipy.stats import pearsonr\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import re \n",
    "import CPTAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "somatic_mutations = CPTAC.get_somatic()\n",
    "proteomics = CPTAC.get_proteomics()\n",
    "phos = CPTAC.get_phosphoproteomics()\n",
    "trans = CPTAC.get_transcriptomics()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       ARID1A Patient_Type\n",
      "S001 -0.87900        Tumor\n",
      "S002 -1.41000        Tumor\n",
      "S003 -1.04000        Tumor\n",
      "S004  0.82700        Tumor\n",
      "S005  0.38500        Tumor\n",
      "S006  0.01060        Tumor\n",
      "S007  0.20200        Tumor\n",
      "S008  0.06220        Tumor\n",
      "S009  0.73500        Tumor\n",
      "S010  0.51900        Tumor\n",
      "S011 -0.08670        Tumor\n",
      "S012 -1.61000        Tumor\n",
      "S013  0.11200        Tumor\n",
      "S014 -1.54000        Tumor\n",
      "S015  0.76700        Tumor\n",
      "S016  0.73300        Tumor\n",
      "S017  0.50400        Tumor\n",
      "S018  0.19500        Tumor\n",
      "S019 -0.20900        Tumor\n",
      "S020 -0.03840        Tumor\n",
      "S021  0.26900        Tumor\n",
      "S022  0.22500        Tumor\n",
      "S023  0.14900        Tumor\n",
      "S024 -0.99200        Tumor\n",
      "S025  0.39300        Tumor\n",
      "S026 -0.11300        Tumor\n",
      "S027  0.13700        Tumor\n",
      "S028 -0.30200        Tumor\n",
      "S029  0.03300        Tumor\n",
      "S030 -1.19000        Tumor\n",
      "...       ...          ...\n",
      "S071  0.73900        Tumor\n",
      "S072  0.16800        Tumor\n",
      "S073 -1.14000        Tumor\n",
      "S074  0.17400        Tumor\n",
      "S075 -0.10600        Tumor\n",
      "S076  0.22600        Tumor\n",
      "S077 -1.46000        Tumor\n",
      "S078 -0.35600        Tumor\n",
      "S079 -0.76100        Tumor\n",
      "S080 -0.24000        Tumor\n",
      "S081  0.29000        Tumor\n",
      "S082  0.08090        Tumor\n",
      "S083 -2.12000        Tumor\n",
      "S084 -0.01280        Tumor\n",
      "S085  0.26900        Tumor\n",
      "S086  0.42600        Tumor\n",
      "S087 -1.43000        Tumor\n",
      "S088  0.02600        Tumor\n",
      "S089 -1.36000        Tumor\n",
      "S090 -0.00658        Tumor\n",
      "S091 -0.84000        Tumor\n",
      "S092 -0.79300        Tumor\n",
      "S093 -0.06580        Tumor\n",
      "S094 -1.10000        Tumor\n",
      "S095 -0.14100        Tumor\n",
      "S096  0.92700        Tumor\n",
      "S097  0.79500        Tumor\n",
      "S098 -0.65100        Tumor\n",
      "S099 -0.42800        Tumor\n",
      "S100 -1.65000        Tumor\n",
      "\n",
      "[100 rows x 2 columns]\n",
      "        ARID1A Patient_Type\n",
      "S101  0.246000       Normal\n",
      "S102 -0.120000       Normal\n",
      "S103  0.200000       Normal\n",
      "S104  0.262000       Normal\n",
      "S105 -0.233000       Normal\n",
      "S106 -0.064400       Normal\n",
      "S107 -0.132000       Normal\n",
      "S108 -0.185000       Normal\n",
      "S109  0.010800       Normal\n",
      "S110 -0.248000       Normal\n",
      "S111  0.200000       Normal\n",
      "S112 -0.020500       Normal\n",
      "S113 -0.105000       Normal\n",
      "S114  0.041300       Normal\n",
      "S115 -0.129000       Normal\n",
      "S116 -0.453000       Normal\n",
      "S117 -0.065200       Normal\n",
      "S118  0.177000       Normal\n",
      "S119  0.188000       Normal\n",
      "S120 -0.087200       Normal\n",
      "S121 -0.032000       Normal\n",
      "S122  0.130000       Normal\n",
      "S123 -0.054900       Normal\n",
      "S124 -0.288000       Normal\n",
      "S125 -0.574000       Normal\n",
      "S126 -0.368000       Normal\n",
      "S127 -0.139000       Normal\n",
      "S128 -0.215000       Normal\n",
      "S129 -0.068100       Normal\n",
      "S130 -0.083900       Normal\n",
      "S131 -0.083400       Normal\n",
      "S132  0.024900       Normal\n",
      "S133 -0.169000       Normal\n",
      "S134 -0.313000       Normal\n",
      "S135 -0.062700       Normal\n",
      "S136  0.314000       Normal\n",
      "S137  0.163000       Normal\n",
      "S138  0.120000       Normal\n",
      "S139  0.138000       Normal\n",
      "S140  0.318000       Normal\n",
      "S141  0.163000       Normal\n",
      "S142  0.297000       Normal\n",
      "S143  0.404000       Normal\n",
      "S144  0.142000       Normal\n",
      "S145 -0.000069       Normal\n",
      "S146  0.276000       Normal\n",
      "S147  0.112000       Normal\n",
      "S148  0.269000       Normal\n",
      "S149  0.258000       Normal\n",
      "S150  0.263000       Normal\n",
      "S151 -0.010900       Normal\n",
      "S152  0.000637       Normal\n",
      "S153  0.222000       Normal\n",
      "[0.0692887528650658]\n"
     ]
    }
   ],
   "source": [
    "gene = \"ARID1A\"\n",
    "protein = \"ARID1A\"\n",
    "\n",
    "p_values = []\n",
    "\n",
    "genedf = CPTAC.compare_mutations(proteomics, gene, protein)\n",
    "#print(genedf)\n",
    "tumordf = genedf.loc[genedf[\"Patient_Type\"] == \"Tumor\"].drop(\"Mutation\", axis = 1)\n",
    "normaldf = genedf.loc[genedf[\"Patient_Type\"] != \"Tumor\"].drop(\"Mutation\", axis = 1)\n",
    "print(tumordf)\n",
    "print(normaldf)\n",
    "\n",
    "ttest = scipy.stats.ttest_ind(tumordf[protein], normaldf[protein])\n",
    "p_values.append(ttest[1])\n",
    "\n",
    "print(p_values)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        ARID1A Patient_Type\n",
      "S001 -0.879000        Tumor\n",
      "S002 -1.410000        Tumor\n",
      "S003 -1.040000        Tumor\n",
      "S004  0.827000        Tumor\n",
      "S005  0.385000        Tumor\n",
      "S006  0.010600        Tumor\n",
      "S007  0.202000        Tumor\n",
      "S008  0.062200        Tumor\n",
      "S009  0.735000        Tumor\n",
      "S010  0.519000        Tumor\n",
      "S011 -0.086700        Tumor\n",
      "S012 -1.610000        Tumor\n",
      "S013  0.112000        Tumor\n",
      "S014 -1.540000        Tumor\n",
      "S015  0.767000        Tumor\n",
      "S016  0.733000        Tumor\n",
      "S017  0.504000        Tumor\n",
      "S018  0.195000        Tumor\n",
      "S019 -0.209000        Tumor\n",
      "S020 -0.038400        Tumor\n",
      "S021  0.269000        Tumor\n",
      "S022  0.225000        Tumor\n",
      "S023  0.149000        Tumor\n",
      "S024 -0.992000        Tumor\n",
      "S025  0.393000        Tumor\n",
      "S026 -0.113000        Tumor\n",
      "S027  0.137000        Tumor\n",
      "S028 -0.302000        Tumor\n",
      "S029  0.033000        Tumor\n",
      "S030 -1.190000        Tumor\n",
      "...        ...          ...\n",
      "S124 -0.288000       Normal\n",
      "S125 -0.574000       Normal\n",
      "S126 -0.368000       Normal\n",
      "S127 -0.139000       Normal\n",
      "S128 -0.215000       Normal\n",
      "S129 -0.068100       Normal\n",
      "S130 -0.083900       Normal\n",
      "S131 -0.083400       Normal\n",
      "S132  0.024900       Normal\n",
      "S133 -0.169000       Normal\n",
      "S134 -0.313000       Normal\n",
      "S135 -0.062700       Normal\n",
      "S136  0.314000       Normal\n",
      "S137  0.163000       Normal\n",
      "S138  0.120000       Normal\n",
      "S139  0.138000       Normal\n",
      "S140  0.318000       Normal\n",
      "S141  0.163000       Normal\n",
      "S142  0.297000       Normal\n",
      "S143  0.404000       Normal\n",
      "S144  0.142000       Normal\n",
      "S145 -0.000069       Normal\n",
      "S146  0.276000       Normal\n",
      "S147  0.112000       Normal\n",
      "S148  0.269000       Normal\n",
      "S149  0.258000       Normal\n",
      "S150  0.263000       Normal\n",
      "S151 -0.010900       Normal\n",
      "S152  0.000637       Normal\n",
      "S153  0.222000       Normal\n",
      "\n",
      "[153 rows x 2 columns]\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'site' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-f37d9c9890d4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mphos_boxplot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mboxplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgenedf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"Mutation\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msite\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0mphos_boxplot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_title\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" mutation and \"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0msite\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" transcriptomics levels\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mphos_boxplot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstripplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgenedf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"Mutation\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msite\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mjitter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\".3\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'site' is not defined"
     ]
    }
   ],
   "source": [
    "genedf = CPTAC.compare_mutations(proteomics, gene, protein)\n",
    "genedf = genedf.drop(\"Mutation\", axis = 1)\n",
    "print (genedf)\n",
    "\n",
    "\n",
    "phos_boxplot = sns.boxplot(data=genedf, x=\"Mutation\",y=site)\n",
    "phos_boxplot.set_title(gene + \" mutation and \" + site + \" transcriptomics levels\")\n",
    "phos_boxplot = sns.stripplot(data=genedf, x=\"Mutation\", y=site,jitter=True, color=\".3\")\n",
    "phos_boxplot.set(xlabel=\"Mutation\",ylabel=\"Transcriptomics\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
