{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter CPTAC.embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import math\n",
    "import seaborn as sns\n",
    "import CPTAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "idx\n",
      "S001    FIGO grade 1\n",
      "S002    FIGO grade 1\n",
      "S003    FIGO grade 2\n",
      "S004             NaN\n",
      "S005    FIGO grade 2\n",
      "S006             NaN\n",
      "S007    FIGO grade 1\n",
      "S008    FIGO grade 2\n",
      "S009             NaN\n",
      "S010    FIGO grade 1\n",
      "S011    FIGO grade 1\n",
      "S012    FIGO grade 1\n",
      "S013             NaN\n",
      "S014    FIGO grade 2\n",
      "S015             NaN\n",
      "S016             NaN\n",
      "S017    FIGO grade 1\n",
      "S018    FIGO grade 1\n",
      "S019    FIGO grade 1\n",
      "S020    FIGO grade 2\n",
      "S021    FIGO grade 1\n",
      "S022    FIGO grade 1\n",
      "S023    FIGO grade 1\n",
      "S024    FIGO grade 2\n",
      "S025    FIGO grade 1\n",
      "S026    FIGO grade 2\n",
      "S027    FIGO grade 1\n",
      "S028    FIGO grade 2\n",
      "S029    FIGO grade 2\n",
      "S030    FIGO grade 3\n",
      "S031    FIGO grade 3\n",
      "S032    FIGO grade 1\n",
      "S033    FIGO grade 2\n",
      "S034    FIGO grade 2\n",
      "S035    FIGO grade 2\n",
      "S036    FIGO grade 1\n",
      "S037    FIGO grade 1\n",
      "S038    FIGO grade 2\n",
      "S039    FIGO grade 1\n",
      "S040    FIGO grade 2\n",
      "S041             NaN\n",
      "S042             NaN\n",
      "S043             NaN\n",
      "S044    FIGO grade 1\n",
      "S045    FIGO grade 2\n",
      "S046    FIGO grade 1\n",
      "S047    FIGO grade 2\n",
      "S048    FIGO grade 1\n",
      "S049    FIGO grade 1\n",
      "S050    FIGO grade 2\n",
      "S051    FIGO grade 3\n",
      "S052             NaN\n",
      "S053    FIGO grade 1\n",
      "S054    FIGO grade 3\n",
      "S055    FIGO grade 1\n",
      "S056    FIGO grade 1\n",
      "S057    FIGO grade 2\n",
      "S058             NaN\n",
      "S059    FIGO grade 2\n",
      "S060    FIGO grade 2\n",
      "S061    FIGO grade 1\n",
      "S062    FIGO grade 1\n",
      "S063    FIGO grade 2\n",
      "S064    FIGO grade 2\n",
      "S065    FIGO grade 1\n",
      "S066    FIGO grade 3\n",
      "S067    FIGO grade 1\n",
      "S068    FIGO grade 2\n",
      "S069    FIGO grade 2\n",
      "S070    FIGO grade 3\n",
      "S071             NaN\n",
      "S072             NaN\n",
      "S073    FIGO grade 2\n",
      "S074    FIGO grade 2\n",
      "S075    FIGO grade 1\n",
      "S076    FIGO grade 1\n",
      "S077    FIGO grade 1\n",
      "S078    FIGO grade 2\n",
      "S079    FIGO grade 2\n",
      "S080    FIGO grade 1\n",
      "S081    FIGO grade 2\n",
      "S082    FIGO grade 1\n",
      "S083    FIGO grade 2\n",
      "S084    FIGO grade 2\n",
      "S085    FIGO grade 2\n",
      "S086    FIGO grade 2\n",
      "S087    FIGO grade 1\n",
      "S088    FIGO grade 2\n",
      "S089    FIGO grade 2\n",
      "S090    FIGO grade 1\n",
      "S091    FIGO grade 2\n",
      "S092    FIGO grade 2\n",
      "S093    FIGO grade 1\n",
      "S094    FIGO grade 3\n",
      "S095    FIGO grade 2\n",
      "S096             NaN\n",
      "S097             NaN\n",
      "S098    FIGO grade 2\n",
      "S099    FIGO grade 2\n",
      "S100    FIGO grade 3\n",
      "S101    FIGO grade 2\n",
      "S102             NaN\n",
      "S103             NaN\n",
      "S104             NaN\n",
      "S105             NaN\n",
      "S106             NaN\n",
      "S107             NaN\n",
      "S108             NaN\n",
      "S109             NaN\n",
      "S110             NaN\n",
      "S111             NaN\n",
      "S112             NaN\n",
      "S113             NaN\n",
      "S114             NaN\n",
      "S115             NaN\n",
      "S116             NaN\n",
      "S117             NaN\n",
      "S118             NaN\n",
      "S119             NaN\n",
      "S120             NaN\n",
      "S121             NaN\n",
      "S122             NaN\n",
      "S123             NaN\n",
      "S124             NaN\n",
      "S125             NaN\n",
      "S126             NaN\n",
      "S127             NaN\n",
      "S128             NaN\n",
      "S129             NaN\n",
      "S130             NaN\n",
      "S131             NaN\n",
      "S132             NaN\n",
      "S133             NaN\n",
      "S134             NaN\n",
      "S135             NaN\n",
      "S136             NaN\n",
      "S137             NaN\n",
      "S138             NaN\n",
      "S139             NaN\n",
      "S140             NaN\n",
      "S141             NaN\n",
      "S142             NaN\n",
      "S143             NaN\n",
      "S144             NaN\n",
      "S145             NaN\n",
      "S146             NaN\n",
      "S147             NaN\n",
      "S148             NaN\n",
      "S149             NaN\n",
      "S150             NaN\n",
      "S151             NaN\n",
      "S152             NaN\n",
      "S153             NaN\n",
      "Name: Histologic_Grade_FIGO, dtype: object\n",
      "Proteomics_Participant_ID\n",
      "Proteomics_TMT_batch\n",
      "Proteomics_TMT_plex\n",
      "Proteomics_TMT_channel\n",
      "Proteomics_Parent_Sample_IDs\n",
      "Proteomics_Aliquot_ID\n",
      "Proteomics_Tumor_Normal\n",
      "Proteomics_OCT\n",
      "Country\n",
      "Histologic_Grade_FIGO\n",
      "Myometrial_invasion_Specify\n",
      "Histologic_type\n",
      "Treatment_naive\n",
      "Tumor_purity\n",
      "Case_quality\n",
      "Path_Stage_Primary_Tumor-pT\n",
      "Path_Stage_Reg_Lymph_Nodes-pN\n",
      "Clin_Stage_Dist_Mets-cM\n",
      "Path_Stage_Dist_Mets-pM\n",
      "tumor_Stage-Pathological\n",
      "FIGO_stage\n",
      "LVSI\n",
      "BMI\n",
      "Age\n",
      "Diabetes\n",
      "Race\n",
      "Ethnicity\n",
      "Gender\n",
      "Tumor_Site\n",
      "Tumor_Site_Other\n",
      "Tumor_Focality\n",
      "Tumor_Size_cm\n",
      "Estrogen_Receptor\n",
      "Estrogen_Receptor_%\n",
      "Progesterone_Receptor\n",
      "Progesterone_Receptor_%\n",
      "MLH1\n",
      "MLH2\n",
      "MSH6\n",
      "PMS2\n",
      "p53\n",
      "Other_IHC_specify\n",
      "MLH1_Promoter_Hypermethylation\n",
      "Num_full_term_pregnancies\n",
      "EPIC_Bcells\n",
      "EPIC_CAFs\n",
      "EPIC_CD4_Tcells\n",
      "EPIC_CD8_Tcells\n",
      "EPIC_Endothelial\n",
      "EPIC_Macrophages\n",
      "EPIC_NKcells\n",
      "EPIC_otherCells\n",
      "CIBERSORT_B _cells _naive\n",
      "CIBERSORT_B _cells _memory\n",
      "CIBERSORT_Plasma _cells\n",
      "CIBERSORT_T _cells _CD8\n",
      "CIBERSORT_T _cells _CD4 _naive\n",
      "CIBERSORT_T _cells _CD4 _memory _resting\n",
      "CIBERSORT_T _cells _CD4 _memory _activated\n",
      "CIBERSORT_T _cells _follicular _helper\n",
      "CIBERSORT_T _cells _regulatory _(Tregs)\n",
      "CIBERSORT_T _cells _gamma _delta\n",
      "CIBERSORT_NK _cells _resting\n",
      "CIBERSORT_NK _cells _activated\n",
      "CIBERSORT_Monocytes\n",
      "CIBERSORT_Macrophages _M0\n",
      "CIBERSORT_Macrophages _M1\n",
      "CIBERSORT_Macrophages _M2\n",
      "CIBERSORT_Dendritic _cells _resting\n",
      "CIBERSORT_Dendritic _cells _activated\n",
      "CIBERSORT_Mast _cells _resting\n",
      "CIBERSORT_Mast _cells _activated\n",
      "CIBERSORT_Eosinophils\n",
      "CIBERSORT_Neutrophils\n",
      "CIBERSORT_Absolute _score\n",
      "ESTIMATE_StromalScore\n",
      "ESTIMATE_ImmuneScore\n",
      "ESTIMATE_ESTIMATEScore\n",
      "Stemness_score\n",
      "Pathway_activity_EGFR\n",
      "Pathway_activity_Hypoxia\n",
      "Pathway_activity_JAK.STAT\n",
      "Pathway_activity_MAPK\n",
      "Pathway_activity_NFkB\n",
      "Pathway_activity_PI3K\n",
      "Pathway_activity_TGFb\n",
      "Pathway_activity_TNFa\n",
      "Pathway_activity_Trail\n",
      "Pathway_activity_VEGF\n",
      "Pathway_activity_p53\n",
      "TP53_ATM\n",
      "TP53_CHEK2\n",
      "TP53_MDM4\n",
      "TP53_RPS6KA3\n",
      "TP53_TP53\n",
      "TP53_pathway\n",
      "PI3K_AKT1\n",
      "PI3K_AKT2\n",
      "PI3K_AKT3\n",
      "PI3K_DEPDC5\n",
      "PI3K_DEPTOR\n",
      "PI3K_INPP4B\n",
      "PI3K_MAPKAP1\n",
      "PI3K_MLST8\n",
      "PI3K_MTOR\n",
      "PI3K_NPRL2\n",
      "PI3K_NPRL3\n",
      "PI3K_PDK1\n",
      "PI3K_PIK3CA\n",
      "PI3K_PIK3CB\n",
      "PI3K_PIK3R1\n",
      "PI3K_PIK3R2\n",
      "PI3K_PPP2R1A\n",
      "PI3K_PTEN\n",
      "PI3K_RHEB\n",
      "PI3K_RICTOR\n",
      "PI3K_RPS6\n",
      "PI3K_RPS6KB1\n",
      "PI3K_RPTOR\n",
      "PI3K_STK11\n",
      "PI3K_TSC1\n",
      "PI3K_TSC2\n",
      "PI3K_pathway\n",
      "HRD_BRCA1\n",
      "HRD_BRCA2\n",
      "HRD_BRCA1_or_BRCA2\n",
      "Log2_SNP\n",
      "Log2_INDEL\n",
      "Mutation_signature_C>A\n",
      "Mutation_signature_C>G\n",
      "Mutation_signature_C>T\n",
      "Mutation_signature_T>C\n",
      "Mutation_signature_T>A\n",
      "Mutation_signature_T>G\n",
      "CNV_clustering\n",
      "Purity_Immune\n",
      "Purity_Cancer\n",
      "Purity_Stroma\n",
      "MSI_status\n",
      "POLE_subtype\n",
      "JAK1_MS_INDEL\n",
      "JAK1_Mutation\n",
      "WXS_normal_sample_type\n",
      "WXS_normal_filename\n",
      "WXS_normal_UUID\n",
      "WXS_tumor_sample_type\n",
      "WXS_tumor_filename\n",
      "WXS_tumor_UUID\n",
      "WGS_normal_sample_type\n",
      "WGS_normal_UUID\n",
      "WGS_tumor_sample_type\n",
      "WGS_tumor_UUID\n",
      "RNAseq_R1_sample_type\n",
      "RNAseq_R1_filename\n",
      "RNAseq_R1_UUID\n",
      "RNAseq_R2_sample_type\n",
      "RNAseq_R2_filename\n",
      "RNAseq_R2_UUID\n",
      "miRNAseq_sample_type\n",
      "miRNAseq_UUID\n",
      "Methylation_available\n",
      "Methylation_quality\n"
     ]
    }
   ],
   "source": [
    "clinical = CPTAC.get_clinical()\n",
    "somatic = CPTAC.get_somatic()\n",
    "\n",
    "\n",
    "gene = 'TP53'\n",
    "protein = 'TP53'\n",
    "\n",
    "pd.options.display.max_columns = 4000\n",
    "pd.options.display.max_rows = 4000\n",
    "\n",
    "columns = CPTAC.get_clinical_cols()\n",
    "print(clinical['Histologic_Grade_FIGO'])\n",
    "for value in columns:\n",
    "    print(value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hotspot_mutations = ['p.P250L', 'p.R248W', 'p.S241C', 'p.S241del', 'p.K164E', 'p.R249S', 'p.R273H', 'p.R282W',\n",
    "                    'p.S127Y', 'p.G244D', 'p.A159P']\n",
    "mutated_hotspot = somatic.loc[(somatic['Location'].isin(hotspot_mutations)) & (somatic['Gene'] == gene)]\n",
    "hotspot_patients = mutated_hotspot['Clinical_Patient_Key']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "###Tumor Attributes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jonjarman/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "interest = list()\n",
    "for header in columns:\n",
    "    if header[0:5] == 'Tumor':\n",
    "        interest.append(header)\n",
    "        \n",
    "        \n",
    "p53_mutation = CPTAC.get_cohort_clinical(interest)\n",
    "p53_mutation['Within_Hotspot'] = False\n",
    "p53_mutation.loc[hotspot_patients, 'Within_Hotspot'] = True\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ttest_indResult(statistic=0.38204990429970553, pvalue=0.7035655006095476)\n"
     ]
    }
   ],
   "source": [
    "hotspot = p53_mutation.loc[p53_mutation['Within_Hotspot'] == True]\n",
    "non_hotspot = p53_mutation.loc[p53_mutation['Within_Hotspot'] == False]\n",
    "#hotspot = hotspot.dropna(axis=0,how = 'any')\n",
    "#non_hotspsot = non_hotspot.dropna(axis=0, how = 'any')\n",
    "ttest = scipy.stats.ttest_ind(hotspot['Tumor_Size_cm'], non_hotspot['Tumor_Size_cm'])\n",
    "print(ttest)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
