{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Methods - Analyze Interesting Proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### From our previous tests, we have identified several proteins of interest to investigate more thoroughly. We chose a few that stood out/were known to be related to the cell cycle: AURKA, BORA, XPO1, CDK1, TPX2, PLK1, CABLES1, XRN2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Standard imports, including CPTAC package"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the CPTAC data service package. Available datasets may be\n",
      "viewed using CPTAC.list(). In order to access a specific data set,\n",
      "import a CPTAC subfolder using either 'import CPTAC.Dataset' or 'from\n",
      "CPTAC import Dataset'.\n",
      "******\n",
      "Version: 0.2.5\n",
      "******\n",
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import CPTAC.Endometrial as CPTAC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up initial dataframes and variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene='TP53'\n",
    "somatic = CPTAC.get_somatic()\n",
    "proteomics = CPTAC.get_proteomics()\n",
    "phos = CPTAC.get_phosphoproteomics()\n",
    "transcriptomics = CPTAC.get_transcriptomics()\n",
    "\n",
    "# A list of mutations present in the hotspot according to our Hotspot3D output\n",
    "hotspot_mutations = ['p.P250L', 'p.R248W', 'p.S241C', 'p.241del','p.K164E', 'p.R249S', 'p.R273H', 'p.R282W',\n",
    "                    'p.S127Y', 'p.G244D', 'p.A159P']\n",
    "mutated_hotspot = somatic.loc[(somatic['Location'].isin(hotspot_mutations)) & (somatic['Gene'] == gene)]\n",
    "\n",
    "# Get a list of patient identifiers for those who have mutations in the hotspot\n",
    "hotspot_patients = mutated_hotspot['Clinical_Patient_Key']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## AURKA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Proteomics analysis\n",
    "### Note: AURKA not in phosphoproteomics data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All TP53 mutation and AURKA: 1.5834367365528105e-05\n",
      "Hotspot TP53 mutation and AURKA: 3.53407989762094e-07\n",
      "Non-hotspot TP53 mutation and AURKA: 0.07102126562678672\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAssAAAINCAYAAADIuTJyAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvIxREBQAAIABJREFUeJzs3XmcHXWV///X6SW9pdNZSUICBAg7ymKQLUhQYNiEDLggMyq4zW9c4jqjAyiI8BUcBzQwjqIsogioIyCMyCKETRYDATEsphMCCQnZuzud3vue3x9Vt/v27Vvdt7pvp3p5Px+Pftyuqk9VnVu3qu65n/rUp8zdERERERGR3oqSDkBEREREZLhSsiwiIiIiEkHJsoiIiIhIBCXLIiIiIiIRlCyLiIiIiERQsiwiIiIiEkHJ8ihgZqvNzPP4W5Bj3k+Z2XNmtiOj3MRwWrmZXWlmtWbWGk57Yae/wQSY2XFm9qCZbTOzVPjeF0aUnZPn9s/+uzSc//wc01Lhuv9sZl8ys3E51tvf5/7jHPPMNrPvmdmfzOyN8HNvMbNVZnaTmb2j4BtzjDCzBeF2X5J0LCONmdWYWVM+55iM7by6n3Jdx2WOabmOlyYzW2lmN5vZoX0sd0lY/vyI6bPN7JWwzONmVhNR7jMZ6/5SX+9FAuFnE7ntC7yu9Hn55qFeVyFl7Perk45lNClJOgApqPuBt/uY3mOamZ0B/BRoAR4EtoaT2sLXy4GvAhuAu4Em4M0Cxtuv8IDfA9jT3VfvpHXOAu4BJgCPA28AKaLfeyPw8xzj5wLHEmy/P+aYnp0UZJYrAfYGjgKOBs41s/e5+44cy/nfMIZsT+UYtz/wb8AW4BXgGaAceAdwPvDPZvbP7n5HjnlHhHRy5O5W4OWuZifvi2PIR4CK8P9DzOxwd39+J6w389jZBZgHfBz4JzM7z91/E2dhZjYXeIhgP7kfONvdmyKKfyLr/x/EWddwFv5gPB44wd2XJBuNyOApWR5drox5Yvpg+LrI3X/ax/Tj3H3FoCIbWU4GaoBfufs/9VfY3TcTJJo9hLUfxwKvunuv6Tn0KmdmxwAPAEcC/wFcnGO+r8VI3l4CDgX+6hlPJDKzIuCLwNXAz8zsD+6+Pc9lSuBZ4ACCH5USTzpxfAuYFQ7vjGS5x7FjZuOBXwJnAf9jZve6e3M+CwqvyjwAzAB+C/yTu7dFlD2A4JjeAXQA7zCzee6+dDBvZgz4D+BKYH3SgcjYomYYY9tu4WtUIrwbwBhLlKH/7bLTuPufCRJY6P7xMpjlbXD3Fz3r0Z3unnL3a4BVwHiC2myJwd2b3P1Vd9+pV19GOjM7CDiCoIb3/HD0eWZWtrNjcfdG4HPh4BTyPA7M7EjgUYJE+Qbgw1GJcuiT4etvgNvD/z8RUVZC7r4+PMbqk45FxhYly2NQut0XcEI46pHMdrTptrCAheUj2z2b2T+Y2e/NbIOZtZnZejO7ra+2r2Y2xcwuM7NlZtYQtptdEcZ1TFjm/DCGPcLZXs+KY06M93u6md1nZpvDGNeY2c/D2p3Mcul1fjscdUnG+pbku74hkK5t2qPPUoXREb625DtDuM+k953Z4ee4PmwD+ryZfSCj7LFm9gcz2xJOf8TMjsixzD7b/+Zql5eOI2O4R5vUjPHVYXvRuyxoj99kZo3h/niRmVVkrSuvfTGPmA8ys1vC/a813B//YGanRpTvap9pZnPN7FfhcdZqZq+a2dfDKwKxmFmpmX3ezJ4Jj79mC9rYXmlmk3OU79rWFvismb0QbrdtZna3mR0cN44MXYmjuz9E0DxoEvCPg1jmgLn7WwTNlACm91fezN5L0PRiEnC1u3/K3VN9lC8B/jkcvAm4Mfz/I2ZWHjferONvTzP7ZbiftJjZcjP7arjOvubbw4J7FtaaWYeZ/SCrbL7n0AXhsXJ8OCrzuyXX98duZvZDM3st3A8bzOzJcJ/v1YzKItosZ72X6Wb2k/C9tJrZ6+G+HXvb9iXf2M2s2MzeCuM7pI/l/TYs87ms8WZm55rZA+H2bzWzN83spxbjezBc1n7h5/ZG+DluD4/rO83snLjbYCxRM4yx6Ynw9RSCL4PMts4vEFxCnErQdg96tsftavdsZj8EFhEkWH8B1hK00z0XWGhm57j7HzJXbGaHAf8HzCRoI72EIDHbg6DdIsCfgdpwvR8AqujdLjdXG91ezOy7wDcI2hw/QXCZ953Ax4APmdkH3P3/wuLpdR4KHAK8SHe74lfzWd8QmRC+tkZM/0SY5BQRtKu+392XxV2JmX0S2JfgEudzA4hzj3C+RoJattkEzVB+bWbnEcR/B8E2fZBgGy8g+EI93N3/PoB1ZnqB4PPLtd9mOgT4CbAReI3gx8gUgsvilwNnmtnx7p7+wTDofdHMzgR+DZQBywnaws8G/gE41cwud/dvRsx+KPBDYDPwCMExO5/gcvRs4Av9rT8jjnLgPoLt3hQurwk4Dvg6Qdv497r7qohF3Ax8GHiM4MrLEcCZwAIzO6yP+aLiKaU7cbw5fL0J+B5BTevtOWYbUhb8ABkfDm7op2zm53qJu1+WxyrOIPgMVwGPu7ub2cvAgcDZwK8GGPqeBPtyC8F5dQJBhcj3gfnh+ThXEr8PsCyc70mCvKAuPTHmOfRtgmMl13cL9Pz+OAG4k6DJWy3B/RrjCe7TuAl4b7iOOHYjOAcZwffIBIJj5esE2/fMmMvLKU7s7t5pZr8IYzgf+HKO5U0G3k9wv9BtGeNLCY6Bs4Fmgs93A3Aw8CngHDM7OZ/mOxZUYD0JVBN8n90DOEGzp38guGfgf2NtiLHE3fU3wv+A1QQ7/YKY8y3pa75wmkdM+//C6X8D9s+athBoB7YBkzLGVwNrwvn+B6jImm8aMD/ivc0ZwHY5LZy3EXhP1rR/C6fVAbtkTbs0nHbpID+X88PlLBlMOYITmAOPRmybXH93Z277iOXeQJCg/C9B0ugEX2bHxnyfl2as9wdAcca0fw3HryH4cfTBjGlFBF8EDtyQtcwF/WyTOeH01XH223D6bIIvs6Ks8RMJEkkHvp5jvj73xaiYCS7N14fTvpJjnh3htH/ImnZzxna9NDNe4D1AZ/i3W4zP6nvh8l4BZmWMT39ROvBUxLZ2ggRv74xpZQQ/fh346QCOkbPDeWsBy9he7eF7272P7dzrs4+KO2ofyfVZEiR6DmwCKnNMXxJOvzeMMwV8IcZ7/n04/zczxn0tHPfQALbhpRnv57dAeca0fQgqMRz4bB/z3QSMy7HsgZ5D09toQUTM6cqSDoIft5YxbTeC5N2B8yOOiezxme/lp5nvheA+gu3htLzPbXSfl28ebOwEN1Y7QaJbkmNdnw2n/yZr/JXh+EeB2VnTPk/3sVOSMT6936/OKn9jOP4/cqx/PHB03H1vLP0lHoD+CvAh9p00pf/qcszX3wkt6oumGFgXTj8wYt7rwulfyBj3pXDcU5knmDzf25wBbJc/hfNeETH96XD6RVnj0yfeSwf5uaRPtkviliOo3dkX+O+Mz/Afc2zjDwN7EfRosTfwmfCE7AS1f0V9rLcjax95PWpf6Cf+9PZaTdYXbrivbA6n/yrHvIeF01ZljV/Q17aL+kLoa7/N873sE87/l7j7YlTMwDfD8U9GzHdVOP3BrPE3h+OfzXW8AH8Ip38sz/dWQXfScFKO6VMJkqIeSQU9k+Uzcsz37lyfYZ4x3RvOe3HW+HRC+a0+tnOvzz5iH+m1L2S8nzkZ46aFx9M6gqsgZ0Ysd0nWcXNHjPc7ne4Ee/eI8Tn3rz6WmT7+mshKWsPpF4TTV0TMtwWojlj2QM+h6W20oJ99/qqI6fPC6c9FHBPnR7yXN8mqhAmn/yhqf+pju55P7mR5oLE/FY7vtV8RHOMOnJ4xbnL4mW7P9blmHT/vz7Hfr84qm/5Re2ic/Ut/wZ/aLI8u9xNcAsv1N9BLe7kcSvDrerm7vxxR5tHwNfMGmVPC1xs8PHqHSthG79hw8OaIYjeFrwuGMpYYjrfu9rXtBLW9nyW4NPdVd78zs7C7f97d73D3Ve7e4u4r3f16gkvjdQSX1s+OWpm7l3jQvdouwIkEXzSPmNl/DjD+hz3rpiZ37yRIMiF393npmyh3HeA6ByRsBzjfzC40sx+F7TVvpru3kX0LuLrjw9ebI6an26zON7PiHNP/EHG8pJsG5bvt3kVQg7TO3R/MnuhBry73hIMLcszfQe7PMG4cAJjZTIJzQoreTWbSx2bOtqsF9HrGMbeR4EpHOUEt2+/7mfex8PVDZvbVPNf3cYIfwn/yjBtB3X0DwY8fI0fPOnl6wN035hh/K8E2nmtBt5jZHvQcPd8M8Tn0tPA1qmu+dHOuQ2O2NX7Yc/deMqB9NMJAY785fD0/s3DY7vsIgqt6mcfXCQQ/cB+N+Fwh9/dslGfD1x+b2UmWwA20I5naLI8ucbuOG6i9wteDLEeH/1mmZfy/R/i6M9r/TiG4RJwi6Cc5l5Xha64vkCRk9rOcAhqAl4Hfu3tf/Wf34O5vmtlNBG3jTiO4NNtX+U3An8zscYJ2fl8zs8fzSBayrY0Y3xg13d0bw1xop524zWw68DvgmD6KTehjWlzp/ev1iOmvE3ze5QT7bfYXY1TvGg3ha77JRH9xQN/HxHp378ge6e4NA/wMP05w5eFBd1+TNe1egmYQexIkDQ9nrjJ87S+JzifJTrc/LyZonnMcwY16t5vZ0e6+pY95byI4thYD3zczd/er+ygP3YnSTTmm3UTQpvZ8M/v2ACoUcn6u7t5mZusJPtPZBG2OM0WdH4fyHJr+DvlLHr+FptA75iiFOlb6MtDYbydopna6mU3J2Lc+Hr7+MqxcyF7P6TG/Z6P8J8H+/T6CLg5bLXgA0KPhul/KYxljlpJlGYh07ddbBHeB9yWpG+Myz2JRJ5qhrLEaiHz7Y85rWeFr3l9i4ZfqbQQ1kOcQXAqPI7IHgDynxzGYq2I/I0iUnyS4fPsiQTOldguelBh1I+VApfezgV5NKdR2yyeOvo6JQn5+0J047mdmT+SYnv5++gQ9k+V0P9ZV/Sw/fZNeXzdgZvezvB9B04N9gB/TT3eN7n5tmDAtBv7LzIhKmC3o6Sfde8SXzeyzWUXS73cPgoSmv3PrQOT67KP6kR7Kc2j6O+QO+u95J87xWOh9NJcBxe7u9WZ2J8GN7OcB14Y3k2bf4Jq9ntcImrv05Zl+puPBw3FOtKCbw1MIrhocTXBj87+b2SWe3w2qY5KSZRmIdC3Q+pjJ3RsENzrsR3ePHENlM8GJqoygDVeuPpP3DF/zrbUYSaaEr3n1GpJhU/i6SwFjGYh0c47xEdP3iBjfJzOrIqht7yRof1uXVWTuQJbbj7UE+/xeBIlYtjkEyX8L3U/RHArpmv09+yizU44JM5tPsE0Adg//opxtZjXe3bduuvZwiplNcvdtEfPtE75m11pHcvfXzOzjBInqB8zsOHd/vJ95shNm96DP8myZ/SjP6yeUTxA/WZ6Ta2T4A3BmOLguxvKG8hy6huBY+467L485b9IGE/vNBMny+cC1wEkEFRpLcywrvd++VMBKFNz9GcLkOtw3ziO4KfJSM7vD3V8r1LpGE7VZloF4luCmkMMseLxrvu4PXz8Rox1iOmmK9cMuvFz8ZDgY1f3Q+eHrkjjLHu7CbZvu2/gvMWd/b/ia9ANZ0l++e4fdJ2U7Lce4tHboanOZrYbgvLc9R6IM0NcTGwe0L9LdrjBqP7wgfH0iVzOHAkq3pZxlZu/LnmhmUwi6r4KhPybSieMN7m5RfwT7bwXd3Uqm2/e+Eg5Gtsmn+xh4JE5g7v4ngt5kAL6T5zzXEnSjCXC1mfXoHiz8kfahcPC4Pt7vQWGZfzSziXHiBk42s1yX4z9CsM+vdPeoplK9DPIc2t+xcl/4OugHLSVgMLE/RPCj9XAL+iZPN8G4OaJsO0FtcNx9IS/u3ubuNxPUXBtBl4CSg5Jlic3d2wm+RIqBu8zs3dllzKzKzD5iPTut/xlBzcYxBJegyrPmmRbWOGVKJ00HEF/6cuiXzOzYzAlm9hWCS1D1YVwjipmdZWaH5xg/meBmqcMJ2undmDX982bWq1bLzMrCm5Q+SlDremN2mZ3J3d8gaA85kaBbrS5mtpDuxCSXvvaZDQRdGk4M+37OXO4pwFcGuNy+/JTgjvb5ZtYjbjN7D939JP9XzOXGEt749ONw8IfhDXbpOMoJunMcDzzt7k/mWERBhIljOtH4RT/F09Ozn26Xvgn1u+Fl5czlm5l9mqC/91aCGt+4LiQ4Do4P+9TtV46E+UsZkz9I0HXm63QnoLmW8TLBY77LCWr84qgErsu8ccvM9qY74f9hzOXBwM+h/R0r/0lwfrrQzD6X64etmR1lZsMxmR5w7B70c31LOLiIoJvVHn0rZ5TdQNAb0kTg92a2f471TDKzT4X3YfTJgocJ7Zdj/F50/0iLapsuSXTBob/C/tHdpdUfCX6hRv2dnDXfEgbQdVzG9KvTZQjaff4OuIugBivdd+wpWfPMo7trs80ENTh3EPyybaV3Nz2LwrINBDfT/Cz8m5Lntkn3U9kZvt9fAS+F45rJ3RXWpSTcdVwey/4B3V0lLSE42T5Kd3++dcD7csyX/sxrw21/K0HTgI3h+FbgkzHfZ5/ba6D7GUGCkQqnP0dw9/kL4bjLie46Lr1fpns3+Bnws4zpX83Yb/8c7hPPhMNX9BFPn/sifXR3B5xF0MzCgb+G61wS7pdOcEk3e56bydFN1mD2U4Ik7JFwvkaCdul30N0V5BvAXlnzzIna1v19hhFlP5Gxrj67kCS4cak9LH9w1rRrw/EpghroX4X7yMqMffm8vuKlj27a6O6X9rGI/Tnqc/lCxvK/FI57LBy+LI/tk+5ic2nM4+8Wgit+a8PP9P8IznEefs7Z/Yrntf8wsHPomeH0lnDd6WNlv4wyJ4Txerj/PUhwvD5GkGw7cHs+x0R/74WIbuD6ed+R8wwk9ox5091Tpv9+00cMpeFn6QS90SwN9/F7CM6FreG0/TPmmUPuruNeCMevpOe5P72M2/LdNmPxL/EA9FeADzG/fpa7TtwZ8y1hEMlyWOY9BInam+FBV0dwifR2gpqRqhzz7AJ8l+CBJk0EX9p/J7gb/KisskUE3Xm9Qney0eeXXI71nUHwQ2ILwa/4tQRfLFF9RPd54o2x3vTJdkkhymXNs4Dgy/wFgqSwnaD2chnBl9usiPlOA64n+HGzKZyvIVzONcC+A3iffW6vwexnBF+6T4X7SUO4rH+I+kII56kgqKVdFX7evZZPcAPjUwQ/LhoIkuZ/6iue/vZF+u8b+mCCmtK3wri2EFzSPS2i/M0UOFkO5yslSOieDfeZFoIbQq8ix4/QvrZ1Pp9hjrKPh+X/X57l008auzrHtJMJfrisITj/7CC4IerHwAH9xUvfyfJudCeb78sYn96fc34uYZnMhDnzx1m/xxfdfS478M44xx9Bu/jbCc4JreG++m9A6WD2H2KeQ8N5/pXgvNKU8f4XZJWZQfAD9YWMfXF1uI2/QcYDcPo6Jvp7LxQ4WR5I7FnzPpGxTU7PI5YzCSqi1tF97vgbwXfmwszPl+hk+YzwuFhGcO5vJfjefoCgYiKyT379edcTk0RERGSEMbNLgUuAb7v7pclGIzI6qc2yiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCbZRERERGRCKPuCX5Tp071OXPmJB2GiIiIiAxjzz333GZ3z/Uwnx5GXbI8Z84cli5dmnQYIiIiIjKMmVleD2JRm2URERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQhKlkVEREREIihZFhEZRRobG3n66adZs2ZN0qGIiIwKJUkHICIihfHaa69xySWX0NTUBMB5553Hueeem3BUIiIjm7l70jEU1Lx583zp0qVJhyEio9jixYupra1NOoxeNmzYQHNzc49xxcXFmBmzZ89OKKr45s6dy6JFi5IOQ0RGOTN7zt3n9VdONcsiIqNEZ2dnr3EtLS0JRCIiMnooWRYRiWm41nred999/M///E/X8CGHHEJ9fT0Q1IaLiEh8SpZFREaJU089lerqap599llmz57NGWecwTe+8Y2kwxIRGdGULIuIjCLz589n/vz5SYchIjJqqOs4EREREZEISpZFRERERCIM62TZzMrN7Fkze9HMlpvZt5OOSURERETGjuHeZrkVeK+7N5pZKfCEmd3n7k8nHZiIiIiIjH7DOln24IkpjeFgafg3up6iIiIiIiLD1rBuhgFgZsVm9gKwEXjQ3Z/JUeYzZrbUzJZu2rRp5wcpIiIiIqPSsE+W3b3T3Q8FZgPvNrODc5S53t3nufu8adOm7fwgRURERGRUGvbJcpq71wFLgFMSDkVERERExohhnSyb2TQzmxj+XwGcCLyabFQiIiIiMlYM6xv8gJnAz82smCCx/7W735twTCIiIiIyRgzrZNnd/woclnQcIiIiIjI2DetmGCIiIiIiSVKyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIiIiIhGULIuIiIiIRFCyLCIiIiISQcmyiIiIiEgEJcsiIsPEli1beOCBB1i2bBnunnQ4IiIClCQdgIiIwIoVK7joootoaWkBYMGCBXzlK19JOCoREVHNsojIMHDnnXd2JcoAS5YsYd26dQlGJCIioGRZRGRYaG1t7TWura0tgUhERCSTkmURkWHgtNNOo6io+5R88MEHM2fOnOQCEhERQG2WRUSGhXe9611cddVV/PnPf2aXXXbhxBNPTDokERFBybKIyLCx3377sd9++yUdhoiIZFAzDBERERGRCEqWRUREREQiKFkWEREREYmgZFlEREREJIKSZRERERGRCMM6WTaz3czsETN7xcyWm9kXk45JRERERMaO4d51XAfwVXd/3syqgefM7EF3fznpwERERERk9BvWNcvuvt7dnw//3w68AsxKNioRERERGSuGdbKcyczmAIcBz+SY9hkzW2pmSzdt2rSzQxMRERGRUWpEJMtmNh74X+BL7t6QPd3dr3f3ee4+b9q0aTs/QBEREREZlYZ9smxmpQSJ8q3u/ruk4xERERGRsWNYJ8tmZsANwCvufnXS8YiIiIjI2DKsk2XgWOCjwHvN7IXw77SkgxIRERGRsWFYdx3n7k8AlnQcIjK0Fi9eTG1tbdJhjEorVqwAYNGiRQlHMjrNnTtX21ZklBvWybKIjA21tbX8/W/Ps/v4zqRDGXXGtQcXEFtW/yXhSEafNxuLkw5BRHYCJcsiMizsPr6Ti+c1Jh2GSN4uXzo+6RBEZCcY7m2WRUREREQSo2RZRERERCSCkmURERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQhKlkVEREREIihZFhERERGJoGRZRERERCSCkmURERERkQixkmUzO97M7jWzjWbWbmadOf46hipYEREREZGdqSTfgmZ2OnAXUAy8CbwGKDEWERlGWjuN5fWVlBen2H9CM0WWdEQiIiNb3skycCnQDpzu7g8MTTgiIjJQ9W3FfP/VXdnWFpza96lu5gv7vq2EWURkEOI0wzgYuEOJsojI8PT4puquRBlgxfYKXm2oSDAiEZGRL06y3AhsHapARERkcFo7e5/SWzpVrSwiMhhxkuU/AUcPVSAiIjI4R03dTqmluoYnjevg4JrmBCMSERn54rRZ/jrwrJldDFzh7j5EMYmIyADMqmznawes4+kt1ZQXpThul+2MK9apWkRkMOIky5cAy4FvA58wsxeAuhzl3N0/WYjgREQknlmV7ZxTqRZzIiKFEidZPj/j/znhXy4OKFkWERERkREvTrK855BFISIiIiIyDOWdLLv7G0MZiIiIiIjIcBPrcdciIiLNncbzW6v4e0M5utVbREa7OM0wADCzo4BPAYcBE4F64DngJnf/c2HDExGR4WRzawn/9cpMtncEXx+HTNzBp+duTDgqEZGhEytZNrPLgf8Asnu5P5Sgh4yr3P3CQgUnImPD2rVr2bG9mMuXjk86FOlHA1U0ZXx1vFhXxSVLJ1JKR4JRJeON7cVUrV2bdBgiMsTyboZhZh8ELgTeJKhZ3guoCF8/FY7/upl9aAjiFBGRYSCV42sj1av+RERk9IhTs/wFYANwhLtvzhi/GrjRzH4P/A34HPDrgkUoIqPe7NmzaelYz8XzGpMORfpRu72dxa/N7EqQp5e3ceFB2ygeg/ny5UvHUz57dtJhiMgQi5MsHwLckpUod3H3zWb2G+BjBYlMRESGnbnVrXxl//U8u7WK6pIUx01rGJOJsoiMHXGS5RKgqZ8yTTGXKSIiI8yc8a3MGd/A603UAAAgAElEQVSadBgiIjtFnK7jaoEzzCznPOH404CVhQhMRERERCRpcZLl24ADgLvNbJ/MCWa2N/Bb4EDgV4ULT0REREQkOXGaTFwNnAKcDpxqZuuA9cAMYBZB4v1EWE5EREREZMTLu2bZ3duAk4CLgNeB2cARwG7h8EXA+8JyIiIiIiIjXqyb8dy9Hfgu8F0zGw/UAPXurv6eRERERGTUGXDPFWGCrCRZREREREatODf4iYiIiIiMKZE1y2a2CnDgRHd/PRzOh7v73gWJTkREREQkQX01wygiSJajhqPoWU4iIiIiMipEJsvuPqevYRERERGR0U5tlkVEREREIgw6WTazUjM7zMz2K0RAIiIiIiLDRd7Jspl9yMx+bWaTM8btDSwHlgIvm9nvzGzA3dGJiIiIiAwncWqWPwHs7+5bM8b9FzAXeAT4K3AWcEHhwhMRERERSU6cZPlA4C/pATObAJwG/NrdTwTeDbyKkmURERERGSXiJMvTgPUZw0cT9KZxO3Q9CvtBQH0si4iIiMioECdZ3g7UZAwfT9Dv8hMZ41qA6gLEJSIiIiKSuDg3460ATjWzMoIk+YPAX919c0aZPYCNBYxPRERERCQxcWqWrwf2IkiaXwn/vzGrzJEEvWOIiIiIiIx4eSfL7v5z4EqgkqA5xnXhHwBm9l5gDkHPGCIiMoqsby7lvnUTeWbzeNpTSUcjIrLzxOoT2d0vBC6MmPwEMAnYMdigRERk+Ph7Qzn/vWIGnW4APLe1is/uuyHhqEREdo6CPe7a3dvcvd7dOwq1TBERSd6jGyd0JcoALzdUsr65NMGIRER2njhP8JtjZqeZWVXGuBIz+7aZvWhmfzazfxyaMEVEJClmOcbt/DBERBIRp2b5EuAXQGvGuIuBbwLvAI4Cfm1mRxUuPBERSdp7p9dTYt0Nld85cQczKtoTjEhEZOeJ02b5aOBP6WYWZlYEfJbgqX0nAzOAh4AvAx8ucJwiIpKQvca3ctFBb/HXukoml3XwzolNSYckIrLTxEmWpwNvZAwfCkwFvu3ua4G1ZnY3cFwB4xMRkWFgWnkH75vRkHQYIiI7XZxmGKUEDyNJOzYcfjhj3FpgZgHiEhERERFJXJxkeS3wzozh04DN7v5KxrhdAFU9iIiIiMioEKcZxr3Al83s+0ALcBJwU1aZ/enZVEMkUmtrKy+++CITJkxg//33TzocERERkV7iJMvfAxYCXwmH3yLoIQMAM9sDOAa4pmDRyai1adMmvv71r7N582YA5s+fz7//+78nHJWIiIhIT3kny+6+0czeAbwvHPWou2/PKDKeIJG+v4DxSUyLFy+mtrY26TD6tXXrVhoaulvsPPHEE6xcuZKysrIEo+rf3LlzWbRoUdJhiIiIyE4S93HXzQTNMXJNWw4sL0RQMvqlUqle4zo7OxOIREREJBkrV67k1ltvZevWrSxYsICFCxcmHZLkECtZTjOz/YEDgPHu/ovChiSDMVJqPf/2t79x8cUXdyXN06dP50c/+hGlpXqEroiIjH6tra1ccsklXVdZV61aRWVlJSeffHLCkUm2WMmymR0K/Aw4LGP0L8JpxwP3AR9293sKFqGMSgcffDBXXHEFV1xxBcXFxVx55ZVKlEVEBBg5TQrT1q5dC8Ds2bPznqe5ublHc0SAG264gXvvzXkBv+DUrDB/eXcdZ2b7AkuA/YAfEiTGmR4DtgIfKFRwMroddNBBTJ06lUmTJjFlypSkwxERERmQ5uZmmpubY82Tq4JIlUbDU5ya5UuAccC73P0VM7sEODU90d3dzJ4CjihwjCIiIjKGjLQaz3S8ixcvjjXf3XffzS9+8Qva2to48MADueiii6iurh6KEGUQ4iTL7wN+l/UQkmxvEvS/LDIgDQ0N/PznP2fFihUcdNBBfPSjH6WysjLpsERERArurLPO4qSTTmLHjh1MmzYt6XAkQpxkeSLBU/z6UkRQ+ywyIFdffTXPP/88AKtXr6apqYkvf/nLCUclIiIyNCorK1UpNMzFedz1RmBuP2UOAtYMPBwZyzo6Oli2bFmPcc8++2xC0YiIiIjES5YfBt5vZvvlmmhmRxA01dBDSWRASkpKmDFjRo9xce4sFhERESm0OMnyd4EO4DEz+1dgVwAzOygcvgfYDny/4FHKmPG5z32OmpoaAKZOncq//Mu/JByRiIiIjGVxHnf9mpmdA9wGXBeONuCv4WsdcLa7v1nwKGXESaVS/O53v+PJJ59kl1124aMf/WiPWuLXX3+dW2+9lXXr1lFVVdU1/p3vfCc33ngjmzZtYvr06RQXFycRvoiIiAgQ/3HXfzSzPYGPA0cBU4B64GngJnffWvgQZSS65557uOWWW4DgcZ61tbVcf/31FBcX09rayre+9S3q6+sBaGtr44EHHuh6alFpaSm77rpr3uvatGkTZWVlTJgwofBvRERERMa02I+7dvc6goeS/LDw4fRmZjcCZwAb3f3gnbFOGbzsG/M2bdrEqlWr2GeffXjttde6EuXM8nEf8dna2sqVV17Jc889R0lJCQsXLuRjH/vYoGMXERERSYvzBL9OM/vVUAYT4WbglATWK4Mwa9asHsOlpaVMnz4dgJkzZ1JUVNRn+Xz88Y9/5LnnngOCnjR++9vfsnLlygFGLCIiItJbnJrl7cAbQxVIFHd/zMzm7Oz1yuCce+65rFixgpUrV1JWVsYnP/nJrmYS06ZN44ILLuh6alFZWRnnnHNO7HWsXdu72++1a9ey9957Dzp+2fnebCzm8qXjkw5j1NnQFPwwnV6ZSjiS0efNxmL2TToIERlycZLlZcCBQxXIYJjZZ4DPAOy+++4JRyMAkydP5pprrmH9+vXU1NT06nD9rLPO4sQTT+RrX/sapaWlA2pv/O53v5v77+/uqbCsrIxDDjlk0LHLzjd3bn9duMtAta1YAUD5nH0SjmT02RftuyJjQZxk+SrgHjM7yd0fHKqABsLdrweuB5g3b54nHI5kmDlzZuS0qqoqSktLB7zsI444gi984Qvcf//9VFZWcu655zJx4sQBL0+Ss2jRoqRDGLXS23bx4sUJRyIiMjLFSZZ3Af4I3GdmdwF/Ad4GeiWn7n5LYcIT6dtJJ53ESSedlHQYIiIiMkrFSZZvJkiMDTg7/IOeybKFw0qWRURERGTEi5MsXzBkUfTBzG4DFgBTzWwtcIm735BELCIiIiIytsR5gt/PhzKQPtb7kSTWKyIiIiIS+6EkInE1NDRw7bXX8vzzz7PHHnvw+c9/nr322ivpsERERET6lfdDSdLMbLyZfdTMrjazG8LXj5qZOkiVnG644QaeeeYZ2tvbqa2t5aqrrsJdnZaIiIjI8BerZtnMPgj8GJhIcDNfmgM/MLN/cfffFjA+GWG2bdvG5s2b2WuvvSguLgbg5Zdf7lFm/fr11NXVMWnSpCRCFBEREclb3smymZ0E3AakCHq7WELQddwM4ATgPOA2M6tz94cKH6oMd3feeSe33HILnZ2dzJgxg8suu4wZM2aw//77s2HDhq5yM2bMoKamJsFIRUREhoe3336bNWvWcMABBzB+vC7SD0dxapa/BbQCx7n781nTfm5m1wGPheWULI8x27Zt60qUITj477jjDr74xS/yyU9+km3btvHSSy8xZcoUvvjFL1JUFLsFkIiIyKhy9913c+ONN+LuVFRUcMkll3DggcHDkp966imeeeYZdt11V84444xeT8KVnSdOxnIYcEeORBkAd18K/Bo4vBCByciydevWrkQ5bePGjQC0tbWxevVq3J3Nmzfzk5/8hPb29iTCFBERGRZaWlq49dZbu+7haW5u5tZbbwXgwQcf5Lvf/S4PP/wwv/zlL7niiiuSDHXMi1Oz3Aqs76fMurCcjDF77rkns2bN4q233uoaN3/+fAAeeOABGhoausavXr2apUuXcvTRR+/0OEVExqrFixdTW1ubdBij0ooVK4Dux8vno7Ozk5aWlh7jXn31VRYtWsT69T3TrZdeeol//dd/pbS0dPDBjkBz586NtW0LLU6y/Dgwv58yxxI0xZAxpqioiMsuu4w77riDDRs2cOyxx3LKKacAkEqlepVXbxgiIjtXbW0ty5YvC27Rl8IKv+aWvbUs1mzjbTyl3p0A13fWs/GtjVR1VDGOcV3jHWf5huW4jcHvzrqkA4iXLH8deMrMrgS+4+470hPMrAq4BDgYOKawIcpIMW3aND7/+c/3Gn/yySdz3333sWNHsMvstttuzJs3b2eHJyIiEyG1oHcFhiSjoaOBijcrKG4spn1qO627Bhfnm+qbKHm+hKKOoLVs857NdM7t7GtRo1bRkuTvcYqbLP8V+DfgM2b2PLABmE7QTrmGoFb562aZvcrh7v7JwoQrI9GMGTO49tpreeyxxygvL2fBggWMGzeu/xlFRERGsxJo3qu51+jOmk7q5tdRurWUzspOOqvHZqI8XMRJls/P+H8i8N4cZY4P/zI5oGR5jJs6dSpnn3120mGIiIiMCF7qtE1vSzoMIV6yvOeQRSEiIiIiMgzlnSy7+xtDGYiIiIiIyHBT8FbTZjah0MsUEREREUlC3smymd1vZlP7KXMk8MKgoxIRERERGQbi1CyfBLxoZifkmmhmXyPoDWNmIQITEREREUlanGT50wTdwz1gZt+2sH84M5tiZv8HXAW8Sf8PLhERkYS98MILXHfdddx+++1s37496XBERIatODf43WBmTwG/Bi4Gjjez64AfALsCtwP/4u4664qIDGPPPvssl19+edfw008/zTXXXENWH/kiIkLMG/zc/WVgHnAz8B7gDoLa5k+5+3lKlEVEhr8HHnigx/CqVauora1NKBoRkeFtIL1hTAL2Dv9PV0OMwYeVi4iMTFVVVXmNExGRmMmymZ1K0NvFe4CfAacA9cDPzOxXZlZd+BBFRKSQzjnnHKqru0/XJ598MrvuumuCEYmIDF95t1k2s/8EvgzsAD7i7neE4w8Bfg6cCxxhZue6+3NDEayIiAze7rvvzk9/+lNeeOEFpk6dyr777pt0SCIiw1acx11/FXgOONfdV6ZHuvsW4Awz+yrw/4AngIqCRikiIgVVWVnJMccck3QYIiLDXpxmGNcCx2Qmypnc/b8Iuo1bV4jARERERESSFqfruC/mUeYvZnbY4EISERERERkeBtIbRp/cvaHQyxQRERERSUKfybKZPWxmH8sad6SZLYoo/0UzW1XIAEVEREREktJfzfICYE7WuFOAayLKTwT2GFxIIiIiIiLDQ8GbYYiIiIiIjBZKlkVEREREIsTpZ1lkUJYtW8ZDDz1EVVUVCxcu1BPDREREZNhTsixDorW1lbVr17Lbbrsxbtw4XnrpJS699FLcHYCnn36an/zkJwlHKSL56OzspKioCDNLOhQRkZ1OybIU3AsvvMD3vvc9GhsbmTBhAt/4xjd49NFHuxJlgLq6OpYtW5ZglCLSn9bWVq677joef/xxJk6cyKc+9Snmz5+fdFgiIjtVPsnygqzahAUAZvZNILuaYUFBopIR7cc//jGNjY0ANDQ0cP3113PUUUf1Kjd58uSdHZpIQSxevJja2tqkw8jLihUrAFi0KGePn32qq6ujrq4OgK1bt/K9732P2267jeLi4oLGmG3u3LkDildEZCjklSyTOwn+dtawEyTPnqOsjHKtra28+uqrzJw5k7fffrvHtPXr13P66afz5JNPsmbNGgBOOOEE9t9//yRCFRlTKioqBjxva2trr3FtbW2DWqaIyEjTX7KcnRCL9PL666/zzW9+k4aGBoqKith9991ZvXp11/RjjjmGmpoaFi9ezCuvvEJ1dTV77KHuuGXkGiu1nnfeeSc33XRT13B5eTnXXHMN48ePTzAqkQHqhKpXqxi3aRydlZ3s2G8HnTWdSUclI0CfybK7K1mWft166600NARPOU+lUrz11lucdtpprFq1igMOOICPfOQjABQXF3PwwQcnGaqIxHDmmWeyefNmlixZwpQpU7jggguUKMuIVbmykvJ15QAU1Rcx4cUJbJu/TZ3oSr90g58M2tatW3sMt7e3s3DhQmbMmJFQRCJSCMXFxXz605/m05/+dNKhiAxa6dbSHsNFrUUUNxXTOV61y9I3/Z6SQVuwYEGP4f3331+JsoiIDCsdNR09hlOlKTorlChL/1SzLIN25plnUllZybPPPsusWbM4++yzkw5JRESkh6a5TRS1FFG6uZRURYodB+yAoe3YRUYJJctSECeeeCInnnhi0mGIiEiEtWvXQj0ULRm7F5V3sCPIfDqAl6BIF9iHvzpY62sTDUF7iYwonZ2dLF++nLfeeivpUEREZCTSgyglJtUsy4ixZcsWLrzwQtavXw/A+9//ft14JCKSp9mzZ7PJNpFakEo6lDGndEsplSsqKWotonVmK01zm1RdmaeiJUXMnjU72RgSXbtIDHfeeWdXogxwzz33dD3kREREZDiydqP6xWpKtpdQ1FZExRsVlK8pTzosiUHJsowYW7Zs6TUuu9s6ERGR4aSkoQTr7Nn2o3RbaURpGY5iN8Mws12AecAkIu4jdfdbBhmXSC8LFizgySef7BqeNm0aBx54YIIRiYiI9K1jfAdujnl3wtwxoaOPOWS4yTtZNrNS4MfAx4iukTbAASXLo9AjjzzCM888w+zZs1m4cOGgn+S1fft2Nm/eTGtrK4sXL+aCCy6guro6svyRRx7JhRdeyMMPP8zkyZM5++yzKS3Vr3MRERm+vMxpPKiRqr9XYe1G2y5tNO/RnHRYEkOcmuXvABcAK4FbgTUEna/IGHDvvfdy/fXXdw3fddddHHnkkVxwwQVMnTp1QMu89tpraWxsBOChhx6isbGRCy+8sEeZuro6br/9dt544w3mzZvHwoULOeqoowb+RkRERAbJ2ozixmI6qzvxUu+3fNvMNtpmtAXViWoAO+LESZbPA/4OHObuY+on0eLFi6mtrU06jEStW7eux3BbWxuPP/44zzzzDLvuuuuAlrl69eoew08//TSLFi3qMW79+vW0trYCsHz5cu666y4mTZo0oPWNRXPnzu21TUVEZODGbRjH+L+Nx1KGFzvbD9lO+5T2/mc01G3dCBUnWd4F+NFYS5QBamtrWfbSy6QqJycdSmKqizoYl+Mgb2tr4/mV6/EBnAFqiowS6/5FnnKoff0NUhg7UiU4xuTi1h7zbKlrYNXW1uxFSQ5FTbr5UUSk0Cpfq8RSwXeedRqVf6+k/uj6hKOSoRQnWX4TmDBUgQx3qcrJtBx4RtJhJKazZRtT1zxOUaqt5/iScpr3PR2s72S5tGUblQ1v0FlcRlPNXqRKyuhs3srkdU9T0tFEqqiUolQ7RTjgFJWXsmGPk0it+j+KUt2/2DvGT6dl9vyheIujTvnL9yYdgojI6JKCorae7SiKWtWuYrSL8wnfDJxqZjVDFIsMY+3lk3h779PYNv1wOkqrAOgsLmPbjHn9J8rNW5j2xsOM31ZLzeblTF2zBDxFe8VkNux1Km/vdSodpT1vFixp30FJxw7qdzkEt2A37SypoH7aO4bk/YmIiPSrCNqm96w0ap2hq52jXZya5SuBQ4CHzOzfgefcvWFowpLhyItKaJq4F001ezKuaRNV9asZv/XvFHe00FQzJ3K+qvrXMbqbW5S2baesaSOtVTPAjM7SKtrLahjXuq2rTMqK6SytpKmshubxu1LS3kh72UQw/YIXEZHkNB7YSGdlJyUNJbRPaqdl95akQ5IhFidZTl8LN+AhAMtdo+jursdoj2aeYvLbz1LcEZwgyps24lZM84Tdchcv6r07ZI9rmHoQpa31jGvdRqqolLrph+FFQbdwXjyO9uKx215cRESGkWJo3nvM3b41psVJah8H+u8fRUa9cc1buhLltIrtayKT5caJ+1DRsIbizuBSVUvVDNoqenY3lyqtYNOc91Hc3kTKiilta6Coo4VUiR4JKiIiIsnJO1l29wVDGIeMIJ2llTg9e8BJt2PONq5pE+O3raCtfBLtZRNpq5xGa+Uukcsu6mhml7VPUpRqw62IuumH99nEQ0RERGQoqQGoxNY5bjzbpxzQ1V1c+7gaGifv16tcSdt2pq59nIrGdVTseJvqra/RWVLR5w2BNZte6upxwzxFzcYXwVND80ZERERE+qG2xTIg26cexI6Je1Hc0Up7WU3OBLh8+zosI9E1nPLGt2gsi+6BsLijZzuwolQ7lurAi8cVLngRERGRPEUmy2b2LYI2yv/t7lvD4Xy4u3+nINHJsJYqqSBVUhE5vbO0Mse43M010lqqZjC+bmXG8HQlyiIiIpKYvmqWLyVIlu8AtobD+XBAybLQXD2L5u27UtEYPCq7uWomzdWzI8uXNa6nKiNRbi2fxNaZRw15nCIiIiJR+kqWTwhf38waFsmPFbF11jGUtG0Hdzr6aH4BUL311R43DY5rqUMdsIiIiEiSIpNld3+0r2GRfHWMq86rnKWyb+RzzF3psohIodRB0RLd219wjeHr+D5LyUDUAbOSDUE3+MmwsWPS3ox7e2nXcHP1bqRKyhKMSERk9Jg7d27SIYxaK1asAGCfWfskHMkoNCv5fTd2smxm7wTOAw4Aqtz9xHD8HODdwIPuvi1yASIRmmrm0FFaSXnjejrKamiasHvugu6UNW0ECPps7qMrOhERCSxatCjpEEat9LZdvHhxwpHIUIiVLJvZZcCFdPfPnHmFvAi4DfgScG1BopMxp61yF9r6eGiJpTqYuuZRxrUEv8fayiexabcFUFS8kyIUERGRsSTvhktmdi5wMfAgcCjw3czp7r4KWAqcWcgARTJVbF/TlSgDjGvZRuX2NQlGJCIiIqNZnFb+i4Ba4Cx3/yvQlqPMK4Aa7MiQKersvdvlGiciIiJSCHGS5XcA97t7X5nJOmD64EISidZcPZuUdbceShWV9Nl3s4iIiMhgxGmzbEB2317ZpgMtAw9HpG+dpVVs2uMEqupWAbBj4t50llZS3riOCZteorijhaaaOdRPe6du/BMREZFBi5MsrwCOiZpoZsXAfGD5YIOSkau8cR2V9atJFY2jcfK+/T6IZCA6ymqon35Y13BRRwuT1z2NefBbbvy2FXSUVrFjkrpJEhERkcGJ0wzj18DhZvbViOn/AcwFfjXoqGTkcGfcjo1U1L9B+fa1THnrz1Q0rqOqYTVT1zyKpdqHPIRxLVu7EuWucc2bh3y9IiIiMvrFqVn+AfBB4Htm9iHCbuPM7PvAccA84Gng+kIHKcPXpPXPULl9LQBOz2YPxZ2tlO3YSEv10D56p618Eo5hGT0ZtpdPHtJ1ioiIyNiQd82yuzcDJwC/AA4neACJAV8B3gX8EjjF3TuGIE4Zhkpa67sSZaBHsprWWVpRmJV59EOvUyUVbJt5BJ0l5bgVsaNmDo1qgiEiIiIFEOuhJO5eD5xvZl8BjgCmAPXAs+6+aQjiGxbWrl1LUVM95S/fm3Qow0oJKch6FkjKoSisYG5JFVO86s/ZRWIZRyeVRe0UAa1ezA4vAXrfuOcEj4+HUtj6NuVb/zCItY4ORU1bWLtWv11FREQGI/bjrgHcfStwf4FjkRGmA6PDjRILan3dYXuqBCgiBaRiNYnvzXDGF7V3dWpRbp10powW72u3VQ8YIiIiUjh5J8tm1glc6u7f6aPMRcC33fvMZmIxs1OAHxLUYf7M3a8s1LLzNXv2bDa0ltBy4Bk7e9XDXmtnO1X1qyjuaKapejfaK6ZElq1oWMOEzcuxVDs7Ju7F9ikHYql2KsIn8DVX74YXj+sqX7bjbWztEz2WUTRhGi2zjs25fOtsB7zHMsay8pfvZfbsGUmHISIiMqLF7Wc5n2q7glXthd3R/TdwErAW+IuZ/d7dXy7UOmRwvLiUxsn79VuuuG0Hk9Y/29WuecKWV+gsqaB6y6uUdDQBUL3lNTbOObEr2W0vn0TKiinyzq7ltFVMzRGEU7PxRarqVgKwo2bPoGs59bMsIiIigzS46+S9TaKwDyV5N1Dr7qvCJwfeDpxVwOXLTjKueXOvGwAr69/oSpQBSjqaqGx4s2s4VVzG1l2Pon1cNamiEhon7k3jpN5PUy/b8Tbj62oxPGi6Ub+K8sZ1Q/dmREREZMzos2bZzN6TNWpOjnEQNJHYHfgn4LUCxQYwC1iTMbwWODK7kJl9BvgMwO67717A1UuhtFdMxul52aGztBJatvQ5X+v4mWwcP7PPMqWt9TnHDXWXdSIiIjL69dcMYwl0VQc68PHwL5f047CjHloyELmuo/fqQ8zdryfs33nevHnRfYxJYjrGVVM3/XAmbP4bRakOdtTsyfbJB1DWvJnijmYAOksqaJ6wW++Z3amsX01Z0ybayiexY9LeYN0XRVqrZuCb/9a1szjQUqW2uiIiIjJ4/SXLl0FXheC3CJLnR3OU6wS2AI+4+6sFjG8tkJk9zQZ0fX2Eapq4F001ewYDYXvijXucSMX2oOlFc/XupIrLes1XveVlJmx5BYDK7W8yrrWObTOP6JreXj6RrbseRfXW18Cdxsn70l6hh5KIiIjI4PWZLLv7pen/zezjwF3uvniog8rwF2AfM9sTeAs4FzhvJ65fCi3rprtUSRk7crRDzlRVv7rHcEXDm2ybfjgUdffg3FI9m5bq2QULU0RERARi9Ibh7nsOZSAR6+wws88T9OlcDNzo7st3dhySrFTxuK6mGsFwaY9mGCIiIiJDZUD9IZvZfOAwYCLBE/yed/cn+p5rYNz9D4AexzaG1U89mCnrnsI8hQMNUw9Wt3AiIiKyU8RKls3scOCXQLpjXSO84c7MXgM+5u5LCxqhjGlFna2kisfx9p6nMK5lG+1lE+kcV5V0WCIiIjJGxHmC31zgYWAC8ET4/3pgJnACcBzwoJm9291XDEGsMsZU1q9m4obnMU/RWVzO5tnzlSiLiIjIThWnZvmbwHjgw+7+m6xpl5GakFMAACAASURBVJrZBwgeGnIx0d3LieQn1UnNxhcxTwFQ3NlCzeaX2DL7uIQDExERkbEkzl1SJxL0hpGdKAPg7r8F7g7LiQxKUaqdolR7j3HF7U0RpUVERESGRpxkeSrQXx/Kr4blZCzwFKQ6h2TRqZJyWit67krNYddw1tnGhI1/ZcraJ6jaVguu59CIiIjI0IjTDGMTcGA/ZfYHNg88HBkpqrbVMmHzcizVQdOE3amb8a6Cd+e2ddejqd7yCiVt9bRWzaBx0r4ATFn3FGVNmwAo3/E2Ral2tk85oKDrFhEREYF4NcsPA2ea2bm5JprZOcBZwEOFCEyGr5K27dRsfIGiVDuGU9XwBlV1qwq+nlRJGfXTD2XLbsfTOHk/MKOoo6UrUU6rzHpoiYiIiEihxKlZvowgGb7V/v/27jzMrqpM1Pj7FalAQgLEBIIhJggRMCqJGPQySTFLtJ1oG4e+NyB9W1raXKQHFSdo9V4aHNo44wSi3e1VBAeITGbAIK1BEEIIowExAknIACQhqaqv/9i7ypOTc5Kq1HAqVe/vec6T2muvvfZ3Ts6q89U6a68dcR4wj2I1jP2BFuBY4Bngk70cowaY5k1rqV7luHnT2n45d3tTM0kTQXtnWVPrxmIqhmsvS5KkXtadO/g9FBEnA98BjikfCZ150/3ALJeNG/yeHzGOJAj+PFf4+ZH79s/Jm3ajbdjuDKu4o19TtrPbludoGz6qf2KQJElDRrduSpKZvwFeGhFHA0cAe1Pcwe/OzFzUB/FpAGpvHsHTBxxVzFlu28yGfQ5i496T++38m0fuy7D1j/05nqbhtA0b0W/nlyRJQ0d3bkryWmB9Zt6VmbcBt/VdWBroNo2awKZRExpy7nX7voLdNj/L7puepm23PVi7/xHQtFtDYpEkDV2PP/44y5YtY/PmzQwfPrzR4aiPdGdkeR7wNeC9fRSL1CXtw0awavKJxa2wm5p7fRUOSZJ25JZbbmHOnDlkuXzp2LFjO/e1trZy7733MmrUKA4++OBGhahe0p1keRWwcYe1pH7SvtvujQ5BkjREfe973+tMlAHWri0udF+zZg0f+tCHWLFiBQDHH388//AP/9CQGNU7upMszweO7qM4JEmSAJgzZw4PPfRQo8PYrqeffnqr7dbWVmbPns2aNWtYt25dZ/mCBQt44IEH2H33gTXAM2XKFGbPnt3oMHYJ3fn++iPAoRHxiYho7quAJEmSBrrRo0dvtd3UVKRUbW3b3tm2Vpl2Hd0ZWf4QsAS4EDgnIn4HPAFU32s4M/OcXopPkiQNMbvCiGdmsmDBAu677z4OO+wwWlpaiAiWLFnCRz7yEdrbi/sBvOAFL+BLX/rSgBtZVtd1J1k+q+Ln/ctHLQmYLEuSpEErImhpaaGlpWWr8pe//OVcfPHF3HzzzYwePZo3velNJsq7uO4kyy/usygkSZIGiWnTpjFt2rRGh6Fe0p07+D3al4Fo8Bn2/HqaN61h84ix3l1PkiTtkrqULEfEJOBIiikWv8nMP/RpVNrl7bnmIfZ56i4AkuDpCa9h0+iJDY5KkiSpe3aYLEfEp4HzgSiLMiI+l5n/1KeRadeVyV6r7u3cDIrt7iTLw597in2e+h27tW5g4+gXsXa/ad6lT5Ik9bvtLh0XEe8ELqBIlJcB95c/XxAR7+j78LRrSiK3XianqX1Ll4+O9lbGrvgVzZvX0dS+hT3XPcLop+/v7SAlSZJ2aEfrLJ8DtAInZ+bLMnMqcBrQjiteDD3tbYxY/xh7rnmIptZN9etFE8/tdeBWRc/tfVCXT9O8ae02yfUez/6xO5FKkiT1ih1NwzgcuDYz53UUZObNEfFjoKUvA9MAk8m4Pyxk902rAdhr1b2snHwircNH16y+bvwr2bLHGIZvWsPzI/dl414v6vKptuy+N+2xG00Vo9PDn1/Hnmse4rkxU3r2PCRJkrphRyPLYyimXlRbBuzT++FooNp9w1OdiTIU0yr2XPtw/QMi2LDPi1m7/xHdSpQBcrdm1rzwyG3udjNqzYPdakeSJKmndjSy3ATUmmy6hT9f8DckNG14mj2W/qzRYTRMM21QdX3dsNW/Z49Vj/eo3di0HoDcY6+qPVm8+yrfZZs3DOn/g+5q2vA09e8dJEmSuqIrS8dVD/ANOVOm+NV/ZrJixQq2bCn+dooIDpo4nuHDh/eo3QcffAaAlxy8bVK3du1a1q5d27m9/75jmTJq2/Wa29ra2LhxI8OGDWOPPfboUTyDy/6+dyVJ6qHIrJ8LR0Q73U+WMzO7c2fAXjVjxoxcvHhxo04/qG3YsIH58+fzzDPPcNxxxzFhwoQetzl79mwA5syZU3P/0qVLefjhh3nFK17BgQceuM3+hx9+mA9/+MNs2LABgJkzZ3Luuef2OC5JkjS4RcQdmTljR/W6ktR2d7rFkJqeMZSMHDmSmTNn9us5p06dytSpU+vu/8EPftCZKAPMnTuXM844g3333bc/wpMkSYPcdpPlzNzRBYBSQ1UmylBMF9m4cWODopEkSYONybB2aaeddtpW21OnTmXSpEkNikaSJA02DZtbLPWGY445hk984hMsWrSI8ePHc/rppzc6JEmSNIiYLKvXrF69mi9/+cssXbqUQw45hPPOO4/99tuvz887bdo0pk2b1ufnkSRJQ4/TMNRrvvjFL/Kb3/yG5557jjvvvJPPfe5zjQ5JkiSpRxxZVo898cQTNDU1sWTJkq3K7733XjKTCBdIkSRJuyaTZe201tZWLr30Um6//XYA9t57b55//vnO/VOmTDFRliRJuzSnYWin3XrrrZ2JMsC6des6b1Ry0EEHcf755zcqNEmSpF7hyLJ22p/+9Kdtyk4//XRmzpxJc3NzAyKSJEnqXY4sa6e95jWvoanpz2+hYcOGceSRR5ooS5KkQcORZe20gw8+mI9+9KNce+21rF+/nunTp7PPPvs0OixJkqRe48iyemTq1KmsXr2a3//+91xzzTXMnj2b9evXNzosSZKkXmGyrB657bbbePzxxzu3n3rqKebNm9fAiCRJknqPybJ6ZMuWLduUtba2NiASSZKk3meyrB455phjGDt2bOf26NGjaWlpaVxAkiRJvcgL/NQjo0eP5rOf/Sy/+MUvaGtr48QTT9wqeZYkSdqVmSyrx8aMGcMZZ5zR6DAkSZJ6ndMwJEmSpDpMliVJkqQ6nIahhliyZAmPPvooW7Zs8Y5/kiRpwDJZVr/79re/zTXXXNO5vd9++3X52Pvvv5/Vq1czffp0Ro4c2RfhSZIkdTJZVr967rnn+OlPf7pV2bp167p07Jw5c7j55psB2Hvvvbnkkks44IADej1GSZKkDs5ZVr9ZsWIFt912G21tbVuVZ+YOj3388cc7E2UoEuwf/ehHvR6jJElSJUeW1S+uv/56vva1r5GZRMRW+/baa68dHv/ss892qUySJKk3ObKsPtfW1sZ3v/vdzhHkzGTcuHG89a1vZf/992fUqFE7bOOQQw5h8uTJW5VNnz69T+KVJEnqYLKsPtfW1sbGjRu3KT/rrLPYY489utRGU1MTn/rUpxg/fnxn2de//nWWLl3aa3FKkiRVM1lWnxs+fDjHH3/8VmWnnnpqt9tZuXIlTz75ZOd2a2sr1113XY/jkyRJqsc5y+oX5513HlOmTOGhhx7i8MMP54QTTuh2G01N2/5tVz3/WZIkqTeZLKtfNDc384Y3vKFHbbz4xS9mxowZLF68uLPN17/+9b0RniRJUk1Ow9Au5YILLuict7xlyxa++c1vsnnz5gZHJUmSBiuTZe1SFi1atNW85QceeIBFixaxdu1aLr74Yt785jcze/ZsHnzwwQZGKUmSBguTZe1Saq2tvH79er75zW9yxx130N7ezvLly7n00ktpb29vQISSJGkwMVnWLuXYY49lxIgRndt77rknxx57LMuWLduq3pNPPsnatWv7OzxJkjTIeIGfdinjx4/nsssu4/rrr6epqYmZM2cyduxYpk6dutX0jBe+8IWMGTOmgZFKkqTBwGRZu5xJkyZx7rnnblV2zjnnsHHjRn77298yefJkzjvvPJeVkyRJPWayrEFhr7324sILL2x0GJIkaZBxzrIkSZJUh8myJEmSVIfJsiRJklSHybIkSZJUh8myJEmSVIfJsiRJklSHybIkSZJUh8myJEmSVIfJsiRJklTHgE2WI+JtEXFvRLRHxIxGxyNJkqShZ8Amy8AS4K3AwkYHIkmSpKFpWKMDqCcz7wOIiEaHIkmSpCFqII8sd1lE/G1ELI6IxStXrmx0OJIkSRokGjqyHBE3A/vX2PXhzPxxV9vJzMuBywFmzJiRvRSeJEmShriGJsuZeXIjzy9JkiRtz6CYhiFJkiT1hQGbLEfEWyLiceAo4LqIuKHRMUmSJGloGcirYVwDXNPoOCRJkjR0DdiRZUmSJKnRTJYlSZKkOgbsNAypli1btnD99dfzyCOPMG3aNE488cRGhyRJkgYxk2XtUj7/+c+zcGFxB/R58+axcuVKzjzzzAZHJUmSBqvIHFz38JgxY0YuXry40WE0zJw5c3jooYcaHUaXPfjggwC85CUv2WHd9vZ2Hnvssa3Khg0bxsSJE/sktlqmTJnC7Nmz++18kiSpb0TEHZk5Y0f1HFlWQ40YMaLLdSOCiKDyD7ymJqfdS5KkvmOyPMgM9lHP6667jssvv5zMpLm5mQsvvJAjjjii0WFJkqRBymRZu5TXv/71vPKVr2T58uW89KUvZcyYMY0OSZIkDWImy9rlTJgwgQkTJjQ6DEmSNAQ44VOSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWZIkSarDZFmSJEmqw2RZkiRJqsNkWQ21atUq3ve+97F69epGhyJJkrQNk2X1mUcffZRly5bR3t5et86VV17J3XffzZVXXtmPkUmSJHXNsEYHoMEnM/nMZz7DwoULATjooIP45Cc/yahRo7aqt2rVKubOnUtmMnfuXGbNmsXYsWMbEbIkSVJNjiyr1y1ZsqQzUQZ45JFHuOGGG7apd+WVV5KZALS3tzu6LEmSBhyTZfW6WvOPa5XddNNNbNmyBYAtW7Zw44039nlskiRJ3WGyrF43Y8aMraZcNDU18drXvnabeqeccgrNzc0ANDc3c+qpp/ZbjJIkSV3hnGX1ulGjRnHJJZdw7bXXsmHDBl73utdx2GGHbVNv1qxZzJ07FygS6lmzZvV3qJIkSdtlsqw+MWnSJGbPnr3dOuPGjeOEE07ghhtu4IQTTvDiPkmSNOCYLGtAykwWLFjAfffdx2GHHUZLSwsR0eiwJEnSEGOyrIZZtWoV8+bNA2DevHm85z3v6Rxd/s53vsPVV18NwNy5c1m+fDlnn312w2KVJElDkxf4qWG2t3Tcz3/+863qVm9LkiT1B5NlNcz2lo4bMWLEVnWrtyVJkvqDybIaZntLx73rXe/qnKMcEbzzne9sSIySJGloc86yGmZ7S8eddNJJHHroodx///0ceuihTJw4sVFhSpKkIcyRZTXMuHHjOP3004kITj/99G2Wjps4cSInnXSSibIkSWoYR5bVULNmzWL58uXekESSJA1IJstqqHHjxvGFL3yh0WFIkiTV5DQMSZIkqQ6TZUmSJKkOk2VJkiSpDpNlSZIkqQ6TZUmSJKkOk2VJkiSpDpNlSZIkqQ6TZUmSJKkOk2VJkiSpDpNlSZIkqQ6TZUmSJKkOk2VJkiSpDpNlSZIkqY4BmyxHxGURsSwi7o6IayJin0bHJEmSpKFlwCbLwE3AyzPzcOAB4EMNjkeSJElDzIBNljPzxsxsLTdvByY2Mh5JkiQNPQM2Wa7ybmBuvZ0R8bcRsTgiFq9cubIfw5IkSdJgFpnZuJNH3AzsX2PXhzPzx2WdDwMzgLdmF4KNiJXAo70aqPraOGBVo4OQBjH7mNT37Ge7nsmZue+OKjU0Wd6RiJgFnAuclJkbGh2P+kZELM7MGY2OQxqs7GNS37OfDV7DGh1APRHxOuADwPEmypIkSWqEgTxn+YvAaOCmiLgrIr7a6IAkSZI0tAzYkeXMnNLoGNRvLm90ANIgZx+T+p79bJAa0HOWJUmSpEYayNMwJEmSpIYyWZYkSZLqMFlWl0XE5yLi/IrtGyLiGxXbn4mICyPih+V2S0T8rE5byyNiXETsExHv7fvopf7RxX5yQURMqOgr0yNiZkWdiyLiH+u0f1s347kiIv6yquzZHRxzYES8szvn6WIs50fEyN5uV0NHREyMiB9HxIMR8XBEfD4ihnfhuAu7UOcbETG1RvlZEfHFbsS4zWdfrX5Y47he7x9lLEf3ZptDkcmyuuM24GiAiGiiWID9ZRX7jwZuyczt/kKosg9gsqzBpCv9ZFFmrqjoK9OBmXRBZvbHB9+BQK8ny8D5gMmydkpEBPAj4NrMfAlwCDAK+FQXDt9hspyZf5OZS3sWZY/0Rf9oofx9pJ1nsqzuWMSfO93LgCXAMxExJiJ2B14KrImIJdUHRsTYiLgxIu6MiK8BUe66BDi4XB7wsoi4KiLeVHHc9yLijeVf9j+OiJ9HxP0R8fGKOn8dEb8u2/haROzWR89f6oqu9JM7y9HbJeWo2L8AZ5bv4TPLY6dGxPyIeCQiZnc03jEqXI4YzY+IH0bEsrKvBN0QhcvKOO6pOPclwHFlPO+PiJdV9LG7I+IlZfzLIuLKsuyHHaNiEXFS2dfviYhvRcTu5XOYAMyLiHk79cpqqDsR2JSZ3wbIzDbg/cC7I2Jk9QhwRPys7CeXACPK9+/3ImLPiLguIn5XvvfPLOvPj4gZ5c9nR8QDEbEAOKaizX0j4uqI+E35OIZu6kr/iIjdytHojr75/ooY/y0ibiv3vbosf0FEXFv2xdsj4vCIOJDixm7vL5/7cTvxmguTZXVDZq4AWiNiEkUy8Cvgv4CjKG5Jfjewuc7hHwd+mZmvBH4CTCrLPwg8nJnTM/OfgG8AZwNExN7lea4v674aeBfFKNzbImJGRLwUOBM4JjOnA21lHakhutJPMnNzRf3NwMeA75f94PvlrsOA0yje9x+PiOYap3slxWjUVOAgKj7Uq1xWfljeFRF3VZS/laI/TQNOLuu9kKJf3lrG8zmKD9zPl31sBvB4efyhwOWZeTiwHnhvROwBXAGcmZmvoFii9O8ycw6wAjghM0/Ywcso1fIy4I7KgsxcDzwG1F1uNjM/CGws38/vAl4HrMjMaZn5cuDnlfXLPnAxRX86haJ/dfg88LnMPBI4g+Izq5bjqvrcG8u2u9o/pgMHZObLy3rfrmh7z/IbpvcC3yrLLgbuLPvihcB3MnM58NUy3umZeWu910jbZ7Ks7uoYNetIAn5Vsb29uZSvBb4LkJnXAWtqVcrMBcCUiNgPeAdwdWa2lrtvyszVmbmR4qu4Y4GTgFcBvyl/IZ1EkTRIjbSz/aTSdZn5fGauAp4Cxteo8+vMfDwz24G7KKZP1PJP5Yfl9DLh7XAs8B+Z2ZaZTwILgCNrHP8r4MKI+AAwueyDAH/IzEXlz98t2zsU+H1mPlCWX0nR/6WeCqDWerf1yuu5Bzg5Iv41Io7LzHVV+18DzM/MleUfs9+v2Hcy8MXy8+YnwF4RMbrGOW6t6nM/Kcu72j8eAQ6KiC9EcUfj9RX7/gMgMxeW59+Hou9dVZb/AhhbDjipF5gsq7s65mO+guLr5dspRsyOpkgQtqerv8yuohgdPput/5quPj4pfkleWfFL6dDMvKiL55H6Sk/6SYfnK35uo/ZNpLpSZ3u6NG0jM/+dYmRsI3BDRJzYsau6alfblHbCvRTfbHSKiL2AFwEPA61sndfsUauRMlF9FUXS/P8i4mO1qtWJoQk4quIz54DMfKYbz6GrfW4NxTc+84Hz2HoEu6v9zhtp9BKTZXXXIuANwNPlaNTTFBfpHUUx+lTPQsrpERFxOjCmLH+G4rbmla6g+GqZzLy3ovyUcl7WCODNZSy3AH9ZjkR3zNuavPNPT+oV3e0ntfpBf1hIMVd6t4jYl2KE69fV8UTEQcAj5VfFPwEOL3dNioijyp/fAfwSWAYcGBEdX4v/T4oRa6rblbrpFmBkRPwvgCiuT/kMcEVmbgCWA9MjoikiXkQxhanDlo6pTBExAdiQmd8FPg0cUXWe/wJaorjWphl4W8W+G4G/79iIiOl0T5f6R0SMA5oy82rgo1UxdsyxPhZYV46MV37GtgCryikq9rleYLKs7rqH4ur+26vK1pVfF9dzMfDaiPgtcCrFHDMyczWwqLxQ4bKy7EngPrYeVYbig/gqiq+br87MxeWVyx8BboyIu4GbgBf28DlKPdXdfjKP4oK+ygv8+sM1FNca/A74BfDPmflEWdZaXgD1fooP5yXlV8+HAd8pj78PmFX2vRcAX8nMTRTfCv0gIu4B2inmTUJxO+C5XuCnnZHFLYffQnHNyoPAA8Am/rzSxSLg9xR97dPAbysOvxy4OyK+R/GNz6/L9/OHgU9WnedPwEUUf9jeXNXObGBGeSHdUor5/N15Dl3tHwcA88sYrwA+VNHMmiiWkPwqcE5ZdlFHXBQX6M4qy38KvMUL/HrG211rwIniivp7gCM65pJFxFnAjMz8++0dK6l/lFfa/6y8QEpSP4iI+cA/ZubiRscylDiyrAElIk6m+JrqCzUuupAkSepXjixLkiRJdTiyLEmSJNVhsixJkiTVYbIsSZIk1WGyLEn9JCKWR8TyRscxFETEWRGR5Uo6krTTTJYlNVyZ1GREtEfEwdupN6+i7lm9dN75PW2nor35EbFLXTVdJvBZLgW3y4iIljLuixodi6TBzWRZ0kDRSnHL1nNq7YyIlwDHl/V2VSeVD/W9a4CXlv9K0k4zWZY0UDwJLAbOjohhNfb/DUUy/bN+jaoXZebDmflwo+MYCjJzXWYuc712ST1lsixpIPk6sD/whsrCiGimuH3rbcC9tQ7c3nzgiLio/Mq+pdw+q2K6xPEVUzu2+lq/rHd1RDwSERsjYn1ELIqIv65q/8CyvePL7cr25u8oxojYPSI+WN5Cd0N5nlsj4q9q1D2wbPeK8uf/jIhVEbEpIhZHxBuqj+lNEfGq8jV5KiKej4hHI+LLEVHzNvMRMTIiPlDG9kxEPBsR90XEnIgYX1HvkIi4pKy3sqLtyyNiYlWbV1DcIhzg41Wvd0tZp+6c5e48h/J1zvK1fk9E3FO+1k+Wse290y+mpF1CrdEbSWqU/wA+SzGKfG1F+RuB8cAHgSm9cJ67gIuBjwOPAldU7Jtf8fNXgKXAQuBPwFhgJnBVRByamR8t660t2zsLmFz+3GH59gKJiOHADRSJ9jLgS8BI4C+B70fE9My8sMahk4FfA48AVwEvAM4EfhwRJ2fmvBrH9EiZiF9NMcL/Q4rX7lXA3wFviohjMnN5Rf0xFEntNOB+4FvAZuBg4N3Ajyi+UQB4K3BuWf+2st7LKN4LfxERMzLzj2XdjvfGLGABW/+fLWc7uvscKlwKnAb8FLgROAH43xTvxxO3d05Ju7jM9OHDh4+GPoAEHi9//gbFvOSJFft/DqyjSCI/WdY/q6qN5cDyOu1fVB7TUuO887cT18E1yoYDtwBbgAOq9s0vfq3WbW+bGIEPlXFcDwyrKN+vrJ/A0RXlB5ZlCXy8qq3TOtrqxmvfcY4Dd1BvFLAKaAOOq9r3gbKNG6vK/70s/wrQVLVvNLB3xfYBwO41zntqec6vVJW3lG1fVCfes6rfJzv5HK4oyx8DJlWUD6P4IyqBVze6D/nw4aPvHk7DkDTQfB3YjWLkkYiYDJwCfC8zN/RnIFljfnFmbqYY/R1G71ys926KhOuCzOy8eDEznwI+UW7+TY3jHqX4w6EythsokrpX90Jc1d5EMbL+/cy8tWrfZyiS7lMiYhJAROxHMdL9J+AfM7O9KtZnsmI+cWb+MTOfrz5pZt5IMfXmtP5+DlX+JTMfq4irFfh2udkXr7ekAcJkWdKAkpn/BdwDvDsimigSxSaKJLpfRcSkiPhSRCwr5xJnOTf56rLKAT1sfzTF1/grMnNZjSq/KP99ZY19d2VmW43yPwBjehJXHUdUxdSpTBwXlpsdsR5J8f+2MDOf21HjUfjriLi5nLPcWvF6v4Ievtal7j6HSotrlP2h/LcvXm9JA4RzliUNRF8H5gCvA84G7sjMO/szgIg4iGJO8BjgVop5qusovsI/kGK+7O49PE3HxWF/qrO/o3yfGvvW1jmmlb4ZCOlurB3//rFG3Vo+C5xftnNDedzGct9ZFHO0e6q3X++ObwJ260lQkgY2k2VJA9FVwL8CX6MYUfyXLhzTTjGfuJZayc+OXEDxlf3ZmXlF5Y6IeAdFstxTHdMQ9q+z/4VV9Rqpu7F2JJc7HBEup2zMBpZQzM9+pmr/O7oXal270ustaYBwGoakAScz11KsVDAReI5ilYwdWQOMj2KZuWoz6hzTTv1RwY5VN66use/4Ose0AUREl0Yay6TwYeCAKG66Uu2E8t/fdqW9PtYxst9SvSOKdbGPLTc7Yv01xev72ojYcwdtH0TxeXRjjUR5Yrm/WscUlO6M6nb3OUiSybKkAesjwFuA06oTqDp+TfFt2dmVheU6u8fUOWY18KI6+5aX/7ZUtXcatS+462gPoNYFYvV8i2IZs8sqk+yIGAd8tKJOo10LPA28IyL+R9W+8ykS2ps7LoLLzJXACMPrGQAAAhRJREFUf1KM1n66nH/eKSJGVaxRvLz899iq12AUxZScWt+C7sxr3a3nIEngNAxJA1SZsHQnafkCRaL8lYg4ieLiq2nA0RR3/at1s45bgLdHxE+BOyjmoC7MzIXAl8v2fhARV1PMoX05xTzq/0+x0kOt9t4G/CgirqeYc/toZl61nbg/DZxOsVLD78rjRpbt7Adcmpm/7PKrsPM+HRHP1tn3scx8LCLeDfwAWBARP6D4/3kVxfJuTwDvqTru7yles3OBloi4gWL95BdTrG7xRoql+56IiP8E3g7cFRE3UswvPgXYRLEu9vSqtu+n+D95e0RsLmNJ4KrMfLTWk8jMZ3fiOUga4kyWJQ0Kmbk0Ik4G/i/wFxSJ763AURQ3vKiVLP8figTrJIqbjTRR3FBkYWbeHREnUCzPNpPi9+XvyrbWUjtZ/gbFhWhvB/65PGYBxRzsenFvjohTKOZIvxN4Xxn774DzM7MrU1B6wxnb2fdvwGOZ+eOIOAa4kCLZ3Zsiwfwq8InMXFF5UGauiYijKUZtzwT+lmL6xB8oRsuXVlQ/h+IGK2cC5wErgZ8AH6PGVJjMbIuItwCXAH9FsW5zAL+kWFavpu4+B0mKzNxxLUmSJGkIcs6yJEmSVIfJsiRJklSHybIkSZJUh8myJEmSVIfJsiRJklSHybIkSZJUh8myJEmSVIfJsiRJklSHybIkSZJUx38DRzle/1xrCYYAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 842.4x595.44 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'AURKA'\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## BORA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Note: BORA not found in proteomics data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Patient_Type'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3077\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3078\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3079\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Patient_Type'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-f5e6a6bb3760>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# 1. Create a dataframe combining phosphosite data with TP53 mutation status; select for tumor samples\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mp53_mutations_protein\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCPTAC\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompare_mutations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mphos\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprotein\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mp53_mutations_protein\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mp53_mutations_protein\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mp53_mutations_protein\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Patient_Type'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'Tumor'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Patient_Type'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;31m# 2. Make a new column for mutation location relative to hotspot\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2686\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2687\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2688\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2690\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2693\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2694\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2695\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2696\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2697\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionality\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   2487\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2488\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2489\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2490\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2491\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/internals.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   4113\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4114\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4115\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4116\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4117\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/envs/ptmresearch/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3078\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3079\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3080\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3081\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3082\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtolerance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtolerance\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Patient_Type'"
     ]
    }
   ],
   "source": [
    "protein = 'BORA'\n",
    "pattern = re.compile(protein)\n",
    "\n",
    "# 1. Create a dataframe combining phosphosite data with TP53 mutation status; select for tumor samples\n",
    "p53_mutations_protein = CPTAC.compare_mutations(phos, protein, gene)\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].drop('Sample_Status', axis=1)\n",
    "\n",
    "# 2. Make a new column for mutation location relative to hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "\n",
    "# 3. Begin constructing a dataframe for plotting, starting only with the mutation location\n",
    "plotdf = p53_mutations_protein[['Within_Hotspot']]\n",
    "\n",
    "# 4. Set a cutoff for significance\n",
    "pcutoff = 0.05/len(p53_mutations_protein.columns)\n",
    "\n",
    "# 5. Go through each site on the protein and check for significance with TP53 mutation\n",
    "for site in p53_mutations_protein.columns:\n",
    "    added_to_plot = False\n",
    "    if (site != 'Within_Hotspot' and site != 'Mutation'):\n",
    "        # Set up ttests for each of the hotspot/non-hotspot/wt comparisons\n",
    "        sitedf = p53_mutations_protein[[site, 'Within_Hotspot']].dropna(axis=0)\n",
    "        all_mut = sitedf.loc[sitedf['Within_Hotspot'] != 'Wildtype']\n",
    "        hotspot_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Within Hotspot']\n",
    "        other_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Outside Hotspot']\n",
    "        wt = sitedf.loc[sitedf['Within_Hotspot'] == 'Wildtype']\n",
    "        \n",
    "        # Check for a lack of data at the site\n",
    "        if all_mut.empty:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "        if wt.empty:\n",
    "            print(\"Not enough wildtype values for site \" + site)\n",
    "            continue\n",
    "        if len(all_mut[site]) < 2 or len(hotspot_mut[site]) < 2 or len(other_mut[site]) < 2:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "            \n",
    "        # Perform the ttests\n",
    "        all_mut_ttest = scipy.stats.ttest_ind(all_mut[site], wt[site])\n",
    "        htspt_ttest = scipy.stats.ttest_ind(hotspot_mut[site], wt[site])\n",
    "        other_mut_ttest = scipy.stats.ttest_ind(other_mut[site], wt[site])\n",
    "        if all_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" all mutations vs wildtype: \" + str(all_mut_ttest[1]))\n",
    "            plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "            added_to_plot = True\n",
    "        else: print(site + \" all mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if htspt_ttest[1] < pcutoff:\n",
    "            print(site + \" hotspot mutations vs wildtype: \" + str(htspt_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "                added_to_plot = True\n",
    "        else: print(site + \" hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if other_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" non-hotspot mutations vs wildtype: \" + str(other_mut_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "        else: print(site + \" non-hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "    print('\\n')\n",
    "    \n",
    "plotdf = pd.melt(plotdf, id_vars = ['Within_Hotspot'], var_name = 'Site', value_name = 'Phos')\n",
    "plotdf = plotdf.dropna(axis = 0)\n",
    "            \n",
    "a4_dims = (23.4, 16.54)\n",
    "fig, ax = plt.subplots(figsize=a4_dims)\n",
    "\n",
    "#Create the plot\n",
    "boxplt = sns.boxplot(x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], data=plotdf)\n",
    "boxplt = sns.stripplot(data=plotdf, x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], dodge=True, jitter=True, color='.3')\n",
    "\n",
    "#Add styling\n",
    "boxplt.set_xlabel('Phosphorylation site on ' + protein, fontsize = '30')\n",
    "boxplt.set_ylabel('Phosphorylation', fontsize='40')\n",
    "boxplt.tick_params(labelsize='20')\n",
    "\n",
    "#Adjust legend\n",
    "handles, labels = ax.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], title='Mutation Within Hotspot', fontsize='20', frameon=False)\n",
    "boxplt.get_legend().get_title().set_fontsize('20')\n",
    "boxplt.get_legend().set_bbox_to_anchor((1, 1, 0, 0))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### We were interested in what effect the type of mutation (missense vs truncation) had on BORA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "protein = 'BORA'\n",
    "pattern = re.compile(protein)\n",
    "\n",
    "# 1. Create a dataframe combining phosphosite data with TP53 mutation status; select for tumor samples\n",
    "p53_mutations_protein = CPTAC.compare_mutations(phos, protein, gene)\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].drop('Sample_Status', axis=1)\n",
    "\n",
    "# 2. Make a new column for mutation location relative to hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "\n",
    "# 3. Classify each mutation as missense or truncation\n",
    "p53_mutations_protein = p53_mutations_protein.replace({'Mutation': {'Frame_Shift_Del': 'Truncating',\n",
    "                                      'Frame_Shift_Ins': 'Truncating',\n",
    "                                      'Nonsense_Mutation': 'Truncating',\n",
    "                                      'Missense_Mutation': 'Missense'}})\n",
    "\n",
    "# 4. Begin constructing a dataframe for plotting, starting only with the mutation location\n",
    "plotdf = p53_mutations_protein[['Mutation']]\n",
    "\n",
    "# 5. Set a cutoff for significance\n",
    "pcutoff = 0.05/len(p53_mutations_protein.columns)\n",
    "\n",
    "# 6. Go through each site on the protein and check for significance with TP53 mutation\n",
    "for site in p53_mutations_protein.columns:\n",
    "    added_to_plot = False\n",
    "    if (site != 'Within_Hotspot' and site != 'Mutation'):\n",
    "        # Set up ttests for each of the wt/missense/truncating comparisons\n",
    "        sitedf = p53_mutations_protein[[site, 'Mutation']].dropna(axis=0)\n",
    "        all_mut = sitedf.loc[sitedf['Mutation'] != 'Wildtype']\n",
    "        missense_mut = sitedf.loc[sitedf['Mutation'] == 'Missense']\n",
    "        truncating_mut = sitedf.loc[sitedf['Mutation'] == 'Truncating']\n",
    "        wt = sitedf.loc[sitedf['Mutation'] == 'Wildtype']\n",
    "        \n",
    "        # Check for a lack of data at the site\n",
    "        if all_mut.empty:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "        if wt.empty:\n",
    "            print(\"Not enough wildtype values for site \" + site)\n",
    "            continue\n",
    "        if len(all_mut[site]) < 2 or len(missense_mut[site]) < 2 or len(truncating_mut[site]) < 2:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "            \n",
    "        # Perform the ttests\n",
    "        all_mut_ttest = scipy.stats.ttest_ind(all_mut[site], wt[site])\n",
    "        missense_ttest = scipy.stats.ttest_ind(missense_mut[site], wt[site])\n",
    "        truncating_mut_ttest = scipy.stats.ttest_ind(truncating_mut[site], wt[site])\n",
    "        if all_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" all mutations vs wildtype: \" + str(all_mut_ttest[1]))\n",
    "            plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "            added_to_plot = True\n",
    "        else: print(site + \" all mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if missense_ttest[1] < pcutoff:\n",
    "            print(site + \" missense mutations vs wildtype: \" + str(missense_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "                added_to_plot = True\n",
    "        else: print(site + \" missense mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if truncating_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" truncating mutations vs wildtype: \" + str(truncating_mut_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "        else: print(site + \" truncating mutations vs wildtype: not significant\")\n",
    "            \n",
    "    print('\\n')\n",
    "    \n",
    "plotdf = pd.melt(plotdf, id_vars = ['Mutation'], var_name = 'Site', value_name = 'Phos')\n",
    "plotdf = plotdf.dropna(axis = 0)\n",
    "            \n",
    "a4_dims = (23.4, 16.54)\n",
    "fig, ax = plt.subplots(figsize=a4_dims)\n",
    "\n",
    "#Create the plot\n",
    "boxplt = sns.boxplot(x='Site', y='Phos', hue='Mutation', hue_order = ['Wildtype', 'Missense', 'Truncating'], data=plotdf)\n",
    "boxplt = sns.stripplot(data=plotdf, x='Site', y='Phos', hue='Mutation', hue_order = ['Wildtype', 'Missense', 'Truncating'], dodge=True, jitter=True, color='.3')\n",
    "\n",
    "#Add styling\n",
    "boxplt.set_xlabel('Phosphorylation site on ' + protein, fontsize = '30')\n",
    "boxplt.set_ylabel('Phosphorylation', fontsize='40')\n",
    "boxplt.tick_params(labelsize='20')\n",
    "\n",
    "#Adjust legend\n",
    "handles, labels = ax.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], ['Wildtype', 'Missense', 'Truncating'], title='TP53 Mutation Type', fontsize='20', frameon=False)\n",
    "boxplt.get_legend().get_title().set_fontsize('20')\n",
    "boxplt.get_legend().set_bbox_to_anchor((1, 1, 0, 0))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## XPO1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'XPO1'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## XRN2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'XRN2'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPX2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Proteomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'TPX2'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Phosphoproteomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein = 'TPX2'\n",
    "pattern = re.compile(protein)\n",
    "\n",
    "# 1. Create a dataframe combining phosphosite data with TP53 mutation status; select for tumor samples\n",
    "p53_mutations_protein = CPTAC.compare_mutations(phos, protein, gene)\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].drop('Sample_Status', axis=1)\n",
    "\n",
    "# 2. Make a new column for mutation location relative to hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype', 'Within_Hotspot'] = 'Wildtype'\n",
    "\n",
    "# 3. Begin constructing a dataframe for plotting, starting only with the mutation location\n",
    "plotdf = p53_mutations_protein[['Within_Hotspot']]\n",
    "\n",
    "# 4. Set a cutoff for significance\n",
    "pcutoff = 0.05/len(p53_mutations_protein.columns)\n",
    "\n",
    "# 5. Go through each site on the protein and check for significance with TP53 mutation\n",
    "for site in p53_mutations_protein.columns:\n",
    "    added_to_plot = False\n",
    "    if (site != 'Within_Hotspot' and site != 'Mutation'):\n",
    "        # Set up ttests for each of the hotspot/non-hotspot/wt comparisons\n",
    "        sitedf = p53_mutations_protein[[site, 'Within_Hotspot']].dropna(axis=0)\n",
    "        all_mut = sitedf.loc[sitedf['Within_Hotspot'] != 'Wildtype']\n",
    "        hotspot_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Within Hotspot']\n",
    "        other_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Outside Hotspot']\n",
    "        wt = sitedf.loc[sitedf['Within_Hotspot'] == 'Wildtype']\n",
    "        \n",
    "        # Check for a lack of data at the site\n",
    "        if all_mut.empty:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "        if wt.empty:\n",
    "            print(\"Not enough wildtype values for site \" + site)\n",
    "            continue\n",
    "        if len(all_mut[site]) < 2 or len(hotspot_mut[site]) < 2 or len(other_mut[site]) < 2:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "            \n",
    "        # Perform the ttests\n",
    "        all_mut_ttest = scipy.stats.ttest_ind(all_mut[site], wt[site])\n",
    "        htspt_ttest = scipy.stats.ttest_ind(hotspot_mut[site], wt[site])\n",
    "        other_mut_ttest = scipy.stats.ttest_ind(other_mut[site], wt[site])\n",
    "        if all_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" all mutations vs wildtype: \" + str(all_mut_ttest[1]))\n",
    "            plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "            added_to_plot = True\n",
    "        else: print(site + \" all mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if htspt_ttest[1] < pcutoff:\n",
    "            print(site + \" hotspot mutations vs wildtype: \" + str(htspt_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "                added_to_plot = True\n",
    "        else: print(site + \" hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if other_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" non-hotspot mutations vs wildtype: \" + str(other_mut_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "        else: print(site + \" non-hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "    print('\\n')\n",
    "    \n",
    "plotdf = pd.melt(plotdf, id_vars = ['Within_Hotspot'], var_name = 'Site', value_name = 'Phos')\n",
    "plotdf = plotdf.dropna(axis = 0)\n",
    "            \n",
    "a4_dims = (23.4, 16.54)\n",
    "fig, ax = plt.subplots(figsize=a4_dims)\n",
    "\n",
    "#Create the plot\n",
    "boxplt = sns.boxplot(x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], data=plotdf)\n",
    "boxplt = sns.stripplot(data=plotdf, x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], dodge=True, jitter=True, color='.3')\n",
    "\n",
    "#Add styling\n",
    "boxplt.set_xlabel('Phosphorylation site on ' + protein, fontsize = '30')\n",
    "boxplt.set_ylabel('Phosphorylation', fontsize='40')\n",
    "boxplt.tick_params(labelsize='20')\n",
    "\n",
    "#Adjust legend\n",
    "handles, labels = ax.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], title='Mutation Within Hotspot', fontsize='20', frameon=False)\n",
    "boxplt.get_legend().get_title().set_fontsize('20')\n",
    "boxplt.get_legend().set_bbox_to_anchor((1, 1, 0, 0))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLK1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Proteomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'PLK1'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Phosphoproteomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein = 'PLK1'\n",
    "pattern = re.compile(protein)\n",
    "\n",
    "# 1. Create a dataframe combining phosphosite data with TP53 mutation status; select for tumor samples\n",
    "p53_mutations_protein = CPTAC.compare_mutations(phos, protein, gene)\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].drop('Sample_StatusPatient_Type', axis=1)\n",
    "\n",
    "# 2. Make a new column for mutation location relative to hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "\n",
    "# 3. Begin constructing a dataframe for plotting, starting only with the mutation location\n",
    "plotdf = p53_mutations_protein[['Within_Hotspot']]\n",
    "\n",
    "# 4. Set a cutoff for significance\n",
    "pcutoff = 0.05/len(p53_mutations_protein.columns)\n",
    "\n",
    "# 5. Go through each site on the protein and check for significance with TP53 mutation\n",
    "for site in p53_mutations_protein.columns:\n",
    "    added_to_plot = False\n",
    "    if (site != 'Within_Hotspot' and site != 'Mutation'):\n",
    "        # Set up ttests for each of the hotspot/non-hotspot/wt comparisons\n",
    "        sitedf = p53_mutations_protein[[site, 'Within_Hotspot']].dropna(axis=0)\n",
    "        all_mut = sitedf.loc[sitedf['Within_Hotspot'] != 'Wildtype']\n",
    "        hotspot_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Within Hotspot']\n",
    "        other_mut = sitedf.loc[sitedf['Within_Hotspot'] == 'Outside Hotspot']\n",
    "        wt = sitedf.loc[sitedf['Within_Hotspot'] == 'Wildtype']\n",
    "        \n",
    "        # Check for a lack of data at the site\n",
    "        if all_mut.empty:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "        if wt.empty:\n",
    "            print(\"Not enough wildtype values for site \" + site)\n",
    "            continue\n",
    "        if len(all_mut[site]) < 2 or len(hotspot_mut[site]) < 2 or len(other_mut[site]) < 2:\n",
    "            print(\"Not enough mutated values for site \" + site)\n",
    "            continue\n",
    "            \n",
    "        # Perform the ttests\n",
    "        all_mut_ttest = scipy.stats.ttest_ind(all_mut[site], wt[site])\n",
    "        htspt_ttest = scipy.stats.ttest_ind(hotspot_mut[site], wt[site])\n",
    "        other_mut_ttest = scipy.stats.ttest_ind(other_mut[site], wt[site])\n",
    "        if all_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" all mutations vs hotspot: \" + str(all_mut_ttest[1]))\n",
    "            plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "            added_to_plot = True\n",
    "        else: print(site + \" all mutations vs hotspot: not significant\")\n",
    "            \n",
    "        if htspt_ttest[1] < pcutoff:\n",
    "            print(site + \" hotspot mutations vs wildtype: \" + str(htspt_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "                added_to_plot = True\n",
    "        else: print(site + \" hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "        if other_mut_ttest[1] < pcutoff:\n",
    "            print(site + \" non-hotspot mutations vs wildtype: \" + str(other_mut_ttest[1]))\n",
    "            if added_to_plot is False:\n",
    "                plotdf = plotdf.join(p53_mutations_protein[site])\n",
    "        else: print(site + \" non-hotspot mutations vs wildtype: not significant\")\n",
    "            \n",
    "    print('\\n')\n",
    "    \n",
    "plotdf = pd.melt(plotdf, id_vars = ['Within_Hotspot'], var_name = 'Site', value_name = 'Phos')\n",
    "plotdf = plotdf.dropna(axis = 0)\n",
    "            \n",
    "a4_dims = (23.4, 16.54)\n",
    "fig, ax = plt.subplots(figsize=a4_dims)\n",
    "\n",
    "#Create the plot\n",
    "boxplt = sns.boxplot(x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], data=plotdf)\n",
    "boxplt = sns.stripplot(data=plotdf, x='Site', y='Phos', hue='Within_Hotspot', hue_order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], dodge=True, jitter=True, color='.3')\n",
    "\n",
    "#Add styling\n",
    "boxplt.set_xlabel('Phosphorylation site on ' + protein, fontsize = '30')\n",
    "boxplt.set_ylabel('Phosphorylation', fontsize='40')\n",
    "boxplt.tick_params(labelsize='20')\n",
    "\n",
    "#Adjust legend\n",
    "handles, labels = ax.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], title='Mutation Within Hotspot', fontsize='20', frameon=False)\n",
    "boxplt.get_legend().get_title().set_fontsize('20')\n",
    "boxplt.get_legend().set_bbox_to_anchor((1, 1, 0, 0))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CABLES1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'CABLES1'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDK1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'TP53'\n",
    "protein = 'CDK1'\n",
    "\n",
    "#1.  merge the mutation status of P53 and the proteomics data of P53\n",
    "p53_mutations_protein = CPTAC.compare_mutations(proteomics, protein, gene)\n",
    "#2. now set up a new column for mutations that are within the hotspot\n",
    "p53_mutations_protein['Within_Hotspot'] = 'Outside Hotspot'\n",
    "p53_mutations_protein.loc[hotspot_patients, 'Within_Hotspot'] = 'Within Hotspot'\n",
    "p53_mutations_protein.loc[p53_mutations_protein['Mutation'] == 'Wildtype_Tumor', 'Within_Hotspot'] = 'Wildtype'\n",
    "p53_mutations_protein = p53_mutations_protein.loc[p53_mutations_protein['Sample_Status'] == 'Tumor'].dropna(axis=0)\n",
    "\n",
    "#3.  now make two new DFs, one for people with a hotspot mutation, one for all the wildtype cancer patients.\n",
    "allmutated = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] != 'Wildtype']\n",
    "mutatedhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Within Hotspot']\n",
    "wt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Wildtype']\n",
    "nonhtspt = p53_mutations_protein.loc[p53_mutations_protein['Within_Hotspot'] == 'Outside Hotspot']\n",
    "\n",
    "ttestall = scipy.stats.ttest_ind(allmutated[protein], wt[protein])\n",
    "ttesthtspt = scipy.stats.ttest_ind(mutatedhtspt[protein], wt[protein])\n",
    "ttest_non_htspt = scipy.stats.ttest_ind(nonhtspt[protein], wt[protein])\n",
    "print(\"All TP53 mutation and \" + protein + \": \" + str(ttestall[1]))\n",
    "print(\"Hotspot TP53 mutation and \" + protein + \": \" + str(ttesthtspt[1]))\n",
    "print(\"Non-hotspot TP53 mutation and \" + protein + \": \" + str(ttest_non_htspt[1]))\n",
    "    \n",
    "a4_dims = (11.7, 8.27)\n",
    "fig, ax = plt.subplots(figsize = a4_dims)\n",
    "boxplt = sns.boxplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], ax = ax)\n",
    "boxplt = sns.stripplot(data = p53_mutations_protein, x = 'Within_Hotspot', y = protein, order = ['Wildtype', 'Within Hotspot', 'Outside Hotspot'], jitter = True, color = '.3')\n",
    "boxplt.set_xlabel('Mutation Location', fontsize='20')\n",
    "boxplt.set_ylabel('Protein Expression', fontsize='20')\n",
    "boxplt.set_title(\"Effect of \" + gene + \" mutation on \" + protein + \" protein levels\", fontsize='22')\n",
    "plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
